<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Bill Cresko">

<title>Week 4a - Statistics for Bioengineering – BioEngineering Statistics Spring 2025</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">BioEngineering Statistics Spring 2025</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Syllabus.html"> 
<span class="menu-text">Syllabus</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Policies.html"> 
<span class="menu-text">Policies</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Schedule.html"> 
<span class="menu-text">Schedule</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Lectures.html"> 
<span class="menu-text">Lectures</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Homework.html"> 
<span class="menu-text">Homeworks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Software.html"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Resources.html"> 
<span class="menu-text">Resources</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#sampling-parameter-estimation-error-and-power" id="toc-sampling-parameter-estimation-error-and-power" class="nav-link active" data-scroll-target="#sampling-parameter-estimation-error-and-power">Sampling, parameter estimation, error and power</a>
  <ul class="collapse">
  <li><a href="#this-week" id="toc-this-week" class="nav-link" data-scroll-target="#this-week">This week</a></li>
  <li><a href="#poisson-probability-distribution" id="toc-poisson-probability-distribution" class="nav-link" data-scroll-target="#poisson-probability-distribution"><strong>Poisson Probability Distribution</strong></a></li>
  <li><a href="#poisson-probability-distribution-1" id="toc-poisson-probability-distribution-1" class="nav-link" data-scroll-target="#poisson-probability-distribution-1">Poisson Probability Distribution</a></li>
  <li><a href="#poisson-probability-distribution-2" id="toc-poisson-probability-distribution-2" class="nav-link" data-scroll-target="#poisson-probability-distribution-2">Poisson Probability Distribution</a></li>
  <li><a href="#poisson-probability-distribution-with-gene-length-by-bins-of-500-nucleotides" id="toc-poisson-probability-distribution-with-gene-length-by-bins-of-500-nucleotides" class="nav-link" data-scroll-target="#poisson-probability-distribution-with-gene-length-by-bins-of-500-nucleotides">Poisson Probability Distribution with gene length by bins of 500 nucleotides</a></li>
  <li><a href="#poisson-probability-distribution-with-increasing-parameter-values-of-lambda" id="toc-poisson-probability-distribution-with-increasing-parameter-values-of-lambda" class="nav-link" data-scroll-target="#poisson-probability-distribution-with-increasing-parameter-values-of-lambda">Poisson Probability Distribution with increasing parameter values of <span class="math inline">\(\lambda\)</span></a></li>
  <li><a href="#testing-poisson-distributions" id="toc-testing-poisson-distributions" class="nav-link" data-scroll-target="#testing-poisson-distributions">Testing Poisson Distributions</a></li>
  </ul></li>
  <li><a href="#some-other-common-probability-distributions" id="toc-some-other-common-probability-distributions" class="nav-link" data-scroll-target="#some-other-common-probability-distributions">Some other common probability distributions</a>
  <ul class="collapse">
  <li><a href="#geometric-distribution" id="toc-geometric-distribution" class="nav-link" data-scroll-target="#geometric-distribution"><strong>Geometric Distribution</strong></a></li>
  <li><a href="#geometric-distribution-1" id="toc-geometric-distribution-1" class="nav-link" data-scroll-target="#geometric-distribution-1"><strong>Geometric Distribution</strong></a></li>
  <li><a href="#testing-geometric-distributions" id="toc-testing-geometric-distributions" class="nav-link" data-scroll-target="#testing-geometric-distributions">Testing Geometric Distributions</a></li>
  <li><a href="#testing-geometric-distributions-1" id="toc-testing-geometric-distributions-1" class="nav-link" data-scroll-target="#testing-geometric-distributions-1">Testing Geometric Distributions</a></li>
  <li><a href="#negative-binomial-distribution" id="toc-negative-binomial-distribution" class="nav-link" data-scroll-target="#negative-binomial-distribution"><strong>Negative Binomial Distribution</strong></a></li>
  <li><a href="#negative-binomial-distribution-1" id="toc-negative-binomial-distribution-1" class="nav-link" data-scroll-target="#negative-binomial-distribution-1"><strong>Negative Binomial Distribution</strong></a></li>
  </ul></li>
  <li><a href="#continuous-probability-distributions" id="toc-continuous-probability-distributions" class="nav-link" data-scroll-target="#continuous-probability-distributions">Continuous Probability Distributions</a>
  <ul class="collapse">
  <li><a href="#continuous-probability-distributions-1" id="toc-continuous-probability-distributions-1" class="nav-link" data-scroll-target="#continuous-probability-distributions-1"><strong>Continuous probability distributions</strong></a></li>
  <li><a href="#continuous-probabilities" id="toc-continuous-probabilities" class="nav-link" data-scroll-target="#continuous-probabilities">Continuous probabilities</a></li>
  <li><a href="#uniform-distribution" id="toc-uniform-distribution" class="nav-link" data-scroll-target="#uniform-distribution"><strong>Uniform Distribution</strong></a></li>
  <li><a href="#uniform-distribution-1" id="toc-uniform-distribution-1" class="nav-link" data-scroll-target="#uniform-distribution-1"><strong>Uniform Distribution</strong></a></li>
  <li><a href="#exponential-distribution" id="toc-exponential-distribution" class="nav-link" data-scroll-target="#exponential-distribution">Exponential Distribution</a></li>
  <li><a href="#exponential-distribution-1" id="toc-exponential-distribution-1" class="nav-link" data-scroll-target="#exponential-distribution-1">Exponential Distribution</a></li>
  <li><a href="#exponential-distribution-2" id="toc-exponential-distribution-2" class="nav-link" data-scroll-target="#exponential-distribution-2">Exponential Distribution</a></li>
  <li><a href="#gamma-distribution" id="toc-gamma-distribution" class="nav-link" data-scroll-target="#gamma-distribution"><strong>Gamma Distribution</strong></a></li>
  <li><a href="#gamma-distribution-1" id="toc-gamma-distribution-1" class="nav-link" data-scroll-target="#gamma-distribution-1">Gamma Distribution</a></li>
  </ul></li>
  <li><a href="#normal-gaussian-probability-distribution" id="toc-normal-gaussian-probability-distribution" class="nav-link" data-scroll-target="#normal-gaussian-probability-distribution">Normal (Gaussian) probability distribution</a>
  <ul class="collapse">
  <li><a href="#normal-pdf---mu-and-sigma" id="toc-normal-pdf---mu-and-sigma" class="nav-link" data-scroll-target="#normal-pdf---mu-and-sigma">Normal PDF - (<span class="math inline">\(\mu\)</span> and <span class="math inline">\(\sigma\)</span>)</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"></a></li>
  <li><a href="#section-1" id="toc-section-1" class="nav-link" data-scroll-target="#section-1"></a></li>
  <li><a href="#estimates-of-mean-and-variance" id="toc-estimates-of-mean-and-variance" class="nav-link" data-scroll-target="#estimates-of-mean-and-variance">Estimates of mean and variance</a></li>
  <li><a href="#normal-pdf" id="toc-normal-pdf" class="nav-link" data-scroll-target="#normal-pdf">Normal PDF</a></li>
  <li><a href="#why-is-the-normal-special-in-biosciences" id="toc-why-is-the-normal-special-in-biosciences" class="nav-link" data-scroll-target="#why-is-the-normal-special-in-biosciences">Why is the Normal special in biosciences?</a></li>
  <li><a href="#why-is-the-normal-special-in-biosciences-1" id="toc-why-is-the-normal-special-in-biosciences-1" class="nav-link" data-scroll-target="#why-is-the-normal-special-in-biosciences-1">Why is the Normal special in biosciences?</a></li>
  <li><a href="#why-is-the-normal-special-in-biosciences-2" id="toc-why-is-the-normal-special-in-biosciences-2" class="nav-link" data-scroll-target="#why-is-the-normal-special-in-biosciences-2">Why is the Normal special in biosciences?</a></li>
  <li><a href="#parent-offspring-resemblance" id="toc-parent-offspring-resemblance" class="nav-link" data-scroll-target="#parent-offspring-resemblance">Parent-offspring resemblance</a></li>
  <li><a href="#genetic-model-of-complex-traits" id="toc-genetic-model-of-complex-traits" class="nav-link" data-scroll-target="#genetic-model-of-complex-traits">Genetic model of complex traits</a></li>
  <li><a href="#distribution-of-f_2-genotypes-really-just-binomial-sampling" id="toc-distribution-of-f_2-genotypes-really-just-binomial-sampling" class="nav-link" data-scroll-target="#distribution-of-f_2-genotypes-really-just-binomial-sampling">Distribution of <span class="math inline">\(F_2\)</span> genotypes | really just binomial sampling</a></li>
  <li><a href="#why-else-is-the-normal-special" id="toc-why-else-is-the-normal-special" class="nav-link" data-scroll-target="#why-else-is-the-normal-special">Why else is the Normal special?</a></li>
  <li><a href="#z-scores-of-normal-variables" id="toc-z-scores-of-normal-variables" class="nav-link" data-scroll-target="#z-scores-of-normal-variables">z-scores of normal variables</a></li>
  <li><a href="#z-scores-of-normal-variables-1" id="toc-z-scores-of-normal-variables-1" class="nav-link" data-scroll-target="#z-scores-of-normal-variables-1">z-scores of normal variables</a></li>
  <li><a href="#common-theme-in-r-for-distributions" id="toc-common-theme-in-r-for-distributions" class="nav-link" data-scroll-target="#common-theme-in-r-for-distributions">Common theme in R for distributions</a></li>
  </ul></li>
  <li><a href="#sampling-and-estimation" id="toc-sampling-and-estimation" class="nav-link" data-scroll-target="#sampling-and-estimation">Sampling and estimation</a>
  <ul class="collapse">
  <li><a href="#parameter-estimation" id="toc-parameter-estimation" class="nav-link" data-scroll-target="#parameter-estimation">Parameter Estimation</a></li>
  <li><a href="#parameter-estimation-1" id="toc-parameter-estimation-1" class="nav-link" data-scroll-target="#parameter-estimation-1">Parameter Estimation</a></li>
  <li><a href="#accuracy-vs.-precision" id="toc-accuracy-vs.-precision" class="nav-link" data-scroll-target="#accuracy-vs.-precision">Accuracy vs.&nbsp;Precision</a></li>
  <li><a href="#accuracy-vs.-precision-1" id="toc-accuracy-vs.-precision-1" class="nav-link" data-scroll-target="#accuracy-vs.-precision-1">Accuracy vs.&nbsp;Precision</a></li>
  </ul></li>
  <li><a href="#the-central-limit-theorem" id="toc-the-central-limit-theorem" class="nav-link" data-scroll-target="#the-central-limit-theorem">The Central Limit Theorem</a>
  <ul class="collapse">
  <li><a href="#the-central-limit-theorem-1" id="toc-the-central-limit-theorem-1" class="nav-link" data-scroll-target="#the-central-limit-theorem-1">The Central Limit Theorem</a></li>
  <li><a href="#standard-error-of-the-mean-sem" id="toc-standard-error-of-the-mean-sem" class="nav-link" data-scroll-target="#standard-error-of-the-mean-sem">Standard Error of the Mean (SEM)</a></li>
  <li><a href="#calculating-the-standard-error-of-the-mean-sem" id="toc-calculating-the-standard-error-of-the-mean-sem" class="nav-link" data-scroll-target="#calculating-the-standard-error-of-the-mean-sem">Calculating the Standard Error of the Mean (SEM)</a></li>
  <li><a href="#estimation-and-confidence-intervals" id="toc-estimation-and-confidence-intervals" class="nav-link" data-scroll-target="#estimation-and-confidence-intervals">Estimation and confidence intervals</a></li>
  <li><a href="#estimation-and-confidence-intervals-1" id="toc-estimation-and-confidence-intervals-1" class="nav-link" data-scroll-target="#estimation-and-confidence-intervals-1">Estimation and confidence intervals</a></li>
  <li><a href="#estimation-and-confidence-intervals-2" id="toc-estimation-and-confidence-intervals-2" class="nav-link" data-scroll-target="#estimation-and-confidence-intervals-2">Estimation and confidence intervals</a></li>
  <li><a href="#confidence-intervals" id="toc-confidence-intervals" class="nav-link" data-scroll-target="#confidence-intervals">Confidence Intervals</a></li>
  <li><a href="#coefficient-of-variation" id="toc-coefficient-of-variation" class="nav-link" data-scroll-target="#coefficient-of-variation">Coefficient of Variation</a></li>
  </ul></li>
  <li><a href="#basics-of-bootstrapping-via-re-sampling" id="toc-basics-of-bootstrapping-via-re-sampling" class="nav-link" data-scroll-target="#basics-of-bootstrapping-via-re-sampling">Basics of bootstrapping via re-sampling</a>
  <ul class="collapse">
  <li><a href="#bootstrapping" id="toc-bootstrapping" class="nav-link" data-scroll-target="#bootstrapping">Bootstrapping</a></li>
  <li><a href="#the-etymology-of-the-term-bootstrap" id="toc-the-etymology-of-the-term-bootstrap" class="nav-link" data-scroll-target="#the-etymology-of-the-term-bootstrap">The etymology of the term ‘bootstrap’</a></li>
  <li><a href="#why-the-bootstrap-is-good" id="toc-why-the-bootstrap-is-good" class="nav-link" data-scroll-target="#why-the-bootstrap-is-good">Why the bootstrap is good</a></li>
  <li><a href="#easy-steps-for-bootstrapping-in-r" id="toc-easy-steps-for-bootstrapping-in-r" class="nav-link" data-scroll-target="#easy-steps-for-bootstrapping-in-r">Easy steps for bootstrapping in R</a></li>
  <li><a href="#bootstrapping-1" id="toc-bootstrapping-1" class="nav-link" data-scroll-target="#bootstrapping-1">Bootstrapping</a></li>
  </ul></li>
  <li><a href="#loops-in-r---a-good-way-to-create-even-more-complicated-null-distributions" id="toc-loops-in-r---a-good-way-to-create-even-more-complicated-null-distributions" class="nav-link" data-scroll-target="#loops-in-r---a-good-way-to-create-even-more-complicated-null-distributions">Loops in R - a good way to create even more complicated null distributions</a>
  <ul class="collapse">
  <li><a href="#loops-in-r" id="toc-loops-in-r" class="nav-link" data-scroll-target="#loops-in-r">Loops in R</a></li>
  <li><a href="#for-loops" id="toc-for-loops" class="nav-link" data-scroll-target="#for-loops">For loops</a></li>
  <li><a href="#while-loops" id="toc-while-loops" class="nav-link" data-scroll-target="#while-loops">While loops</a></li>
  <li><a href="#if-and-else-statements" id="toc-if-and-else-statements" class="nav-link" data-scroll-target="#if-and-else-statements">If and Else Statements</a></li>
  <li><a href="#a-programming-joke-for-conditional-statements" id="toc-a-programming-joke-for-conditional-statements" class="nav-link" data-scroll-target="#a-programming-joke-for-conditional-statements">A programming joke for conditional statements</a></li>
  <li><a href="#r-interlude---simulate-a-population-and-sample-it" id="toc-r-interlude---simulate-a-population-and-sample-it" class="nav-link" data-scroll-target="#r-interlude---simulate-a-population-and-sample-it">R INTERLUDE - Simulate a population and sample it!</a></li>
  <li><a href="#simulations-to-compare-parameter-estimates" id="toc-simulations-to-compare-parameter-estimates" class="nav-link" data-scroll-target="#simulations-to-compare-parameter-estimates">Simulations to compare parameter estimates</a></li>
  <li><a href="#calculating-parameters" id="toc-calculating-parameters" class="nav-link" data-scroll-target="#calculating-parameters">Calculating parameters</a></li>
  <li><a href="#sampling-exercise" id="toc-sampling-exercise" class="nav-link" data-scroll-target="#sampling-exercise">Sampling Exercise</a></li>
  <li><a href="#randomness-in-sampling" id="toc-randomness-in-sampling" class="nav-link" data-scroll-target="#randomness-in-sampling">Randomness in Sampling</a></li>
  <li><a href="#surveying-your-sampling" id="toc-surveying-your-sampling" class="nav-link" data-scroll-target="#surveying-your-sampling">Surveying your Sampling</a></li>
  <li><a href="#histograms" id="toc-histograms" class="nav-link" data-scroll-target="#histograms">Histograms</a></li>
  <li><a href="#surveying-your-sampling-1" id="toc-surveying-your-sampling-1" class="nav-link" data-scroll-target="#surveying-your-sampling-1">Surveying your Sampling</a></li>
  <li><a href="#challenge---bootstrapping-to-produce-a-confidence-interval" id="toc-challenge---bootstrapping-to-produce-a-confidence-interval" class="nav-link" data-scroll-target="#challenge---bootstrapping-to-produce-a-confidence-interval">CHALLENGE - Bootstrapping to produce a confidence interval</a></li>
  </ul></li>
  <li><a href="#section-2" id="toc-section-2" class="nav-link" data-scroll-target="#section-2"></a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="Week4a.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Week 4a - Statistics for Bioengineering</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Bill Cresko </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nycflights23)</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_minimal</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="sampling-parameter-estimation-error-and-power" class="level1">
<h1>Sampling, parameter estimation, error and power</h1>
<section id="this-week" class="level2">
<h2 class="anchored" data-anchor-id="this-week">This week</h2>
<ul>
<li><p>Finish parameter estimation</p></li>
<li><p>t-test</p></li>
<li><p>Hypothesis testing</p></li>
<li><p>Power</p></li>
<li><p>Start linear models</p></li>
</ul>
</section>
<section id="poisson-probability-distribution" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="poisson-probability-distribution"><strong>Poisson Probability Distribution</strong></h2>
<ul>
<li><p>Another common situation in bioengineering is when each trial is discrete, but the number of observations of each outcome is observed/counted</p></li>
<li><p>Some examples are</p>
<ul>
<li>counts of snails in several plots of land</li>
<li>observations of the firing of a neuron in a unit of time</li>
<li>the number of muscle contractions during a bout of exercise</li>
<li>count of genes in a genome binned to units of 500kb</li>
</ul></li>
<li><p>Just like before you have ‘successes’, but</p>
<ul>
<li>now you count them for each replicate</li>
<li>the replicates now are units of area or time</li>
<li>the values can now range from 0 to a large number</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="poisson-probability-distribution-1">Poisson Probability Distribution</h2>
<p><br></p>
<ul>
<li>For example, you can examine 100 participants
<ul>
<li>count the number of jumps they can perform in 2 minutes</li>
<li>what is the probability of observing a particpant with <code>r</code> jumps?</li>
</ul></li>
<li><code>Pr(Y=r)</code> is the probability that the number of occurrences of an event <code>y</code> equals a count <code>r</code> in the total number of trials</li>
</ul>
<p><br></p>
<p><span class="math display">\[Pr(Y=r) = \frac{e^{-\mu}\mu^r}{r!}\]</span></p>
</section>
<section id="poisson-probability-distribution-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="poisson-probability-distribution-2">Poisson Probability Distribution</h2>
<ul>
<li>Note that the Poisson is a single parameter function because <span class="math inline">\(\mu = \sigma^2\)</span></li>
<li>The two together are often just represented by <span class="math inline">\(\lambda\)</span></li>
</ul>
<p><span class="math display">\[Pr(y=r) = \frac{e^{-\lambda}\lambda^r}{r!}\]</span></p>
<ul>
<li>This means that for a variable that is truly Poisson distributed:
<ul>
<li>the mean and variance should be equal to one another</li>
<li>variables that are approximately Poisson distributed but have a larger variance than mean are often called ‘overdispersed’</li>
<li>quite common in RNA-seq and microbiome data</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-with-gene-length-by-bins-of-500-nucleotides" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="poisson-probability-distribution-with-gene-length-by-bins-of-500-nucleotides">Poisson Probability Distribution with gene length by bins of 500 nucleotides</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.004.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="poisson-probability-distribution-with-increasing-parameter-values-of-lambda" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="poisson-probability-distribution-with-increasing-parameter-values-of-lambda">Poisson Probability Distribution with increasing parameter values of <span class="math inline">\(\lambda\)</span></h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.005.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="testing-poisson-distributions" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="testing-poisson-distributions">Testing Poisson Distributions</h2>
<p>Number of counts (x) given a mean and variance of lambda</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dpois</span>(<span class="at">x=</span><span class="dv">2</span>, <span class="at">lambda=</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1839397</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">dpois</span>(<span class="at">x=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">lambda=</span><span class="dv">3</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:40.0%"></p>
</figure>
</div>
</div>
</div>
<hr>
</section>
</section>
<section id="some-other-common-probability-distributions" class="level1">
<h1>Some other common probability distributions</h1>
<section id="geometric-distribution" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="geometric-distribution"><strong>Geometric Distribution</strong></h2>
<ul>
<li>If a single event has two possible outcomes the probability of having to observe <code>k</code> trials before the first “one” appears is given by the <strong>geometric distribution</strong></li>
<li>The probability that the first “one” would appear on the first trial is <code>p</code>, but the probability that the first “one” appears on the second trial is <code>(1-p)*p</code></li>
<li>By generalizing this procedure, the probability that there will be <code>k-1</code> failures before the first success is:</li>
</ul>
<p><span class="math display">\[P(X=k)=(1-p)^{k-1}p\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{1}{p}\)</span></li>
<li>variance = <span class="math inline">\(\frac{(1-p)}{p^2}\)</span></li>
</ul>
</section>
<section id="geometric-distribution-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="geometric-distribution-1"><strong>Geometric Distribution</strong></h2>
<ul>
<li>Example: If the probability of extinction of an endangered population is estimated to be 0.1 every year, what is the expected time until extinction?</li>
<li>The distribution gives the probability of extinction in a given year (requiring that the population did not go extinct in all of the years prior)</li>
<li>If we want to know the probability of the population going exticnt by year 4, we simply add up the probabilities for years 1-3 using “or” rules</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.017.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="testing-geometric-distributions" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="testing-geometric-distributions">Testing Geometric Distributions</h2>
<ul>
<li>dgeom gives the density (probability) of an event (p) after (x) failures</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dgeom</span>(<span class="at">x=</span><span class="dv">20</span>, <span class="at">p=</span><span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.01215767</code></pre>
</div>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">dgeom</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,<span class="fl">0.1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-6-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:40.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="testing-geometric-distributions-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="testing-geometric-distributions-1">Testing Geometric Distributions</h2>
<ul>
<li>pgeom gives the cumulative probability of event (p) in (q) trials</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pgeom</span>(<span class="at">q=</span><span class="dv">20</span>, <span class="at">p=</span><span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.890581</code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">pgeom</span>(<span class="at">q=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>, <span class="at">p=</span><span class="fl">0.1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="negative-binomial-distribution" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="negative-binomial-distribution"><strong>Negative Binomial Distribution</strong></h2>
<ul>
<li>Extension of the geometric distribution describing the waiting time until <code>r</code> “ones” have appeared.</li>
<li>Generalizes the geometric distribution</li>
<li>Probability of the <span class="math inline">\(r^{th}\)</span> “one” appearing on the <span class="math inline">\(k^{th}\)</span> trial:</li>
</ul>
<p><span class="math display">\[P(X=k)=(\frac{k-1}{r-1})p^{r-1}(1-p)^{k-r}p\]</span></p>
<p><br></p>
<p>which simplifies to</p>
<p><span class="math display">\[P(X=k)=(\frac{k-1}{r-1})p^{r}(1-p)^{k-r}\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{r}{p}\)</span></li>
<li>variance = <span class="math inline">\(r(1-p)/p^2\)</span></li>
</ul>
</section>
<section id="negative-binomial-distribution-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="negative-binomial-distribution-1"><strong>Negative Binomial Distribution</strong></h2>
<ul>
<li>Example: If a predator must capture 10 prey before it can grow large enough to reproduce</li>
<li>What would the mean age of onset of reproduction be if the probability of capturing a prey on any given day is 0.1?</li>
<li>Notice that the variance is quite high (~1000) and that the distribution looks quite skewed</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.018.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="continuous-probability-distributions" class="level1">
<h1>Continuous Probability Distributions</h1>
<section id="continuous-probability-distributions-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="continuous-probability-distributions-1"><strong>Continuous probability distributions</strong></h2>
<p>P(observation lies within dx of x) = f(x)dx</p>
<p><span class="math display">\[P(a\leq X \leq b) = \int_{a}^{b} f(x) dx\]</span></p>
<p>Remember that the indefinite integral sums to one</p>
<p><span class="math display">\[\int_{-\infty}^{\infty} f(x) dx = 1\]</span></p>
</section>
<section id="continuous-probabilities" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="continuous-probabilities">Continuous probabilities</h2>
<p><br></p>
<p><code>E[X]</code> may be found by integrating the product of <code>x</code> and the probability density function over all possible values of <code>x</code>:</p>
<p><span class="math display">\[E[X] = \int_{-\infty}^{\infty} xf(x) dx \]</span></p>
<p><br></p>
<p><span class="math inline">\(Var(X) = E[X^2] - (E[X])^2\)</span>, where the expectation of <span class="math inline">\(X^2\)</span> is</p>
<p><span class="math display">\[E[X^2] = \int_{-\infty}^{\infty} x^2f(x) dx \]</span></p>
</section>
<section id="uniform-distribution" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="uniform-distribution"><strong>Uniform Distribution</strong></h2>
<p><br></p>
<p><span class="math display">\[E[X] = \int_{a}^{b} x\frac{1}{b-a} dx = \frac{(a+b)}{2} \]</span></p>
<p><br></p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.019.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="uniform-distribution-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="uniform-distribution-1"><strong>Uniform Distribution</strong></h2>
<ul>
<li>Means that the probability is equal for all possible outcomes</li>
<li>Like drawing m&amp;m out of a bag with equal proportions of colors</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dunif</span>(<span class="at">x=</span><span class="dv">1</span>,<span class="at">min=</span><span class="dv">0</span>, <span class="at">max=</span><span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">dunif</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="dv">0</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="exponential-distribution" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="exponential-distribution">Exponential Distribution</h2>
<p><br></p>
<p><span class="math display">\[f(x)=\lambda e^{-\lambda x}\]</span></p>
<p><br></p>
<ul>
<li><code>E[X]</code> can be found be integrating <span class="math inline">\(xf(x)\)</span> from 0 to infinity</li>
</ul>
<p><br></p>
<ul>
<li>leading to the result that</li>
</ul>
<p><br></p>
<ul>
<li><span class="math inline">\(E[X] = \frac{1}{\lambda}\)</span></li>
<li><span class="math inline">\(E[X^2] = \frac{1}{\lambda^2}\)</span></li>
</ul>
</section>
<section id="exponential-distribution-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="exponential-distribution-1">Exponential Distribution</h2>
<ul>
<li>For example, let equal the instantaneous death rate of an individual.</li>
<li>The lifespan of the individual would be described by an exponential distribution (assuming that does not change over time).</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.020.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="exponential-distribution-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="exponential-distribution-2">Exponential Distribution</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dexp</span>(<span class="dv">10</span>, <span class="at">rate =</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.03678794</code></pre>
</div>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(<span class="fu">dexp</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>, <span class="at">rate =</span> <span class="fl">0.1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<hr>
</section>
<section id="gamma-distribution" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="gamma-distribution"><strong>Gamma Distribution</strong></h2>
<ul>
<li>The gamma distribution generalizes the exponential distribution.</li>
<li>It describes the waiting time until the <span class="math inline">\(r^{th}\)</span> event for a process that occurs randomly over time at a rate <span class="math inline">\(\lambda\)</span> :</li>
</ul>
<p><br></p>
<p><span class="math display">\[f(x) = \frac{e^{-\lambda x}\lambda x^{r-1}}{(r-1)!}\lambda\]</span></p>
<p><br></p>
<p><span class="math display">\[ Mean =  \frac{r}{\lambda} \]</span> <span class="math display">\[ Variance = \frac{r}{\lambda^2} \]</span></p>
</section>
<section id="gamma-distribution-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="gamma-distribution-1">Gamma Distribution</h2>
<ul>
<li><p><strong>Example</strong>: If, in a PCR reaction, DNA polymerase synthesizes new DNA strands at a rate of 1 per millisecond, how long until 1000 new DNA strands are produced?</p></li>
<li><p>Assume that DNA synthesis does not deplete the pool of primers or nucleotides in the chamber, so that each event is independent of other events in the PCR chamber.</p></li>
<li><p>The gamma distribution generalizes the exponential distribution.</p></li>
<li><p>It describes the waiting time until the <span class="math inline">\(r^{th}\)</span> event for a process that occurs randomly over time at a rate <span class="math inline">\(\lambda\)</span></p></li>
</ul>
</section>
</section>
<section id="normal-gaussian-probability-distribution" class="level1 smaller">
<h1 class="smaller">Normal (Gaussian) probability distribution</h1>
<section id="normal-pdf---mu-and-sigma" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="normal-pdf---mu-and-sigma">Normal PDF - (<span class="math inline">\(\mu\)</span> and <span class="math inline">\(\sigma\)</span>)</h2>
<p><span class="math display">\[f(x) = \frac{1}{\sqrt{2\pi\sigma}} \, \mathrm{e}^{-\frac{(x - \mu)^2}{2\sigma^2}}\]</span></p>
<p>where <span class="math display">\[\large \pi \approx 3.14159\]</span></p>
<p><span class="math display">\[\large \mathrm{e} \approx 2.71828\]</span></p>
<p>To write that a variable (v) is distributed as a normal distribution with mean <span class="math inline">\(\mu\)</span> and variance <span class="math inline">\(\sigma^2\)</span>, we write the following:</p>
<p><span class="math display">\[\large v \sim \mathcal{N} (\mu,\sigma^2)\]</span></p>
</section>
<section id="section" class="level2">
<h2 class="anchored" data-anchor-id="section"></h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.011.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="section-1" class="level2">
<h2 class="anchored" data-anchor-id="section-1"></h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.012.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="estimates-of-mean-and-variance" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="estimates-of-mean-and-variance">Estimates of mean and variance</h2>
<p>Estimate of the mean from a single sample</p>
<p><span class="math display">\[\Large \bar{x} = \frac{1}{n}\sum_{i=1}^{n}{x_i} \]</span></p>
<p>Estimate of the variance from a single sample</p>
<p><span class="math display">\[\Large s^2 = \frac{1}{n-1}\sum_{i=1}^{n}{(x_i - \bar{x})^2} \]</span></p>
</section>
<section id="normal-pdf" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="normal-pdf">Normal PDF</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.010.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="why-is-the-normal-special-in-biosciences" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="why-is-the-normal-special-in-biosciences">Why is the Normal special in biosciences?</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.013.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="why-is-the-normal-special-in-biosciences-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="why-is-the-normal-special-in-biosciences-1">Why is the Normal special in biosciences?</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.015.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="why-is-the-normal-special-in-biosciences-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="why-is-the-normal-special-in-biosciences-2">Why is the Normal special in biosciences?</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.014.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="parent-offspring-resemblance" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="parent-offspring-resemblance">Parent-offspring resemblance</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.016.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:45.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="genetic-model-of-complex-traits" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="genetic-model-of-complex-traits">Genetic model of complex traits</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.017.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="distribution-of-f_2-genotypes-really-just-binomial-sampling" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="distribution-of-f_2-genotypes-really-just-binomial-sampling">Distribution of <span class="math inline">\(F_2\)</span> genotypes | really just binomial sampling</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.018.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="why-else-is-the-normal-special" class="level2">
<h2 class="anchored" data-anchor-id="why-else-is-the-normal-special">Why else is the Normal special?</h2>
<ul>
<li><p>It is the basis of estimation and precision of the expected value of <strong>all distributions</strong></p></li>
<li><p>Provides a mathematical basis for moving from <strong>single samples to point estimates</strong>.</p></li>
<li><p>Provides a way to use simulation to generate <strong>empirical sample and test distributions</strong> through Monte Carlo approaches</p></li>
</ul>
</section>
<section id="z-scores-of-normal-variables" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="z-scores-of-normal-variables">z-scores of normal variables</h2>
<ul>
<li>Often we want to make variables more comparable to one another</li>
<li>For example, consider measuring the leg length of mice and of elephants
<ul>
<li>Which animal has longer legs in absolute terms?</li>
<li>Proportional to their body size?</li>
</ul></li>
<li>A good way to answer these last questions is to use ‘z-scores’</li>
</ul>
</section>
<section id="z-scores-of-normal-variables-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="z-scores-of-normal-variables-1">z-scores of normal variables</h2>
<ul>
<li>z-scores are standardized to a mean of 0 and a standard deviation of 1</li>
<li>We can modify any normal distribution to have
<ul>
<li>a mean of 0 and</li>
<li>a standard deviation of 1</li>
</ul></li>
<li>Another term for this is the standard normal distribution</li>
</ul>
<p><span class="math display">\[\huge z_i = \frac{(x_i - \bar{x})}{s}\]</span></p>
</section>
<section id="common-theme-in-r-for-distributions" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="common-theme-in-r-for-distributions">Common theme in R for distributions</h2>
<table class="caption-top table">
<colgroup>
<col style="width: 11%">
<col style="width: 22%">
<col style="width: 21%">
<col style="width: 16%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th></th>
<th style="text-align: center;"><p>d</p>
<p>probability mass function</p></th>
<th style="text-align: center;"><p>p</p>
<p>cumulative distribution</p></th>
<th style="text-align: center;"><p>q</p>
<p>quantile function</p></th>
<th style="text-align: center;"><p>r</p>
<p>pseudorandom number generate</p></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>binomial</td>
<td style="text-align: center;">dbinom</td>
<td style="text-align: center;">pbinom</td>
<td style="text-align: center;">qbinom</td>
<td style="text-align: center;">rbinom</td>
</tr>
<tr class="even">
<td>poisson</td>
<td style="text-align: center;">dpois</td>
<td style="text-align: center;">ppois</td>
<td style="text-align: center;">qpois</td>
<td style="text-align: center;">rpois</td>
</tr>
<tr class="odd">
<td>exponential</td>
<td style="text-align: center;">dexp</td>
<td style="text-align: center;">pexp</td>
<td style="text-align: center;">qexp</td>
<td style="text-align: center;">rexp</td>
</tr>
<tr class="even">
<td>normal</td>
<td style="text-align: center;">dnorm</td>
<td style="text-align: center;">pnorm</td>
<td style="text-align: center;">qnorm</td>
<td style="text-align: center;">rnorm</td>
</tr>
</tbody>
</table>
<hr>
</section>
</section>
<section id="sampling-and-estimation" class="level1">
<h1>Sampling and estimation</h1>
<section id="parameter-estimation" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="parameter-estimation">Parameter Estimation</h2>
<ul>
<li>Estimation is the process of inferring a population parameter from sample data</li>
<li>The value of one sample estimate is almost never the same as the population parameter because of random sampling error</li>
<li>Most will be close, but some will be far away</li>
<li>Sampling distribution of an estimate
<ul>
<li>all values we might have obtained from our sample</li>
<li>probabilities of occurrence</li>
</ul></li>
<li>Standard error of an estimate
<ul>
<li>standard deviation of a sampling distribution</li>
<li>measures the precision of the parameter estimate</li>
<li>NO ESTIMATE IS USEFUL WITHOUT IT!</li>
</ul></li>
</ul>
</section>
<section id="parameter-estimation-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="parameter-estimation-1">Parameter Estimation</h2>
<ul>
<li><strong>Parametric</strong> (a few special exceptions, like the sample mean and its standard error)</li>
<li><strong>Resampling</strong> - bootstrapping and randomization to create empirical null distributions</li>
<li><strong>Ordinary Least Squares (OLS)</strong> - optimized procedure that produces one definitive result, easy to use but no estimates of confidence</li>
<li><strong>Maximum Likelihood (ML)</strong> - Can provide model-based estimates with confidence, but harder to calculate</li>
<li><strong>Bayesian Approaches</strong> - Incorporates prior information into ML estimation</li>
</ul>
</section>
<section id="accuracy-vs.-precision" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="accuracy-vs.-precision">Accuracy vs.&nbsp;Precision</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.024.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="accuracy-vs.-precision-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="accuracy-vs.-precision-1">Accuracy vs.&nbsp;Precision</h2>
<ul>
<li><strong>Accuracy</strong> is the closeness of an estimated value to its true value</li>
<li><strong>Precision</strong> is the closeness of repeated estimates to one another</li>
<li>Our goal is to have <strong>unbiased estimates</strong> that are the most precise</li>
<li>We have to <strong>estimate parameters</strong> and <strong>test hypotheses</strong> by taking <strong>samples</strong> that approximate the underlying distribution</li>
<li>The goal of <strong>replication</strong> is to quantify variation at as many levels in a study as possible</li>
<li>The goal of <strong>randomization</strong> is to avoid bias as much as possible</li>
</ul>
</section>
</section>
<section id="the-central-limit-theorem" class="level1 smaller">
<h1 class="smaller">The Central Limit Theorem</h1>
<section id="the-central-limit-theorem-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="the-central-limit-theorem-1">The Central Limit Theorem</h2>
<ul>
<li>For most real world data sets we can’t empirically determine a sampling distribution by taking many actual samples, because we often have just the one sample.</li>
<li>Fortunately, we can rely on the <strong>Central Limit Theorem</strong> to make some assumptions about sampling distributions, particularly when estimating a mean from a single sample, or when estimating most any parameter using a “pseudo” or re-sampling process we refer to as “<strong>bootstrapping</strong>”</li>
</ul>
</section>
<section id="standard-error-of-the-mean-sem" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="standard-error-of-the-mean-sem">Standard Error of the Mean (SEM)</h2>
<p><span class="math display">\[\huge \sigma_{\bar{x}} \approx s_{\bar{x}} = \frac{s}{\sqrt{n}} \]</span></p>
<ul>
<li>where <span class="math inline">\(s_{\bar{x}}\)</span> is the estimated standard error of the distribution of the mean estimates</li>
<li>which is usually just referred to as the ’standard error of the mean (SEM)</li>
<li>note that this <strong>is not</strong> the standard deviation of the original distribution</li>
<li>importantly, the SEM will go down as the sample size increases</li>
</ul>
</section>
<section id="calculating-the-standard-error-of-the-mean-sem" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="calculating-the-standard-error-of-the-mean-sem">Calculating the Standard Error of the Mean (SEM)</h2>
<ul>
<li>Think conceptually - how will SEM change as sample size increases?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">32</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>true_pop <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="at">n=</span><span class="dv">1000</span>, <span class="at">mean=</span><span class="dv">2</span>, <span class="at">sd=</span><span class="dv">5</span>)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>samps_var <span class="ot">&lt;-</span> <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">50</span>, <span class="fu">sample</span>(true_pop, <span class="at">size =</span> <span class="dv">5</span>))</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>samps_var_means <span class="ot">&lt;-</span> <span class="fu">apply</span>(samps_var, <span class="dv">2</span>, mean)</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>sem <span class="ot">=</span> <span class="fu">sd</span>(samps_var_means)<span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">length</span>(samps_var_means))</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sem)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.2646669</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">32</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>true_pop <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">1000</span>, <span class="at">lambda =</span> <span class="dv">5</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>samps_var <span class="ot">&lt;-</span> <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">50</span>, <span class="fu">sample</span>(true_pop, <span class="at">size =</span> <span class="dv">5</span>))</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>samps_var_means <span class="ot">&lt;-</span> <span class="fu">apply</span>(samps_var, <span class="dv">2</span>, mean)</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>sem <span class="ot">=</span> <span class="fu">sd</span>(samps_var_means)<span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">length</span>(samps_var_means))</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sem)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1299607</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">32</span>)</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>true_pop <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">1000</span>, <span class="at">lambda =</span> <span class="dv">5</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>samps_var <span class="ot">&lt;-</span> <span class="fu">replicate</span>(<span class="at">n =</span> <span class="dv">50</span>, <span class="fu">sample</span>(true_pop, <span class="at">size =</span> <span class="dv">50</span>))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a>samps_var_means <span class="ot">&lt;-</span> <span class="fu">apply</span>(samps_var, <span class="dv">2</span>, mean)</span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>sem <span class="ot">=</span> <span class="fu">sd</span>(samps_var_means)<span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">length</span>(samps_var_means))</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(sem)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.04635888</code></pre>
</div>
</div>
</section>
<section id="estimation-and-confidence-intervals" class="level2 vcenter flexbox">
<h2 class="vcenter flexbox anchored" data-anchor-id="estimation-and-confidence-intervals">Estimation and confidence intervals</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.026.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="estimation-and-confidence-intervals-1" class="level2 vcenter flexbox">
<h2 class="vcenter flexbox anchored" data-anchor-id="estimation-and-confidence-intervals-1">Estimation and confidence intervals</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.027.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="estimation-and-confidence-intervals-2" class="level2 vcenter flexbox">
<h2 class="vcenter flexbox anchored" data-anchor-id="estimation-and-confidence-intervals-2">Estimation and confidence intervals</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.028.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="confidence-intervals" class="level2">
<h2 class="anchored" data-anchor-id="confidence-intervals">Confidence Intervals</h2>
<ul>
<li>A confidence interval is a range of values about a parameter estimate, such that we are X% certain that the true population parameter value lies within that interval.</li>
<li>For now, know that for a normally distributed sample, a confidence interval about the population mean can be calculated using the t.test() function in base R.</li>
<li>The 95% confidence interval is commonly reported in statistical analysis results, by convention, but other values are occasionally reported as well.</li>
</ul>
</section>
<section id="coefficient-of-variation" class="level2">
<h2 class="anchored" data-anchor-id="coefficient-of-variation">Coefficient of Variation</h2>
<ul>
<li>To make standard deviations comparable across populations with very different means, we can instead compare a standardized metric called the “coefficient of variation” (CV), which is simply the sample standard deviation divided by the sample mean (and usually expressed as a % by multiplying by 100).</li>
</ul>
</section>
</section>
<section id="basics-of-bootstrapping-via-re-sampling" class="level1">
<h1>Basics of bootstrapping via re-sampling</h1>
<section id="bootstrapping" class="level2">
<h2 class="anchored" data-anchor-id="bootstrapping">Bootstrapping</h2>
<p>-Unfortunately, most other kinds of estimates (anything not the mean) do not have this amazing property, but we can rely on another approach to calculate the standard error. - This involves generating your own sampling distribution for the estimate using the “<strong>bootstrap</strong>,” a method invented by Efron (1979). - We call the bootstrap, and other methods that do not rely on distributional assumptions of the variable itself, “<strong>nonparametric</strong>” approaches.</p>
</section>
<section id="the-etymology-of-the-term-bootstrap" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="the-etymology-of-the-term-bootstrap">The etymology of the term ‘bootstrap’</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.030.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="why-the-bootstrap-is-good" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="why-the-bootstrap-is-good">Why the bootstrap is good</h2>
<ul>
<li>Can be applied to almost any sample statistic
<ul>
<li>This includes means, proportions, correlations, regression</li>
</ul></li>
<li>Works when there is no ready formula for a standard error
<ul>
<li>For example the median, trimmed mean, correlation, eigenvalue, etc.</li>
</ul></li>
<li>Is nonparametric, so doesn’t require normally-distributed data</li>
<li>Works for estimates based on complicated sampling procedures or calculations</li>
</ul>
</section>
<section id="easy-steps-for-bootstrapping-in-r" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="easy-steps-for-bootstrapping-in-r">Easy steps for bootstrapping in R</h2>
<ol type="1">
<li>Take a random sample (with replacement) from your sample data</li>
<li>Calculate the estimate using the measurements in the bootstrap sample (step 1). This is the first bootstrap replicate estimate</li>
<li>Repeat steps 1 and 2 a large number of times (1000 times is reasonable)</li>
<li>Calculate the sample standard deviation of all the bootstrap replicate estimates obtained in step 3 (SSD = sd(sample)/sqrt(sample size))</li>
</ol>
</section>
<section id="bootstrapping-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="bootstrapping-1">Bootstrapping</h2>
<ul>
<li>The resulting quantity is called the “bootstrap standard error”</li>
<li>The bootstrap can be applied to almost any sample statistic, including means, proportions, correlations, and regression parameters.</li>
<li>It works when there is no ready formula for a standard error, for example when estimating the median, trimmed mean, correlation, eigenvalue, etc.</li>
<li>It is nonparametric, so doesn’t require normally-distributed data, as mentioned.</li>
<li>It works well for parameter estimates that are based on complicated sampling procedures or calculations. For example, it is used to assess confidence in local relationships within phylogenetic trees.</li>
</ul>
</section>
</section>
<section id="loops-in-r---a-good-way-to-create-even-more-complicated-null-distributions" class="level1">
<h1>Loops in R - a good way to create even more complicated null distributions</h1>
<section id="loops-in-r" class="level2">
<h2 class="anchored" data-anchor-id="loops-in-r">Loops in R</h2>
<ul>
<li>R is very good at performing repetitive tasks.</li>
<li>If we want a set of operations to be repeated several times we use what’s known as a loop.</li>
<li>When you create a loop, R will execute the instructions in the loop a specified number of times or until a specified condition is met.</li>
<li>There are two common types of loop in R: the <code>for loop</code> and the <code>while loop</code></li>
</ul>
</section>
<section id="for-loops" class="level2">
<h2 class="anchored" data-anchor-id="for-loops">For loops</h2>
<ul>
<li>The most commonly used loop structure when you want to repeat a task a defined number of times is the for loop. The most basic example of a for loop is:</li>
<li>How does this appear to be working?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Notice the sequence of parentheses and brackets used in this example</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>) {</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5</code></pre>
</div>
</div>
</section>
<section id="while-loops" class="level2">
<h2 class="anchored" data-anchor-id="while-loops">While loops</h2>
<ul>
<li>Another type of loop that you may use is the while loop.</li>
<li>The while loop is used when you want to keep looping until a specific logical condition is satisfied.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>i <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="cf">while</span> (i <span class="sc">&lt;=</span> <span class="dv">4</span>) {</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  i <span class="ot">&lt;-</span> i <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 1
[1] 2
[1] 3
[1] 4
[1] 5</code></pre>
</div>
</div>
</section>
<section id="if-and-else-statements" class="level2">
<h2 class="anchored" data-anchor-id="if-and-else-statements">If and Else Statements</h2>
<ul>
<li>Conditional statements are how you inject some logic into your code.</li>
<li>The most commonly used conditional statement is <code>if.</code>
<ul>
<li>Whenever you see an <code>if</code> statement, read it as ‘If X is TRUE, then do a thing’.</li>
</ul></li>
<li>Another statement is <code>else</code>, which extends the logic to ‘If X is TRUE, do a thing, or else do something different’.</li>
</ul>
</section>
<section id="a-programming-joke-for-conditional-statements" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="a-programming-joke-for-conditional-statements">A programming joke for conditional statements</h2>
<p>A programmer’s partner says: ‘Please go to the store and buy a carton of milk and if they have eggs, get six.’</p>
<p>The programmer returned with 6 cartons of milk.</p>
<p>When the partner sees this, and exclaims ‘Why the heck did you buy six cartons of milk?’</p>
<p>The programmer replied ‘They had eggs’</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>eggs <span class="ot">&lt;-</span> <span class="cn">TRUE</span> <span class="co"># Whether there were eggs in the store</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (eggs <span class="sc">==</span> <span class="cn">TRUE</span>) { <span class="co"># If there are eggs</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  n.milk <span class="ot">&lt;-</span> <span class="dv">6</span> <span class="co"># Get 6 cartons of milk</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> { <span class="co"># If there are not eggs</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  n.milk <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="co"># Get 1 carton of milk</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  }</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
<section id="r-interlude---simulate-a-population-and-sample-it" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="r-interlude---simulate-a-population-and-sample-it">R INTERLUDE - Simulate a population and sample it!</h2>
</section>
<section id="simulations-to-compare-parameter-estimates" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="simulations-to-compare-parameter-estimates">Simulations to compare parameter estimates</h2>
<ul>
<li>Let’s use our distribution functions from last time to set up some data to play with</li>
<li>Let’s imagine our data is made up of counts, with an average of 3 counts - which distribution would fit that data best?</li>
<li>Ex: number of hours per day spent doing homework by UO undergraduates</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">32</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>true_pop <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">10000</span>, <span class="at">lambda =</span> <span class="dv">3</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(true_pop, <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">16</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-33-1.png" class="img-fluid figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="calculating-parameters" class="level2">
<h2 class="anchored" data-anchor-id="calculating-parameters">Calculating parameters</h2>
<ul>
<li>How would we calculate the mean and range for this population?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">mean</span>(true_pop)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.9817</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">range</span>(true_pop)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  0 12</code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(true_pop)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3</code></pre>
</div>
</div>
<ul>
<li>how about the variance and the standard deviation?</li>
</ul>
</section>
<section id="sampling-exercise" class="level2">
<h2 class="anchored" data-anchor-id="sampling-exercise">Sampling Exercise</h2>
<ul>
<li>Since we are working with simulated data, we can also afford to simulate our sampling!</li>
<li>Try taking a sample from our <code>true_pop</code> dataset and change the sample size, then calculate the mean and range for your sample and see how it compares to the true values.</li>
<li>How many college students are you including in your survey?</li>
</ul>
</section>
<section id="randomness-in-sampling" class="level2">
<h2 class="anchored" data-anchor-id="randomness-in-sampling">Randomness in Sampling</h2>
<ul>
<li>Because of the randomness of sampling, you may get close to the true estimates even with a small sample size</li>
<li>But your results will change each time you take a new sample of the same size</li>
<li>How do we get a feel for how accurate each sample size is? Or which sample size is recommended?</li>
</ul>
</section>
<section id="surveying-your-sampling" class="level2">
<h2 class="anchored" data-anchor-id="surveying-your-sampling">Surveying your Sampling</h2>
<p>Step 1 - take 50 samples of size 10 Step 2 - Calculate the mean from each sample Step 3 - Plot the distribution of sample mean estimates</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">32</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>true_pop <span class="ot">&lt;-</span> <span class="fu">rpois</span>(<span class="at">n=</span><span class="dv">10000</span>, <span class="at">lambda =</span> <span class="dv">3</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>samps_var <span class="ot">&lt;-</span> <span class="fu">replicate</span>(<span class="dv">50</span>, <span class="fu">sample</span>(true_pop, <span class="dv">10</span>))</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>samps_var_means <span class="ot">&lt;-</span> <span class="fu">apply</span>(samps_var, <span class="dv">2</span>, mean)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="histograms" class="level2">
<h2 class="anchored" data-anchor-id="histograms">Histograms</h2>
<div class="columns">
<div class="column" style="width:50%;">
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(true_pop)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div><div class="column" style="width:50%;">
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(samps_var_means)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week4a_files/figure-html/unnamed-chunk-37-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</div>
</div>
</section>
<section id="surveying-your-sampling-1" class="level2">
<h2 class="anchored" data-anchor-id="surveying-your-sampling-1">Surveying your Sampling</h2>
<ul>
<li>This sampling variation is what we have to deal with, and account for, as empirical scientists.</li>
<li>If this had been a real-world scenario, we likely would be basing our estimate for the true mean on just a single sample mean.</li>
<li>Getting close to the idea of <strong>power</strong> - does our experimental design have the power to detect the parameters we are interested in?</li>
</ul>
</section>
<section id="challenge---bootstrapping-to-produce-a-confidence-interval" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="challenge---bootstrapping-to-produce-a-confidence-interval">CHALLENGE - Bootstrapping to produce a confidence interval</h2>
<ul>
<li>The 2.5th and 97.5th percentiles of the bootstrap sampling distribution are a passable 95% confidence interval</li>
<li>Note that no transformations or normality assumptions needed</li>
<li>You can use the <code>quantile()</code> function to calculate these</li>
<li>You can also use <code>qnorm</code></li>
<li>Use R to figure out the bootstrap distribution for other parameters (such as variance).</li>
<li>Try to produce bootstrap distributions for the mean and variance of gene expresssion of one gene from the stickleback data set</li>
<li>Plot the resulting distributions</li>
<li>Determine the value of the 2.5th and 97.5th percentiles</li>
</ul>
<hr>
</section>
</section>
<section id="section-2" class="level1 smaller">
<h1 class="smaller"></h1>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>