<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.57">

  <meta name="author" content="Bill Cresko">
  <title>BioEngineering Statistics Spring 2025 – Week 6a - Statistics for Bioengineering</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Week 6a - Statistics for Bioengineering</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Bill Cresko 
</div>
</div>
</div>

</section>
<section id="this-week" class="slide level2">
<h2>This week</h2>
<ul>
<li><p>Model 1 and 2 regression</p></li>
<li><p>Multiple linear regression</p></li>
<li><p>Analysis of Variance (ANOVA)</p></li>
<li><p>Key principles of experimental design</p></li>
<li><p>How to report statistics in papers</p></li>
</ul>
</section>
<section>
<section id="linear-models-and-regression" class="title-slide slide level1 center">
<h1>Linear Models and Regression</h1>

</section>
<section id="general-linear-model---two-or-more-continuous-variables" class="slide level2">
<h2>General linear model - two or more continuous variables</h2>
<p><span class="math display">\[ y_i = \beta_0 + \beta_1 x_1 + \epsilon_i \]</span></p>

<img data-src="images/images_4b.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"><p><span class="math display">\[ y_i = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + ... + \beta_n x_n + \epsilon_i \]</span></p>
<p><span class="math display">\[ y_i = \beta_0 + \beta_1 x_1 + \beta_2 x_2 + \beta_3 x_1x_2 + \epsilon_i \]</span></p>
</section>
<section id="generalized-linear-models---non-linear-in-the-coefficients" class="slide level2 smaller">
<h2>Generalized linear models - non-linear in the coefficients</h2>
<ul>
<li><p><strong>Polynomial regression</strong> <span class="math display">\[y = \beta_0 + \beta_1 x + \beta_2 x^2 +...+ \beta_n x^n +\epsilon\]</span></p></li>
<li><p><strong>Logistic regression</strong> <span class="math display">\[log(\frac{\pi_i}{1-\pi_1})=\beta_0 + \beta_1 x_1 + \beta_2 x_2 +...+ \beta_n x_n +\epsilon\]</span></p></li>
<li><p><strong>Poisson regression</strong> <span class="math display">\[log(y) = \beta_0 + \beta_1 x + \epsilon\]</span></p></li>
<li><p><strong>Exponential regression</strong> <span class="math display">\[y = \alpha^{\beta x} + \epsilon\]</span></p></li>
<li><p><strong>Power regression</strong> <span class="math display">\[y = \alpha x^\beta + \epsilon\]</span></p></li>
</ul>
</section></section>
<section>
<section id="model-1-and-model-2-regression" class="title-slide slide level1 center">
<h1>Model 1 and Model 2 regression</h1>

</section>
<section id="hypothesis-tests-in-regression" class="slide level2 smaller">
<h2>Hypothesis tests in regression</h2>
<p><span class="math display">\[H_0 : \beta_0 = 0\]</span> <span class="math display">\[H_A : \beta_1 \neq 0\]</span></p>
<p>reduced model (<span class="math inline">\(H_0\)</span>) -</p>
<p><span class="math display">\[y_i = \beta_0 + 0x_i + \epsilon_i\]</span></p>
<p>full model (<span class="math inline">\(H_A\)</span>) -</p>
<p><span class="math display">\[y_i = \beta_0 + \beta_1x_i + \epsilon_i\]</span><br>
<br>
</p>
<ol type="1">
<li>fits a “reduced” model without slope term (H0)</li>
<li>fits the “full” model with slope term added back</li>
<li>compares fit of full and reduced models using an <strong><em>F test statistic</em></strong></li>
<li>compares the residual variance of the full vs.&nbsp;reduced models</li>
</ol>
</section>
<section id="section" class="slide level2">
<h2></h2>

<img data-src="images/images_5a.001.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="section-1" class="slide level2">
<h2></h2>

<img data-src="images/images_5a.002.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="section-2" class="slide level2">
<h2></h2>

<img data-src="images/images_5a.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="r-interlude-model-i-and-ii-regression" class="slide level2 smaller">
<h2>R INTERLUDE | Model I and II regression</h2>
<ul>
<li>First we fit a standard model I ordinary least squares</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a>zfish_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"Drerio_development_complete.csv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">","</span>)</span>
<span id="cb1-2"><a></a></span>
<span id="cb1-3"><a></a>length <span class="ot">&lt;-</span> zfish_data<span class="sc">$</span>Length_cm</span>
<span id="cb1-4"><a></a>weight <span class="ot">&lt;-</span> zfish_data<span class="sc">$</span>Weight_mg</span>
<span id="cb1-5"><a></a></span>
<span id="cb1-6"><a></a>size_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(weight<span class="sc">~</span>length)</span>
<span id="cb1-7"><a></a><span class="fu">summary</span>(size_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = weight ~ length)

Residuals:
    Min      1Q  Median      3Q     Max 
-4.1164 -1.1004  0.0976  0.9950  4.5447 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  2.42136    0.12820   18.89   &lt;2e-16 ***
length       3.06303    0.03775   81.13   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.48 on 598 degrees of freedom
Multiple R-squared:  0.9167,    Adjusted R-squared:  0.9166 
F-statistic:  6582 on 1 and 598 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</section>
<section id="r-interlude-model-i-and-ii-regression-1" class="slide level2 smaller">
<h2>R INTERLUDE | Model I and II regression</h2>
<ul>
<li>We can fit the line</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-2"><a></a></span>
<span id="cb3-3"><a></a><span class="fu">ggplot</span>(size_lm, <span class="fu">aes</span>(<span class="at">x =</span> weight, <span class="at">y =</span> length)) <span class="sc">+</span></span>
<span id="cb3-4"><a></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span>                  </span>
<span id="cb3-5"><a></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb3-6"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb3-7"><a></a>    <span class="at">title =</span> <span class="st">"Linear regression of zebrafish weight on standard length"</span>,</span>
<span id="cb3-8"><a></a>    <span class="at">x =</span> <span class="st">"zebrafish length in cm"</span>,</span>
<span id="cb3-9"><a></a>    <span class="at">y =</span> <span class="st">"zebrafish weight in gm"</span></span>
<span id="cb3-10"><a></a>  ) <span class="sc">+</span></span>
<span id="cb3-11"><a></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week6a_files/figure-revealjs/unnamed-chunk-7-1.png" width="960" class="r-stretch"></section>
<section id="r-interlude-model-i-and-ii-regression-2" class="slide level2 smaller">
<h2>R INTERLUDE | Model I and II regression</h2>
<ul>
<li>We can fit a <code>LOESS</code> line</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-2"><a></a></span>
<span id="cb4-3"><a></a><span class="fu">ggplot</span>(size_lm, <span class="fu">aes</span>(<span class="at">x =</span> weight, <span class="at">y =</span> length)) <span class="sc">+</span></span>
<span id="cb4-4"><a></a>  <span class="fu">geom_point</span>(<span class="at">color =</span> <span class="st">"blue"</span>, <span class="at">size =</span> <span class="dv">2</span>, <span class="at">alpha =</span> <span class="fl">0.2</span>) <span class="sc">+</span>                  </span>
<span id="cb4-5"><a></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"loess"</span>, <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">se =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb4-6"><a></a>  <span class="fu">labs</span>(</span>
<span id="cb4-7"><a></a>    <span class="at">title =</span> <span class="st">"Linear regression of zebrafish weight on standard length"</span>,</span>
<span id="cb4-8"><a></a>    <span class="at">x =</span> <span class="st">"zebrafish length in cm"</span>,</span>
<span id="cb4-9"><a></a>    <span class="at">y =</span> <span class="st">"zebrafish weight in gm"</span></span>
<span id="cb4-10"><a></a>  ) <span class="sc">+</span></span>
<span id="cb4-11"><a></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week6a_files/figure-revealjs/unnamed-chunk-8-1.png" width="960" class="r-stretch"></section>
<section id="smoothers-and-local-regression" class="slide level2 smaller">
<h2>Smoothers and local regression</h2>
<ul>
<li><strong>loess</strong> (locally estimated scatterplot smoothing) - fit least squares regression lines to successive subsets of the observations weighted according to their distance from the focal observation.<br>
</li>
<li><strong>kernel smoothers</strong> - new smoothed y-values are computed as the weighted averages of points within a defined window of the original x-values. Weightings are determined by the type of kernel smoother specified, the larger the window, the greater the degree of smoothing.<br>
</li>
<li><strong>splines</strong> - joins together a series of polynomial fits that have been generated after the data is split up into a number of smaller windows.</li>
</ul>
</section>
<section id="r-interlude-model-i-and-ii-regression-3" class="slide level2 smaller">
<h2>R INTERLUDE | Model I and II regression</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a></a><span class="fu">library</span>(lmodel2)</span>
<span id="cb5-2"><a></a></span>
<span id="cb5-3"><a></a>size_lm_ModII <span class="ot">&lt;-</span> <span class="fu">lmodel2</span>(weight <span class="sc">~</span> length)</span>
<span id="cb5-4"><a></a>size_lm_ModII</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Model II regression

Call: lmodel2(formula = weight ~ length)

n = 600   r = 0.9574514   r-square = 0.9167132 
Parametric P-values:   2-tailed = 0    1-tailed = 0 
Angle between the two OLS regression lines = 1.418965 degrees

Regression results
  Method Intercept    Slope Angle (degrees) P-perm (1-tailed)
1    OLS  2.421359 3.063034        71.91950                NA
2     MA  1.662671 3.316382        73.22019                NA
3    SMA  2.013729 3.199153        72.64166                NA

Confidence intervals
  Method 2.5%-Intercept 97.5%-Intercept 2.5%-Slope 97.5%-Slope
1    OLS       2.169579        2.673139   2.988886    3.137182
2     MA       1.416796        1.897871   3.237842    3.398487
3    SMA       1.789108        2.233204   3.125864    3.274161

Eigenvalues: 28.63954 0.1960763 

H statistic used for computing C.I. of MA: 4.476936e-05 </code></pre>
</div>
</div>
<ul>
<li>You should also have generated <code>OLS (Ordinary Least Squares)</code>, <code>MA (Moving Average)</code>, and <code>SMA (Standard Major Axis)</code> regression models, and the last plots SMA line from parameter estimates</li>
</ul>
</section>
<section id="r-interlude---on-your-own" class="slide level2 smaller">
<h2>R INTERLUDE - on your own</h2>
<ul>
<li><p>Say you measured green fluorescent protein abundance in cell culture across several key, carefully controlled temperatures.</p></li>
<li><p>You ultimately want to know whether protein expression changes as a function of temperature in your experiment.</p></li>
<li><p>Read in the <code>gfp_temp.tsv</code> data and perform a regression analysis.</p></li>
<li><p>Address the following questions</p>
<ul>
<li>Which is X and which is Y?</li>
<li>Are residual assumptions met?</li>
<li>What model of regression should we use in this case and why?</li>
<li>What is our decision regarding our null hypothesis of no relationship?</li>
</ul></li>
</ul>
</section></section>
<section>
<section id="mutliple-linear-regression" class="title-slide slide level1 center">
<h1>Mutliple Linear regression</h1>

</section>
<section id="multiple-linear-regression---goals" class="slide level2">
<h2>Multiple Linear Regression - Goals</h2>
<ul>
<li><p>To develop a better predictive model than is possible from models based on single independent variables.</p></li>
<li><p>To investigate the relative individual effects of each of the multiple independent variables above and beyond the effects of the other variables.</p></li>
<li><p>The individual effects of each of the predictor variables on the response variable can be depicted by single partial regression lines.</p></li>
<li><p>The slope of any single partial regression line (partial regression slope) thereby represents the rate of change or effect of that specific predictor variable (holding all the other predictor variables constant to their respective mean values) on the response variable.</p></li>
</ul>
</section>
<section id="multiple-linear-regression-additive-and-multiplicative-models-of-2-or-more-predictors" class="slide level2">
<h2>Multiple Linear Regression | Additive and multiplicative models of 2 or more predictors</h2>
<p>Additive model <span class="math display">\[y_i = \beta_0 + \beta_1x_{1} + \beta_2x_{2} + ... + \beta_jx_{j} + \epsilon_i\]</span></p>
<p><br>
</p>
<p>Multiplicative model (with two predictors) <span class="math display">\[y_i = \beta_0 + \beta_1x_1 + \beta_2x_2 + \beta_3x_1x_2 + \epsilon_i\]</span></p>
</section>
<section id="multiple-linear-regression-additive-and-multiplicative-models" class="slide level2">
<h2>Multiple Linear Regression | Additive and multiplicative models</h2>
<p><br></p>

<img data-src="images/images_5a.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="multiple-linear-regression-assumptions" class="slide level2">
<h2>Multiple linear regression assumptions</h2>
<p><br></p>
<ul>
<li>linearity</li>
<li>normality</li>
<li>homogeneity of variance</li>
<li>The absence of strong <strong>multi-collinearity</strong>
<ul>
<li>a predictor variable must not be linearly predicted by - or correlated with - any combination of the other predictor variables.</li>
</ul></li>
</ul>
</section>
<section id="checking-for-multi-collinearity" class="slide level2">
<h2>checking for multi-collinearity</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a><span class="fu">library</span>(car)</span>
<span id="cb7-2"><a></a>zfish_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"Drerio_development_complete.csv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">","</span>)</span>
<span id="cb7-3"><a></a></span>
<span id="cb7-4"><a></a>length <span class="ot">&lt;-</span> zfish_data<span class="sc">$</span>Length_cm</span>
<span id="cb7-5"><a></a>weight <span class="ot">&lt;-</span> zfish_data<span class="sc">$</span>Weight_mg</span>
<span id="cb7-6"><a></a>age <span class="ot">&lt;-</span> zfish_data<span class="sc">$</span>Age_Days</span>
<span id="cb7-7"><a></a>pigment <span class="ot">&lt;-</span> zfish_data<span class="sc">$</span>Pigmentation</span>
<span id="cb7-8"><a></a></span>
<span id="cb7-9"><a></a></span>
<span id="cb7-10"><a></a><span class="fu">scatterplotMatrix</span>(<span class="sc">~</span>length<span class="sc">+</span>pigment<span class="sc">+</span>age, <span class="at">diag=</span><span class="st">"boxplot"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week6a_files/figure-revealjs/unnamed-chunk-11-1.png" width="960" class="r-stretch"></section>
<section id="r-interlude-multiple-regression-with-2-predictor-variables" class="slide level2 smaller">
<h2>R INTERLUDE | multiple regression with 2 predictor variables</h2>
<ul>
<li>Read in the <code>RNAseq_lipid.tsv</code> and examine the continuous variables.</li>
<li>We are interested in whether Gene01 and/or Gene02 expression levels influence lipid content.</li>
<li>First plot <span class="math inline">\(Y\)</span> vs <span class="math inline">\(Gene01\)</span>, <span class="math inline">\(Y\)</span> vs <span class="math inline">\(Gene02\)</span>, then set up and test a multiplicative model.</li>
<li>What are the parameter estimates of interest in this case?</li>
<li>What are the outcomes from our hypothesis tests?</li>
</ul>
</section>
<section id="r-interlude-multiple-regression-with-2-predictor-variables-1" class="slide level2 smaller">
<h2>R INTERLUDE | multiple regression with 2 predictor variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a></a>RNAseq_Data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"RNAseq_lipid.tsv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb8-2"><a></a></span>
<span id="cb8-3"><a></a>y <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Lipid_Conc</span>
<span id="cb8-4"><a></a>g1 <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Gene01</span>
<span id="cb8-5"><a></a>g2 <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Gene02</span>
<span id="cb8-6"><a></a>g3 <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Gene03</span>
<span id="cb8-7"><a></a>g4 <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Gene04</span>
<span id="cb8-8"><a></a></span>
<span id="cb8-9"><a></a>Mult_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> g1<span class="sc">*</span>g2)</span>
<span id="cb8-10"><a></a><span class="fu">summary</span>(Mult_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ g1 * g2)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.52111 -0.16333 -0.00931  0.21515  0.54255 

Coefficients:
              Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  3.2965521  0.3461603   9.523 5.40e-08 ***
g1           0.0558232  0.0040117  13.915 2.34e-10 ***
g2          -0.0082765  0.0290978  -0.284    0.780    
g1:g2        0.0001702  0.0003199   0.532    0.602    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2997 on 16 degrees of freedom
Multiple R-squared:  0.9846,    Adjusted R-squared:  0.9817 
F-statistic: 341.3 on 3 and 16 DF,  p-value: 1.042e-14</code></pre>
</div>
</div>
</section>
<section id="r-interlude-multiple-regression-with-2-predictor-variables-2" class="slide level2 smaller">
<h2>R INTERLUDE | multiple regression with 2 predictor variables</h2>
<ul>
<li>Now get rid of the interaction term, and set up a purely additive model</li>
<li>Did any of our estimates change? Why?</li>
<li>Did the degrees of freedom change? Why?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a></a>Add_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> g1<span class="sc">+</span>g2)</span>
<span id="cb10-2"><a></a><span class="fu">summary</span>(Add_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Also try different combinations of genes</li>
<li>Finally, see if you can make a partial regression plot of just one predictor and the response variable</li>
</ul>
</section>
<section id="r-interlude-adding-a-polynomial" class="slide level2 smaller">
<h2>R INTERLUDE | adding a polynomial</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a></a>Poly_lm_1 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> <span class="fu">poly</span>(g1, <span class="dv">1</span>))</span>
<span id="cb11-2"><a></a><span class="fu">summary</span>(Poly_lm_1)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ poly(g1, 1))

Residuals:
     Min       1Q   Median       3Q      Max 
-0.50579 -0.18804 -0.03178  0.20893  0.54981 

Coefficients:
            Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  7.47450    0.06402  116.75   &lt;2e-16 ***
poly(g1, 1)  9.58694    0.28631   33.48   &lt;2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2863 on 18 degrees of freedom
Multiple R-squared:  0.9842,    Adjusted R-squared:  0.9833 
F-statistic:  1121 on 1 and 18 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</section>
<section id="r-interlude-adding-a-polynomial-1" class="slide level2 smaller">
<h2>R INTERLUDE | adding a polynomial</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a></a>Poly_lm_2 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> <span class="fu">poly</span>(g1, <span class="dv">2</span>))</span>
<span id="cb13-2"><a></a><span class="fu">summary</span>(Poly_lm_2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ poly(g1, 2))

Residuals:
     Min       1Q   Median       3Q      Max 
-0.48903 -0.19681 -0.01246  0.18486  0.54854 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   7.47450    0.06575 113.681   &lt;2e-16 ***
poly(g1, 2)1  9.58694    0.29404  32.604   &lt;2e-16 ***
poly(g1, 2)2  0.07566    0.29404   0.257      0.8    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.294 on 17 degrees of freedom
Multiple R-squared:  0.9843,    Adjusted R-squared:  0.9824 
F-statistic: 531.5 on 2 and 17 DF,  p-value: 4.725e-16</code></pre>
</div>
</div>
</section>
<section id="r-interlude-adding-a-polynomial-2" class="slide level2 smaller">
<h2>R INTERLUDE | adding a polynomial</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a></a>Poly_lm_3 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> <span class="fu">poly</span>(g1, <span class="dv">3</span>))</span>
<span id="cb15-2"><a></a><span class="fu">summary</span>(Poly_lm_3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ poly(g1, 3))

Residuals:
    Min      1Q  Median      3Q     Max 
-0.5201 -0.1758 -0.0155  0.1628  0.5170 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)   7.47450    0.06720 111.233  &lt; 2e-16 ***
poly(g1, 3)1  9.58694    0.30051  31.902 6.51e-16 ***
poly(g1, 3)2  0.07566    0.30051   0.252    0.804    
poly(g1, 3)3  0.15773    0.30051   0.525    0.607    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.3005 on 16 degrees of freedom
Multiple R-squared:  0.9845,    Adjusted R-squared:  0.9816 
F-statistic: 339.4 on 3 and 16 DF,  p-value: 1.089e-14</code></pre>
</div>
</div>
</section>
<section id="r-interlude-adding-a-polynomial-3" class="slide level2 smaller">
<h2>R INTERLUDE | adding a polynomial</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a></a>Poly_lm_10 <span class="ot">&lt;-</span> <span class="fu">lm</span>(y <span class="sc">~</span> <span class="fu">poly</span>(g1, <span class="dv">10</span>))</span>
<span id="cb17-2"><a></a><span class="fu">summary</span>(Poly_lm_10)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = y ~ poly(g1, 10))

Residuals:
     Min       1Q   Median       3Q      Max 
-0.33720 -0.10156  0.00083  0.11606  0.30332 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     7.474500   0.053607 139.431 2.55e-16 ***
poly(g1, 10)1   9.586936   0.239738  39.989 1.90e-11 ***
poly(g1, 10)2   0.075656   0.239738   0.316   0.7595    
poly(g1, 10)3   0.157731   0.239738   0.658   0.5271    
poly(g1, 10)4  -0.065839   0.239738  -0.275   0.7898    
poly(g1, 10)5   0.652130   0.239738   2.720   0.0236 *  
poly(g1, 10)6   0.167389   0.239738   0.698   0.5027    
poly(g1, 10)7  -0.349566   0.239738  -1.458   0.1788    
poly(g1, 10)8  -0.007755   0.239738  -0.032   0.9749    
poly(g1, 10)9   0.036730   0.239738   0.153   0.8816    
poly(g1, 10)10 -0.588592   0.239738  -2.455   0.0364 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.2397 on 9 degrees of freedom
Multiple R-squared:  0.9945,    Adjusted R-squared:  0.9883 
F-statistic: 161.6 on 10 and 9 DF,  p-value: 6.533e-09</code></pre>
</div>
</div>
</section></section>
<section>
<section id="model-selection-when-you-have-many-predictor-variables" class="title-slide slide level1 center">
<h1>Model selection when you have many predictor variables</h1>

</section>
<section id="model-selection" class="slide level2">
<h2>Model selection</h2>

<img data-src="images/images_5b.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="model-selection-the-problems" class="slide level2">
<h2>Model selection | the problems</h2>
<ul>
<li><p>How to decide the complexity of polynomial: straight line regression, quadratic, cubic, ….</p></li>
<li><p>Which variables to keep/ discard when building a multiple regression model?</p></li>
<li><p>Selecting from candidate models representing different biological processes.</p></li>
</ul>
</section>
<section id="model-selection-a-beetle-example" class="slide level2">
<h2>Model selection | a beetle example</h2>

<img data-src="images/images_5b.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="start-with-linear-regression" class="slide level2">
<h2>Start with linear regression</h2>

<img data-src="images/images_5b.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="quadratic-2nd-degree-polynomial" class="slide level2">
<h2>Quadratic (2nd degree) polynomial?</h2>

<img data-src="images/images_5b.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="quadratic-3rd-degree-polynomial" class="slide level2">
<h2>Quadratic (3rd degree) polynomial?</h2>

<img data-src="images/images_5b.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="a-polynomial-of-degree-5" class="slide level2">
<h2>A polynomial of degree 5?</h2>

<img data-src="images/images_5b.009.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="a-polynomial-of-degree-10" class="slide level2">
<h2>A polynomial of degree 10?</h2>

<img data-src="images/images_5b.010.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="the-problem-with-this-approach" class="slide level2">
<h2>The problem with this approach</h2>
<ul>
<li>The log likelihood of the model increases with the number of parameters</li>
<li>So does <span class="math inline">\(r^2\)</span></li>
<li>Isn’t this good - the best fit to the data?</li>
</ul>

<img data-src="images/images_5b.011.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="the-problem-with-this-approach-1" class="slide level2">
<h2>The problem with this approach</h2>
<ul>
<li>Does it violate a principle of parsimony</li>
<li>Fit no more parameters than is necessary.</li>
<li>If two or more models fit the data almost equally well, prefer the simpler model.</li>
</ul>
<blockquote>
<p>“models should be pared down until they are minimal and adequate”</p>
</blockquote>
<p>Crawley 2007, p325</p>
</section>
<section id="lets-consider-our-objectives" class="slide level2">
<h2>Let’s consider our objectives</h2>
<ul>
<li><p>Model should predicts well</p></li>
<li><p>Approximates true relationship between the variables</p></li>
<li><p>Be able to evaluate a wider array of models. Not only or more “reduced” models.</p></li>
<li><p>But an overfitted model may not predict well in the future!</p></li>
</ul>
</section>
<section id="a-polynomial-of-degree-10-1" class="slide level2">
<h2>A polynomial of degree 10?</h2>

<img data-src="images/images_5b.010.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="how-do-we-accomplish-these-goals" class="slide level2 flexbox vcenter">
<h2>How do we accomplish these goals?</h2>
<p>How to accomplish these goals To answer this, we need</p>
<ul>
<li>A criterion to compare models:
<ul>
<li>Mallow’s Cp</li>
<li>AIC (Akaike’s Information Criterion)</li>
<li>BIC (Bayesian Information Criterion)</li>
</ul></li>
<li>A strategy for searching the candidate models</li>
</ul>
</section>
<section id="how-do-we-accomplish-these-goals-1" class="slide level2">
<h2>How do we accomplish these goals?</h2>
<ul>
<li><p>How to decide the complexity of polynomial: straight line regression, quadratic, cubic, ….</p></li>
<li><p>Which variables to keep/ discard when building a multiple regression model?</p></li>
<li><p>Selecting from candidate models representing different biological processes.</p></li>
</ul>
</section>
<section id="section-3" class="slide level2 smaller">
<h2></h2>
<ul>
<li><p><strong>MSresiduals</strong>- represents the mean amount of variation unexplained by the model, and therefore the lowest value indicates the best fit.</p></li>
<li><p><strong>Adjusted</strong> <span class="math inline">\(r^2\)</span> - (the proportion of mean amount of variation in response variable explained by the model) is calculated as adj. r2 which is adjusted for both sample size and the number of terms. Larger values indicate better fit.</p></li>
<li><p><strong>Mallow’s Cp</strong> - is an index resulting from the comparison of the specific model to a model that contain all the possible terms. Models with the lowest value and/or values closest to their respective p (the number of model terms, including the y-intercept) indicate best fit.</p></li>
<li><p><strong>Akaike Information Criteria (AIC)</strong> - there are several different versions of AIC, each of which adds a different constant designed to penalize according to the number of parameters and sample size to a likelihood function to produce a relative measure of the information content of a model. Smaller values indicate more parsimonious models. As a rule of thumb, if the difference between two AIC values (delta AIC) is greater than 2, the lower AIC is a significant improvement in parsimony.</p></li>
<li><p><strong>Schwarz or Bayesian Information Criteria (BIC or SIC)</strong> - is outwardly similar to AIC. The constant added to the likelihood function penalizes models with more predictor terms more heavily (and thus select more simple models) than AIC. It is for this reason that BIC is favored by many researchers.</p></li>
</ul>
</section>
<section id="mallows-cp" class="slide level2 flexbox vcenter">
<h2>Mallow’s Cp</h2>
<ul>
<li>Frequently used in multiple regression</li>
<li>Test case: “all subsets regression”</li>
<li>Strategy: Test all possible models and selection the one with the smallest Cp</li>
<li><code>leaps</code> package in R. Smart search among a potentially huge number of models</li>
<li>Typically we are modeling observational data. Experimental data often allows smart use of variables</li>
<li>Investigating all possible subsets of variables = only intelligent decision is choice of variables ….</li>
</ul>
</section>
<section id="mallows-cp-1" class="slide level2">
<h2>Mallow’s Cp</h2>

<img data-src="images/images_5b.012.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="mallows-cp-2" class="slide level2 flexbox vcenter">
<h2>Mallow’s Cp</h2>
<p><strong>For prediction</strong>: All models with Cp &lt; p predict about equally well. Don’t get carried away with a “best”.</p>
<p><strong>For explanation</strong>: If numerous equally well fitting models fit the data, it is difficult to deduce which predictor “explains” the response.</p>
<p><strong>General caveat</strong>: “regression is not causation”. Experiment needed to get to causal explanation.</p>
</section>
<section id="r-interlude---model-fitting-of-multiple-linear-regression-example" class="slide level2 smaller">
<h2>R INTERLUDE - model fitting of multiple linear regression example</h2>
<ul>
<li>xxx</li>
<li>xxx</li>
<li>xxx</li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section id="analysis-of-variance" class="title-slide slide level1 center">
<h1>Analysis of Variance</h1>

</section>

<section>
<section id="anova" class="title-slide slide level1 center">
<h1>ANOVA</h1>

</section>
<section id="anova-1" class="slide level2 flexbox vcenter">
<h2>ANOVA</h2>
<ul>
<li>Stands for <strong>AN</strong>alysis <strong>o</strong>f <strong>VA</strong>riance</li>
<li>Core statistical procedure in biology</li>
<li>Developed by R.A. Fisher in the early 20th Century</li>
<li>The core idea is to ask how much variation exists within vs.&nbsp;among groups</li>
<li>ANOVAs are linear models that have <strong>categorical predictor</strong> and <strong>continuous response</strong> variables</li>
<li>The categorical predictors are often called <strong>factors</strong>, and can have two or more <strong>levels</strong> (important to specify in R)</li>
<li>Each factor will have a <strong>hypothesis test</strong></li>
<li>The levels of each factor may also need to be tested</li>
</ul>
</section></section>
<section>
<section id="an-example---irises" class="title-slide slide level1 center">
<h1>An Example - Irises</h1>
<ul>
<li>Let’s use an example to think about how ANOVA works under the hood</li>
<li>Our system: iris flowers have sepals and petals of differing lengths</li>
<li>The question: do different species have different sepal lengths?</li>
</ul>

<img data-src="images/week9_irises.png" class="quarto-figure quarto-figure-center r-stretch" style="width:40.0%"></section>
<section id="an-example---irises-1" class="slide level2">
<h2>An Example - Irises</h2>
<ul>
<li>We have data from the R package <code>iris</code></li>
<li>3 Different species: <em>Iris versicolor</em>, <em>Iris setosa</em>, and <em>Iris virginica</em></li>
</ul>
</section>
<section id="an-example---irises-2" class="slide level2">
<h2>An Example - Irises</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a></a><span class="fu">stripchart</span>(iris<span class="sc">$</span>Sepal.Length <span class="sc">~</span> iris<span class="sc">$</span>Species, <span class="at">vertical=</span>T, <span class="at">method=</span><span class="st">"jitter"</span>,</span>
<span id="cb19-2"><a></a>           <span class="at">ylab=</span><span class="st">"sepal length"</span>, <span class="at">xlab=</span><span class="st">"species"</span>, <span class="at">pch=</span><span class="dv">19</span>, <span class="at">cex=</span><span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week6a_files/figure-revealjs/unnamed-chunk-29-1.png" style="width:80.0%" class="r-stretch"></section>
<section id="anovas-and-hypotheses" class="slide level2 smaller">
<h2>ANOVAs and Hypotheses</h2>
<ul>
<li><p>As with any statistical test, we will be testing the probability of accepting or rejecting our null hypothesis, compared to an alternative hypothesis</p></li>
<li><p>What are the null and alternative hypotheses for our example?</p></li>
</ul>
<p><span class="math display">\[H_0: \mu(\text{versicolor_length"}) = \mu(\text{setosa_length}) = \mu(\text{virginica_length})\]</span></p>
<p><span class="math display">\[H_A: \mu(\text{versicolor_length"}) \neq \mu(\text{setosa_length}) \neq \mu(\text{virginica_length})\]</span></p>
<ul>
<li>NOTE: the alternative hypothesis is that at least one group is different</li>
</ul>
</section>
<section id="anova-2" class="slide level2 vcenter flexbox">
<h2>ANOVA</h2>

<img data-src="images/images_5b.015.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="anova-conceptually-similar-to-regression" class="slide level2 vcenter flexbox">
<h2>ANOVA | Conceptually similar to regression</h2>

<img data-src="images/images_5b.016.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="anova-statistical-results-table" class="slide level2 vcenter flexbox">
<h2>ANOVA | Statistical results table</h2>

<img data-src="images/images_5b.017.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="anova-f-ratio-calculation" class="slide level2 vcenter flexbox">
<h2>ANOVA | F-ratio calculation</h2>

<img data-src="images/images_5b.018.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="anova-f-ratio-calculation-1" class="slide level2 vcenter flexbox">
<h2>ANOVA | F-ratio calculation</h2>

<img data-src="images/images_5b.019.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="assumptions-of-anova" class="slide level2">
<h2>Assumptions of ANOVA</h2>
<ol type="1">
<li><p>The response variable (y) is approximately normal in all groups (factor levels)</p>
<ul>
<li>Deviation from normality O.K. if sample sizes and variances across groups are equal</li>
</ul></li>
<li><p>Variances equal across groups</p></li>
<li><p>Observations within a group are independent</p>
<ul>
<li><p>randomly sampled</p></li>
<li><p>no structuring</p></li>
</ul></li>
</ol>
</section>
<section id="lets-practice-with-irises" class="slide level2 smaller">
<h2>Let’s practice with irises</h2>
<ul>
<li>First steps, what are the null and alternative hypotheses for our test? (Already done!)</li>
<li>Next, do our data meet or violate the assumptions of ANOVA tests?</li>
<li>Then, we can set up our ANOVA</li>
<li>What tests should you run (functions, graphing, etc.) in RStudio to evaluate if the data is O.K. to use in an ANOVA test?
<ul>
<li>Plot the data for each group as a histogram - is it normally distributed?</li>
<li>Run variance tests between the groups - do they have equal variance?</li>
<li>Is our data randomly sampled? Assume yes.</li>
</ul></li>
</ul>
</section>
<section id="lets-practice-with-irises-1" class="slide level2">
<h2>Let’s practice with irises</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a></a>set <span class="ot">&lt;-</span> <span class="fu">subset</span>(iris, Species <span class="sc">==</span><span class="st">"setosa"</span>)</span>
<span id="cb20-2"><a></a>vers <span class="ot">&lt;-</span> <span class="fu">subset</span>(iris, Species <span class="sc">==</span><span class="st">"versicolor"</span>)</span>
<span id="cb20-3"><a></a>vir <span class="ot">&lt;-</span> <span class="fu">subset</span>(iris, Species<span class="sc">==</span><span class="st">"virginica"</span>)</span>
<span id="cb20-4"><a></a><span class="fu">hist</span>(set<span class="sc">$</span>Sepal.Length, <span class="at">main =</span> <span class="st">"I. setosa"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="Week6a_files/figure-revealjs/unnamed-chunk-35-1.png" style="width:33.0%"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a></a><span class="fu">hist</span>(vers<span class="sc">$</span>Sepal.Length, <span class="at">main =</span> <span class="st">"I. versicolor"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="Week6a_files/figure-revealjs/unnamed-chunk-35-2.png" style="width:33.0%"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a></a><span class="fu">hist</span>(vir<span class="sc">$</span>Sepal.Length, <span class="at">main=</span><span class="st">"I. virginica"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="Week6a_files/figure-revealjs/unnamed-chunk-35-3.png" style="width:33.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="lets-practice-with-irises-2" class="slide level2">
<h2>Let’s practice with irises</h2>
<ul>
<li>How do we interpret these findings?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a></a>iris_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(Sepal.Length <span class="sc">~</span> Species, iris)</span>
<span id="cb23-2"><a></a><span class="fu">anova</span>(iris_aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Response: Sepal.Length
           Df Sum Sq Mean Sq F value    Pr(&gt;F)    
Species     2 63.212  31.606  119.26 &lt; 2.2e-16 ***
Residuals 147 38.956   0.265                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
<section id="anova-results" class="slide level2">
<h2>ANOVA results</h2>
<ul>
<li>Species is significantly associated with differences in sepal length!</li>
<li>The Mean Square (MS) of our factor (species) explains more of the variance in the response variable (sepal length) than the residuals (random or unmeasured variables)</li>
</ul>
</section>
<section id="nonparametric-tests-similar-to-anova" class="slide level2">
<h2>Nonparametric tests similar to ANOVA</h2>
<ul>
<li>The Kruskal-Wallis test is robust to non-normality and group variance differences</li>
<li>Check out the help page for <code>kruskal.test()</code> and run a model on iris sepal length</li>
</ul>
</section>
<section id="nonparametric-tests-similar-to-anova-1" class="slide level2">
<h2>Nonparametric tests similar to ANOVA</h2>
<ul>
<li>Also a significant difference between species</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
    Kruskal-Wallis rank sum test

data:  Sepal.Length by Species
Kruskal-Wallis chi-squared = 96.937, df = 2, p-value &lt; 2.2e-16</code></pre>
</div>
</div>
</section></section>
<section>
<section id="anova-and-linear-models" class="title-slide slide level1 center">
<h1>ANOVA and linear models</h1>
<ul>
<li>Last time I mentioned that ANOVAs used the same principles as linear models that we discussed last time</li>
<li>We can use the function <code>lm()</code> to also run an analysis of variance</li>
<li>Try it out! Set up a linear model to find the relationship between species and sepal length</li>
<li>Can you identify where in the results you can find information about ANOVA?</li>
</ul>
</section>
<section id="anova-and-linear-models-1" class="slide level2">
<h2>Anova and linear models</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = iris$Sepal.Length ~ iris$Species)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.6880 -0.3285 -0.0060  0.3120  1.3120 

Coefficients:
                       Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)              5.0060     0.0728  68.762  &lt; 2e-16 ***
iris$Speciesversicolor   0.9300     0.1030   9.033 8.77e-16 ***
iris$Speciesvirginica    1.5820     0.1030  15.366  &lt; 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 0.5148 on 147 degrees of freedom
Multiple R-squared:  0.6187,    Adjusted R-squared:  0.6135 
F-statistic: 119.3 on 2 and 147 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
</section></section>
<section id="testing-what-groups-differ-from-one-another" class="title-slide slide level1 center">
<h1>Testing what groups differ from one another</h1>

</section>

<section>
<section id="post-hoc-comparisons" class="title-slide slide level1 center">
<h1>Post Hoc Comparisons</h1>

<img data-src="images/week9_turkey.jpeg" class="r-stretch"></section>
<section id="post-hoc-comparisons-1" class="slide level2">
<h2>Post Hoc Comparisons</h2>
<ul>
<li>So now you’ve got a significant result that says your groups are different from one another!</li>
<li>But <strong>how</strong> are they different? 2 groups could be the same and 1 is different, or all 3 could be different from each other!</li>
</ul>
</section>
<section id="post-hoc-comparisons-2" class="slide level2">
<h2>Post Hoc comparisons</h2>
<ul>
<li>There are several ways to compare groups after an ANOVA</li>
<li>One of the most common ways is by Tukey’s tests (Tukey’s range test, Tukey’s honestly significant difference (HSD) test)</li>
<li>The function <code>TukeyHSD()</code> in R</li>
</ul>

<img data-src="images/week9_Tukeymeme.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="lets-practice---irises" class="slide level2">
<h2>Let’s practice - Irises</h2>
<ul>
<li>The function <code>TukeyHSD()</code> in R</li>
<li>Run post-hoc analyses on your anova model using TukeyHSD function
<ul>
<li>Check out the help page to figure out how to run</li>
</ul></li>
<li>How do you interpret the results?</li>
</ul>
</section>
<section id="lets-practice---irises-1" class="slide level2">
<h2>Let’s practice - Irises</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a></a>iris_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(Sepal.Length <span class="sc">~</span> Species, iris)</span>
<span id="cb27-2"><a></a><span class="fu">TukeyHSD</span>(iris_aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Tukey multiple comparisons of means
    95% family-wise confidence level

Fit: aov(formula = Sepal.Length ~ Species, data = iris)

$Species
                      diff       lwr       upr p adj
versicolor-setosa    0.930 0.6862273 1.1737727     0
virginica-setosa     1.582 1.3382273 1.8257727     0
virginica-versicolor 0.652 0.4082273 0.8957727     0</code></pre>
</div>
</div>
</section>
<section id="lets-practice---using-the-categorical-predictor-variables-in-the-zebrafish-data-set" class="slide level2">
<h2>Let’s practice - using the categorical predictor variables in the zebrafish data set</h2>
<ul>
<li>Read in the data as before</li>
<li>Pick one or two categorical predictor variables to include</li>
<li>Make sure that R sees them as type <code>factor</code></li>
<li>Set up the model and evaluate the overall fit</li>
<li>Do a post-hoc means test with Tukey’s HSD</li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="general-linear-mixed-models" class="title-slide slide level1 center">
<h1>General Linear Mixed Models</h1>

</section>
<section id="what-is-a-glm" class="slide level2">
<h2>What is a GLM?</h2>
<ul>
<li><p>General Linear Model (<strong>GLM</strong>) - two or more continuous variables</p></li>
<li><p>Generalized Linear Model - used to conduct linear regressions on non-continuous data, and non-normal data</p></li>
<li><p>General Linear Mixed Model (<strong>GLMM</strong>) - a continuous response variable with a mix of continuous and categorical predictor variables, can be used to model random effects (for more on this, see Peter Ralph’s Advanced Bio Stats class)</p></li>
</ul>
</section>
<section id="how-do-we-factor-in-categorical-variables" class="slide level2">
<h2>How do we factor in categorical variables?</h2>
<ul>
<li>For a single categorical predictor, we can include effects of each factor level:</li>
</ul>
<p><span class="math display">\[ y = B_0 + B_1(x~level1) + B_2(x~level2) +~... ~+ e \]</span></p>
<ul>
<li>Each factor level (ex: for Wolbachia: Yes or No) becomes it’s own “effector” that will change the value of y</li>
<li>Instead of x being a continuous numerical value, for categorical data it will be either a 0 or 1</li>
</ul>
</section>
<section id="a-generalized-linear-model" class="slide level2">
<h2>A Generalized Linear Model</h2>
<ul>
<li>Check out the help page in RStudio for <code>glm()</code></li>
</ul>
</section>
<section id="first-steps-use-a-glm-to-test-wolbachia-infection" class="slide level2">
<h2>First steps: use a glm to test Wolbachia infection</h2>
<ul>
<li>Set up a <code>glm()</code> to test for the effect of Wolbachia infection on Recombinant Fraction</li>
</ul>
</section>
<section id="first-steps-use-a-glm-to-test-wolbachia-infection-1" class="slide level2">
<h2>First steps: use a glm to test Wolbachia infection</h2>
<ul>
<li>What do these components mean?</li>
</ul>
</section>
<section id="interpreting-the-glm" class="slide level2">
<h2>Interpreting the GLM</h2>
<ul>
<li>Why is our intercept significant?
<ul>
<li>Remember that the null hypothesis is a model with an intercept and slope of 0</li>
</ul></li>
<li>How would you interpret the “WolbachiaYes” coefficient, especially the estimate and the p value?</li>
</ul>
</section>
<section id="second-steps---adding-more-variables" class="slide level2">
<h2>Second steps - adding more variables</h2>
<ul>
<li>Now let’s add Food to our model - how does this change our model results?</li>
</ul>
</section>
<section id="second-steps---adding-more-variables-1" class="slide level2">
<h2>Second steps - adding more variables</h2>
</section>
<section id="interpreting-the-glm-1" class="slide level2">
<h2>Interpreting the GLM</h2>
<ul>
<li>How would you interpret the Intercept and Food coefficients, especially the estimate and the p value?</li>
</ul>
</section>
<section id="interaction-effects-in-glms" class="slide level2">
<h2>Interaction effects in GLMs</h2>
<ul>
<li>You can also model interactions between two categorical variables in glms
<ul>
<li>What if variable x and variable z interact in non-additive ways?</li>
</ul></li>
<li>Use the notation x*z in the formula</li>
</ul>
</section>
<section id="third-steps---adding-interaction-effects" class="slide level2">
<h2>Third steps - adding interaction effects</h2>
<ul>
<li>Let’s finally add an interaction between Wolbachia infection and Food to represent Wolbachia titer in our model</li>
<li>Run the model - what are the results?</li>
</ul>
</section>
<section id="third-steps---adding-interaction-effects-1" class="slide level2">
<h2>Third steps - adding interaction effects</h2>
<ul>
<li>How do we interpret these findings?</li>
</ul>
</section>
<section id="how-do-we-know-which-model-to-use" class="slide level2">
<h2>How do we know which model to use?</h2>
<ul>
<li>Want to be careful not to <em>overfit</em> your linear model (give too many variables)</li>
<li>The Akaike information criterion (AIC) is a metric for comparing different models to find the best fit
<ul>
<li>Notice the AIC score in your output?</li>
<li>For more on this, take Advanced Bio Stats! Or do some reading on the internet.</li>
</ul></li>
</ul>
</section>
<section id="multiple-linear-regression---goals-1" class="slide level2">
<h2>Multiple Linear Regression - Goals</h2>
<ul>
<li>To develop a better predictive model than is possible from models based on single independent variables.</li>
<li>To investigate the relative individual effects of each of the multiple independent variables above and beyond the effects of the other variables.</li>
<li>The individual effects of each of the predictor variables on the response variable can be depicted by single partial regression lines.</li>
<li>The slope of any single partial regression line (partial regression slope) thereby represents the rate of change or effect of that specific predictor variable (holding all the other predictor variables constant to their respective mean values) on the response variable.</li>
</ul>
</section>
<section id="r-interlude-one-way-anova" class="slide level2 smaller">
<h2>R INTERLUDE | One way ANOVA</h2>
<ul>
<li>Again, use the RNAseq_lip.tsv data again.</li>
<li>Let’s test for an effect of Population on Gene01 expression levels</li>
<li>First, let’s look at how the data are distributed</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a></a>RNAseq_Data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'RNAseq_lip.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb29-2"><a></a>g1 <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Gene01</span>
<span id="cb29-3"><a></a>Pop <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>Population</span>
<span id="cb29-4"><a></a><span class="fu">boxplot</span>(g1<span class="sc">~</span>Pop, <span class="at">col=</span><span class="fu">c</span>(<span class="st">"blue"</span>,<span class="st">"green"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Or, to plot all points:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a></a><span class="fu">stripchart</span>(g1<span class="sc">~</span>Pop, <span class="at">vertical=</span>T, <span class="at">pch=</span><span class="dv">19</span>, <span class="at">col=</span><span class="fu">c</span>(<span class="st">"blue"</span>,<span class="st">"green"</span>), </span>
<span id="cb30-2"><a></a>           <span class="at">at=</span><span class="fu">c</span>(<span class="fl">1.25</span>,<span class="fl">1.75</span>), <span class="at">method=</span><span class="st">"jitter"</span>, <span class="at">jitter=</span><span class="fl">0.05</span>)</span>
<span id="cb30-3"><a></a>Pop_Anova <span class="ot">&lt;-</span> <span class="fu">aov</span>(g1 <span class="sc">~</span> Pop)</span>
<span id="cb30-4"><a></a><span class="fu">summary</span>(Pop_Anova)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-one-way-anova-1" class="slide level2 smaller vcenter flexbox">
<h2>R INTERLUDE | One way ANOVA</h2>

<img data-src="images/images_6a.001.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="anova-one-or-more-predictor-variables" class="slide level2">
<h2>ANOVA | One or more predictor variables</h2>
<ul>
<li>One-way ANOVAs just have a single factor</li>
<li>Multi-factor ANOVAs
<ul>
<li>Factorial - two or more factors and their interactions</li>
<li>Nested - the levels of one factor are contained within another level</li>
<li>The models can be quite complex</li>
</ul></li>
<li>ANOVAs use an F-statistic to test factors in a model
<ul>
<li>Ratio of two variances (numerator and denominator)</li>
<li>The numerator and denominator d.f. need to be included (e.g.&nbsp;<span class="math inline">\(F_{1, 34} = 29.43\)</span>)</li>
</ul></li>
<li>Determining the appropriate test ratios for complex ANOVAs takes some work</li>
</ul>
</section>
<section id="anova-assumptions" class="slide level2">
<h2>ANOVA | Assumptions</h2>
<ul>
<li>Normally distributed groups
<ul>
<li>robust to non-normality if equal variances and sample sizes</li>
</ul></li>
<li>Equal variances across groups
<ul>
<li>okay if largest-to-smallest variance ratio &lt; 3:1</li>
<li>problematic if there is a mean-variance relationship among groups</li>
</ul></li>
<li>Observations in a group are independent
<ul>
<li>randomly selected</li>
<li>don’t confound group with another factor</li>
</ul></li>
</ul>
</section></section>
<section>
<section id="different-ways-to-include-factors-in-models" class="title-slide slide level1 center">
<h1>Different ways to include factors in models</h1>

</section>
<section id="anova-fixed-effects-of-factors" class="slide level2">
<h2>ANOVA | Fixed effects of factors</h2>
<ul>
<li>Groups are predetermined, of direct interest, repeatable.</li>
<li>For example:
<ul>
<li>medical treatments in a clinical trial</li>
<li>predetermined doses of a toxin</li>
<li>age groups in a population</li>
<li>habitat, season, etc.</li>
</ul></li>
<li>Any conclusions reached in the study about differences among groups can be applied only to the groups included in the study.</li>
<li>The results cannot be generalized to other treatments, habitats, etc. not included in the study.</li>
</ul>
</section>
<section id="anova-random-effects-of-factors" class="slide level2">
<h2>ANOVA | Random effects of factors</h2>
<ul>
<li>Measurements that come in groups. A group can be:
<ul>
<li>a family made up of siblings</li>
<li>a subject measured repeatedly</li>
<li>a transect of quadrats in a sampling survey</li>
<li>a block of an experiment done at a given time</li>
</ul></li>
<li>Groups are assumed to be randomly sampled from a population of groups.</li>
<li>Therefore, conclusions reached about groups can be generalized to the population of groups.</li>
<li>With random effects, the variance among groups is the main quantity of interest, not the specific group attributes.</li>
</ul>
</section>
<section id="anova-random-effects-of-factors-1" class="slide level2 smaller">
<h2>ANOVA | Random effects of factors</h2>
<ul>
<li>Below are cases where you are likely to treat factors as random effects</li>
<li>Whenever your sampling design is nested
<ul>
<li>quadrats within transects</li>
<li>transects within woodlots</li>
<li>woodlots within districts</li>
</ul></li>
<li>Whenever you divide up plots and apply separate treatments to subplots</li>
<li>Whenever your replicates are grouped spatially or temporally
<ul>
<li>in blocks</li>
<li>in batches</li>
</ul></li>
<li>Whenever you take measurements on related individuals</li>
<li>Whenever you measure subjects or other sampling units repeatedly</li>
</ul>
</section>
<section id="anova-random-effects-of-factors-2" class="slide level2 smaller vcenter flexbox">
<h2>ANOVA | Random effects of factors</h2>

<img data-src="images/images_6a.002.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="anova-random-effects---test-your-understanding" class="slide level2">
<h2>ANOVA | Random effects - test your understanding</h2>
<ul>
<li>Factor is sex (Male vs.&nbsp;Female)</li>
<li>Factor is fish tank (10 tanks in an experiment)</li>
<li>Factor is family (measure multiple sibs per family)</li>
<li>Factor is temperature (10 arbitrary temps over natural range)</li>
</ul>
</section>
<section id="anova-caution-about-fixed-vs.-random-effects" class="slide level2">
<h2>ANOVA | Caution about fixed vs.&nbsp;random effects</h2>
<ul>
<li>Using fixed vs.&nbsp;random effects changes the way that statistical tests are performed in ANOVA</li>
<li>Most statistical packages assume that all factors are fixed unless you instruct it otherwise</li>
<li>Designating factors as random takes extra work and probably a read of the manual</li>
<li>In <code>R</code>, <code>lm</code> assumes that all effects are fixed</li>
<li>For random effects, use <code>lme</code> instead (part of the nlme package)</li>
</ul>
</section></section>
<section>
<section id="means-test-to-compare-levels-of-a-factor" class="title-slide slide level1 center">
<h1>Means test to compare levels of a factor</h1>

</section>
<section id="means-for-greater-than-two-factor-levels" class="slide level2 smaller">
<h2>Means for greater than two factor levels?</h2>
<ul>
<li>The F-ratio test for a single-factor ANOVA tests for any difference among groups.</li>
<li>If we want to understand specific differences, we need further “contrasts”.</li>
<li>Unplanned comparisons (post hoc):
<ul>
<li>Multiple comparisons carried out after the results are obtained.</li>
<li>Used to find where the differences lie (which means differ from which other means)</li>
<li>Comparisons require protection for inflated Type 1 error rates:
<ul>
<li>Tukey tests: compare all pairs of means and control for multiple comparisons</li>
<li>Scheffé contrasts: compare all combinations of means</li>
</ul></li>
</ul></li>
<li>Planned comparisons (a priori):
<ul>
<li>Comparisons between group means that were decided when the experiment was designed (not after the data were in)</li>
<li>Must be few in number to avoid inflating Type 1 error rates</li>
</ul></li>
</ul>
</section>
<section id="planned-a-priori-contrasts" class="slide level2">
<h2>Planned (a priori) contrasts</h2>
<ul>
<li>A well planned experiment often dictates which comparison of means are of most interest, whereas other comparisons are of no interest.</li>
<li>By restricting the comparisons to just the ones of interest, researchers can mitigate the multiple testing problem associated with post-hoc tests.</li>
<li>Some statisticians argue that, in fact, planned comparisons allow researchers to avoid adjusting p-values all together because each test is therefore unique.</li>
<li>Contrasts can also allow more complicated tests of the relationships among means.</li>
<li>Coding a priori contrasts in R is quite easy and just depends upon writing the right series of coefficient contrasts.</li>
</ul>
</section>
<section id="planned-a-priori-contrasts-1" class="slide level2 flexbox vcenter">
<h2>Planned (a priori) contrasts</h2>

<img data-src="images/images_6a.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="understand-the-coefficients-table" class="slide level2 flexbox vcenter">
<h2>Understand the coefficients table</h2>

<img data-src="images/images_6a.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="r-interlude-planned-contrasts" class="slide level2 smaller">
<h2>R INTERLUDE | Planned contrasts</h2>
<ul>
<li>Take the RNAseq data you’ve examined before and create a new four level genotype by combining genotype and microbiota treatment into a single variable</li>
<li>Think about how to do this using <code>dplyr</code> functions.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a></a>RNAseq_Data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"RNAseq.tsv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">''</span>)</span>
<span id="cb31-2"><a></a></span>
<span id="cb31-3"><a></a>x <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>categorical_var</span>
<span id="cb31-4"><a></a>y <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>continuous_var1</span>
<span id="cb31-5"><a></a>z <span class="ot">&lt;-</span> RNAseq_Data<span class="sc">$</span>continuous_var2</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Set up the a priori contrasts specifically testing one group mean against another</li>
<li>These are just examples - you should figure out the logic of the contrasts</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a></a><span class="fu">contrasts</span>(x) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="dv">0</span>, <span class="sc">-</span><span class="dv">1</span>), <span class="fu">c</span>(<span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="sc">-</span><span class="dv">1</span>), <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Confirm that the contrasts are orthogonal</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a></a><span class="fu">round</span>(<span class="fu">crossprod</span>(<span class="fu">contrasts</span>(x)), <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-planned-contrasts-1" class="slide level2 smaller">
<h2>R INTERLUDE | Planned contrasts</h2>
<ul>
<li>Define the contrast labels</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a></a>rnaseq_data_list <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">x =</span> <span class="fu">list</span>(‘xxx vs. xxx’ <span class="ot">=</span> <span class="dv">1</span>, ‘xxx vs. xxx’ <span class="ot">=</span> <span class="dv">2</span>, ‘xxx vs. xxx’ <span class="ot">=</span> <span class="dv">3</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Then fit the fixed effect model</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a></a>RNAseq_aov_fixed <span class="ot">&lt;-</span> <span class="fu">aov</span>(y <span class="sc">~</span> x)</span>
<span id="cb35-2"><a></a><span class="fu">plot</span>(RNAseq_aov_fixed)</span>
<span id="cb35-3"><a></a><span class="fu">boxplot</span>(y <span class="sc">~</span> x)</span>
<span id="cb35-4"><a></a><span class="fu">summary</span>(RNAseq_aov_fixed, <span class="at">split =</span> rnaseq_data_list)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-unplanned-contrasts" class="slide level2 smaller">
<h2>R INTERLUDE | Unplanned contrasts</h2>
<ul>
<li>Remember that this is when you had no hypotheses of differences in means in advance</li>
<li>Read in the perchlorate data from Week 3</li>
<li>Let’s assess the effects of the 4 perchlorate levels on T4</li>
<li>Which perchlorate levels differ in their effect on T4?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a></a>perc <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'perchlorate_data.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb36-2"><a></a></span>
<span id="cb36-3"><a></a>x <span class="ot">&lt;-</span> perc<span class="sc">$</span>Perchlorate_Level</span>
<span id="cb36-4"><a></a>y <span class="ot">&lt;-</span> <span class="fu">log10</span>(perc<span class="sc">$</span>T4_Hormone_Level)</span>
<span id="cb36-5"><a></a></span>
<span id="cb36-6"><a></a>MyANOVA <span class="ot">&lt;-</span> <span class="fu">aov</span>(y <span class="sc">~</span> x)</span>
<span id="cb36-7"><a></a><span class="fu">summary</span> (MyANOVA)</span>
<span id="cb36-8"><a></a><span class="fu">boxplot</span>(y <span class="sc">~</span> x)</span>
<span id="cb36-9"><a></a></span>
<span id="cb36-10"><a></a><span class="fu">install.packages</span>(<span class="st">"multcomp"</span>)</span>
<span id="cb36-11"><a></a><span class="fu">library</span>(multcomp)</span>
<span id="cb36-12"><a></a></span>
<span id="cb36-13"><a></a><span class="fu">summary</span>(<span class="fu">glht</span>(MyANOVA, <span class="at">linfct =</span> <span class="fu">mcp</span>(<span class="at">x =</span> <span class="st">"Tukey"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section>
<section>
<section id="multifactor-anova" class="title-slide slide level1 center">
<h1>Multifactor ANOVA</h1>

</section>
<section id="multifactor-anova-1" class="slide level2 smaller">
<h2>Multifactor ANOVA</h2>
<ul>
<li><strong>Nested ANOVA</strong> or nested design
<ul>
<li>factors might be hierarchical - in other words nested - within one another</li>
<li>The sources of variance are therefore hierarchical too</li>
</ul></li>
<li>The <strong>factorial ANOVA</strong> design is the most common experimental design used to investigate more than one treatment variable
<ul>
<li>In a factorial design every combination of treatments from two (or more) treatment variables is investigated.</li>
<li>The main purpose of a factorial design is to evaluate possible interactions between variables.</li>
<li>An <strong>interaction</strong> between two explanatory variables means that the effect of one variable on the response depends on the state of a second variable.</li>
</ul></li>
</ul>
</section>
<section id="multifactor-anova-key-difference-between-nested-and-factorial-designs" class="slide level2 smaller">
<h2>Multifactor ANOVA | Key difference between nested and factorial designs</h2>
<ul>
<li>Nested designs are hierarchical
<ul>
<li>often contain sub-replicates that are random, uncontrolled, nuisance effects</li>
<li>but the nested factors can be of interest too</li>
</ul></li>
<li>Factorial designs are
<ul>
<li>all pairwise combinations,</li>
<li>and often involve all combinations of factor levels</li>
<li>when each factor is fixed interactions can be assessed</li>
</ul></li>
<li>Completely nested designs therefore have no interaction terms, whereas factorial designs do</li>
<li><strong>Mixed models</strong> can have a combination of fixed and random factors that are more complicated</li>
</ul>
</section></section>
<section>
<section id="nested-anova" class="title-slide slide level1 center">
<h1>Nested ANOVA</h1>

</section>
<section id="nested-anova-walking-stick-example" class="slide level2 smaller">
<h2>Nested ANOVA | Walking stick example</h2>
<ul>
<li>Example 1: Study of “repeatability” (simple nested design)</li>
<li>The walking stick, <em>Timema cristinae</em>, is a wingless herbivorous insect on plants in chaparral habitats of California.</li>
<li>Nosil and Crespi (2006) measured individuals using digital photographs.</li>
<li>To evaluate measurement repeatability they took two separate photographs of each specimen.</li>
<li>After measuring traits on one set of photographs, they repeated the measurements on the second set.</li>
</ul>

<img data-src="images/images_7a.001.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="nested-anova-walking-stick-example-1" class="slide level2 flexbox vcenter">
<h2>Nested ANOVA | Walking stick example</h2>
<p>Each pair of dots represents the two measurements</p>

<img data-src="images/images_7a.002.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="nested-anova-walking-stick-example-2" class="slide level2 flexbox vcenter">
<h2>Nested ANOVA | Walking stick example</h2>

<img data-src="images/images_7a.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="nested-anova-anova-table-of-results" class="slide level2 flexbox vcenter">
<h2>Nested ANOVA | ANOVA Table of Results</h2>

<img data-src="images/images_7a.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="nesting-logic" class="slide level2 flexbox vcenter">
<h2>Nesting Logic</h2>

<img data-src="images/images_7a.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="nesting-equations" class="slide level2 flexbox vcenter">
<h2>Nesting equations</h2>

<img data-src="images/images_7a.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="nesting-hypothesis-tests" class="slide level2 flexbox vcenter">
<h2>Nesting hypothesis tests</h2>

<img data-src="images/images_7a.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="nesting-ms-calculations" class="slide level2 flexbox vcenter">
<h2>Nesting MS calculations</h2>

<img data-src="images/images_7a.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="nested-anova-table-of-results" class="slide level2 flexbox vcenter">
<h2>Nested ANOVA table of results</h2>

<img data-src="images/images_7a.009.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="r-interlude-nested-anova" class="slide level2">
<h2>R INTERLUDE | Nested ANOVA</h2>

<img data-src="images/images_7a.010.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="r-interlude-nested-anova-1" class="slide level2 smaller">
<h2>R INTERLUDE | Nested ANOVA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a></a>andrew_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'andrew.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span>‘\t<span class="st">')</span></span>
<span id="cb37-2"><a></a><span class="st">head(andrew_data)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>There are four variables: ‘TREAT’, ‘PATCH’, ‘QUAD’ and ‘ALGAE’</li>
<li>The main effect factor is TREAT</li>
<li>Make a simplified factor called TREAT2, in which 0% and 33% are a level called “low” and 66% and 100% are “high”</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a></a>andrew_data<span class="sc">$</span>TREAT2 <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(“low”,<span class="dv">40</span>),<span class="fu">rep</span>(“high”,<span class="dv">40</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>The nested factor is PATCH - also need to turn this into a factor</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a></a>andrew_data<span class="sc">$</span>PATCH <span class="ot">&lt;-</span> <span class="fu">factor</span>(andrew_data<span class="sc">$</span>PATCH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-nested-anova-2" class="slide level2 smaller">
<h2>R INTERLUDE | Nested ANOVA</h2>
<ul>
<li>In this case, our response variable is ALGAE</li>
<li>Look at the distribution of ALGAE for the two levels of TREAT2 using boxplots based on the patch means, which are the replicates in this case.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a></a>andrew.agg <span class="ot">&lt;-</span> <span class="fu">with</span>(andrew_data, <span class="fu">aggregate</span>(<span class="fu">data.frame</span>(ALGAE), </span>
<span id="cb40-2"><a></a>                  <span class="at">by =</span> <span class="fu">list</span>(<span class="at">TREAT2=</span>TREAT2, <span class="at">PATCH=</span>PATCH), mean)</span>
<span id="cb40-3"><a></a></span>
<span id="cb40-4"><a></a><span class="fu">library</span>(nlme)</span>
<span id="cb40-5"><a></a>andrew.agg <span class="ot">&lt;-</span> <span class="fu">gsummary</span>(andrew_data, <span class="at">groups=</span>andrew_data<span class="sc">$</span>PATCH)</span>
<span id="cb40-6"><a></a></span>
<span id="cb40-7"><a></a><span class="fu">boxplot</span>(ALGAE <span class="sc">~</span> TREAT2, andrew.agg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Evaluate assumptions based on the boxplots</li>
<li>Is the design balanced (equal numbers of sub-replicates per PATCH)?</li>
</ul>
</section>
<section id="r-interlude-nested-anova-3" class="slide level2 smaller">
<h2>R INTERLUDE | Nested ANOVA</h2>
<ul>
<li>Run the nested ANOVA:</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a></a>nested.aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(ALGAE <span class="sc">~</span> TREAT2 <span class="sc">+</span> <span class="fu">Error</span>(PATCH), <span class="at">data=</span>andrew_data)</span>
<span id="cb41-2"><a></a><span class="fu">summary</span>(nested.aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Do we detect an effect of TREAT2 (high vs low sea urchin density)?</li>
<li>Estimate variance components to assess relative contributions of the random factors</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a></a><span class="fu">library</span>(nlme)</span>
<span id="cb42-2"><a></a><span class="fu">VarCorr</span>(<span class="fu">lme</span>(ALGAE <span class="sc">~</span> <span class="dv">1</span>, <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> TREAT2<span class="sc">/</span>PATCH, andrew_data))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Calculate the % of variation due to between-treatment differences vs.&nbsp;due to among patches within treatment differences.</li>
<li>See pg. 302 in Logan if you need help.</li>
<li>What do these variance component estimates tell us???</li>
</ul>
</section></section>
<section>
<section id="factorial-designs" class="title-slide slide level1 center">
<h1>Factorial Designs</h1>

</section>
<section id="multifactor-anova-2" class="slide level2 smaller">
<h2>Multifactor ANOVA</h2>
<ul>
<li>For example, Relyae (2003) looked at how a moderate dose (1.6mg/L) of a commonly used pesticide, carbaryl (Sevin), affected bullfrog tadpole survival.</li>
<li>In particular, the experiment asked how the effect of carbaryl depended on whether a native predator, the red-spotted newt, was also present.</li>
<li>The newt was caged and could cause no direct harm, but it emitted visual and chemical cues to other tadpoles</li>
<li>The experiment was carried out in 10-L tubs (experimental units), each containing 10 tadpoles.</li>
<li>The four combinations of pesticide treatment (carbaryl vs.&nbsp;water only) and predator treatment (present or absent) were randomly assigned to tubs.</li>
<li>The results showed that survival was high except when pesticide was applied together with the predator.</li>
<li>Thus, the two treatments, predation and pesticide, seem to have interacted.</li>
</ul>

<img data-src="images/images_7a.011.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:30.0%"></section>
<section id="multifactor-anova-3" class="slide level2">
<h2>Multifactor ANOVA</h2>

<img data-src="images/images_7a.012.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="two-factor-factorial-designs" class="slide level2 flexbox vcenter">
<h2>Two Factor Factorial Designs</h2>

<img data-src="images/images_7a.013.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="three-factor-factorial-designs" class="slide level2 flexbox vcenter">
<h2>Three Factor Factorial Designs</h2>

<img data-src="images/images_7a.014.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="factorial-designs-number-of-replicates" class="slide level2 flexbox vcenter">
<h2>Factorial Designs | Number of Replicates</h2>

<img data-src="images/images_7a.015.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="model-1-factorial-anova-both-main-effects-fixed" class="slide level2 flexbox vcenter">
<h2>Model 1 factorial ANOVA | both main effects fixed</h2>

<img data-src="images/images_7a.016.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="model-2-factorial-anova-both-main-effects-fixed" class="slide level2 flexbox vcenter">
<h2>Model 2 factorial ANOVA | both main effects fixed</h2>

<img data-src="images/images_7a.017.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="model-2-factorial-anova-both-main-effects-random" class="slide level2 flexbox vcenter">
<h2>Model 2 factorial ANOVA | both main effects random</h2>

<img data-src="images/images_7a.018.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="the-mean-squares-for-a-factorial-model" class="slide level2 flexbox vcenter">
<h2>The mean squares for a factorial model</h2>

<img data-src="images/images_7a.019.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="the-f-ratios-for-a-factorial-model" class="slide level2 flexbox vcenter">
<h2>The F-ratios for a factorial model</h2>

<img data-src="images/images_7a.020.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="interpretation-significant-main-and-interaction-effects" class="slide level2 flexbox vcenter">
<h2>Interpretation | significant main and interaction effects</h2>

<img data-src="images/images_7a.021.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="interaction-plots" class="slide level2 flexbox vcenter">
<h2>Interaction plots</h2>

<img data-src="images/images_7a.022.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:75.0%"></section>
<section id="r-interlude-2-by-2-fixed-effect-factorial-anova" class="slide level2">
<h2>R INTERLUDE | 2-by-2 fixed effect factorial ANOVA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a></a>rnadata <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'RNAseq.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">''</span>)</span>
<span id="cb43-2"><a></a><span class="fu">head</span>(rnadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>continuous response variable and two main effect categorical variables</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a></a>gene <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Gene80</span>
<span id="cb44-2"><a></a>microbiota <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Microbiota</span>
<span id="cb44-3"><a></a>genotype <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Genotype</span>
<span id="cb44-4"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota)</span>
<span id="cb44-5"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> genotype)</span>
<span id="cb44-6"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fit-the-factorial-linear-model-two-different-ways-to-do-the-same-thing" class="slide level2">
<h2>Fit the factorial linear model | two different ways to do the same thing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a></a>rna_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(gene <span class="sc">~</span> microbiota <span class="sc">+</span> genotype <span class="sc">+</span> microbiota<span class="sc">:</span>genotype)</span>
<span id="cb45-2"><a></a>rna_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Examine the fitted model diagnostics and the ANOVA results table</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a></a><span class="fu">plot</span>(rna_aov)</span>
<span id="cb46-2"><a></a><span class="fu">summary</span>(rna_aov)</span>
<span id="cb46-3"><a></a><span class="fu">anova</span>(rna_aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>What are the general results of our hypothesis tests?</li>
<li>If there is an interaction, can we understand it by looking at the boxplots?</li>
</ul>
</section>
<section id="r-interlude-2-by-3-fixed-effect-factorial-anova" class="slide level2 flexbox vcenter">
<h2>R INTERLUDE | 2-by-3 fixed effect factorial ANOVA</h2>
<ul>
<li>Try the following code to produce an interaction plot for the response variable cell count.</li>
<li>In this case there are 2 genotypes and 3 treatment levels.</li>
<li>Download the <code>IntPlot_data</code> file and <code>IntPlot_Example.R</code></li>
<li>Go through the R script, get a feel for what it’s doing, and try to produce and interpret the interaction plot.</li>
</ul>
</section></section>
<section>
<section id="means-tests-for-multifactorial-anovas" class="title-slide slide level1 center">
<h1>Means tests for multifactorial ANOVAs</h1>

</section>
<section id="means-tests-factor-level-combinations-in-multi-factor-anova" class="slide level2">
<h2>Means tests | factor level combinations in multi-factor ANOVA</h2>
<ul>
<li>The F-ratio test for a single-factor ANOVA tests for any difference among groups.</li>
<li>If we want to understand specific differences, we need further “contrasts”.</li>
<li>Unplanned comparisons (post hoc)</li>
<li>Planned comparisons (a priori)</li>
<li>Now we need to make ‘pseudo-factors’ that combine our levels of interest</li>
</ul>
</section>
<section id="planned-a-priori-contrasts-2" class="slide level2 flexbox vcenter">
<h2>Planned (a priori) contrasts</h2>

<img data-src="images/images_7b.002.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction" class="slide level2 smaller">
<h2>R INTERLUDE | 2x2 Fixed-Effects Factorial ANOVA contrasts &amp; interaction</h2>
<p>continuous response and two main effect variables</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a></a>rnadata <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'RNAseq.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">''</span>)</span>
<span id="cb47-2"><a></a>gene <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Gene80</span>
<span id="cb47-3"><a></a>microbiota <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Microbiota</span>
<span id="cb47-4"><a></a>genotype <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Genotype</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>make new “pseudo factor,” combining genotype and microbiota</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a></a>gxm <span class="ot">&lt;-</span> <span class="fu">interaction</span>(genotype,microbiota)</span>
<span id="cb48-2"><a></a><span class="fu">levels</span>(gxm)</span>
<span id="cb48-3"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> gxm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>specify the following 2 contrasts</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a></a><span class="fu">contrasts</span>(gxm) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="sc">-</span><span class="dv">1</span>), <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction-1" class="slide level2 smaller">
<h2>R INTERLUDE | 2x2 Fixed-Effects Factorial ANOVA contrasts &amp; interaction</h2>
<p>Fit the factorial linear model</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a></a>rna_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(gene <span class="sc">~</span> gxm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Examine the ANOVA table, using supplied contrasts. Figure out the appropriate titles to give them.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a></a><span class="fu">summary</span>(rna_aov, <span class="at">split =</span> <span class="fu">list</span>(<span class="at">gxm =</span> <span class="fu">list</span>(<span class="st">'xxx'</span><span class="ot">=</span><span class="dv">1</span>,<span class="st">'xxx'</span><span class="ot">=</span><span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What does the contrast summary tell you about the nature of the interaction?</p>
</section></section>
<section>
<section id="mixed-effect-models-with-unequal-sample-sizes" class="title-slide slide level1 flexbox vcenter center">
<h1>Mixed effect models with unequal sample sizes</h1>

</section>
<section id="attributes-of-mixed-effects-models" class="slide level2 flexbox vcenter">
<h2>Attributes of mixed effects models</h2>
<ul>
<li>Linear models that include both fixed and random effects.</li>
<li>The model is split into fixed and random parts:
<ul>
<li>Fixed effects influence mean of the response variable Y.</li>
<li>Random effects influence the variance of Y.</li>
</ul></li>
<li>There is a different error variance for each level of grouping.</li>
<li>Estimation and testing is based on restricted maximum likelihood, which can handle unequal sample size.</li>
<li>P-values for fixed effects are conservative when design unbalanced.</li>
<li>Implemented in the <code>nlme</code> &amp; <code>lme4</code> packages in <code>R</code>.</li>
</ul>
</section>
<section id="assumptions-of-mixed-effects-models" class="slide level2 flexbox vcenter">
<h2>Assumptions of mixed-effects models</h2>
<ul>
<li>Variation within groups follows a normal distribution with equal variance among groups.</li>
<li>Groups are randomly sampled from “population” of groups.</li>
<li>Group means follow a normal distribution.</li>
<li>Measurements within groups are independent.</li>
</ul>
</section>
<section id="hypotheses-for-model-3-anova-factorial-design-with-mixed-effects" class="slide level2 flexbox vcenter">
<h2>Hypotheses for Model 3 ANOVA Factorial Design With Mixed Effects</h2>

<img data-src="images/images_7b.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="general-r-syntax-for-two-factor-factorial-designs" class="slide level2 flexbox vcenter">
<h2>General R syntax for two factor factorial designs</h2>

<img data-src="images/images_7b.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="r-interlude-variance-components-with-2-random-factors-using-lme4" class="slide level2 smaller">
<h2>R INTERLUDE | Variance components with 2 random factors using LME4</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a></a>rnadata <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'RNAseq.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">''</span>)</span>
<span id="cb52-2"><a></a><span class="fu">head</span>(rnadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>variables excluding first 5 and last 5 observations</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a></a>gene <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Gene80[<span class="dv">6</span><span class="sc">:</span><span class="dv">75</span>] </span>
<span id="cb53-2"><a></a>microbiota <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Microbiota[<span class="dv">6</span><span class="sc">:</span><span class="dv">75</span>]</span>
<span id="cb53-3"><a></a>genotype <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Genotype[<span class="dv">6</span><span class="sc">:</span><span class="dv">75</span>]</span>
<span id="cb53-4"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota)</span>
<span id="cb53-5"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> genotype)</span>
<span id="cb53-6"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Estimate the variance components using Restricted Maximum Likelihood (REML)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a></a><span class="fu">library</span>(lme4)</span>
<span id="cb54-2"><a></a><span class="fu">lmer</span>(gene <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> microbiota) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> genotype) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> microbiota<span class="sc">:</span>genotype))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Based on the REML sd estimates, what are the relative contributions of the factors to total variance in gene expression?</p>
</section></section>
<section>
<section id="analysis-of-covariance-ancova" class="title-slide slide level1 center">
<h1>Analysis of Covariance (ANCOVA)</h1>

</section>
<section id="brain-body-size-neaderthals-as-compared-to-humans" class="slide level2 flexbox vcenter">
<h2>Brain &amp; body size | neaderthals as compared to humans</h2>

<img data-src="images/images_7b.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="brain-body-size-neaderthals-as-compared-to-humans-1" class="slide level2 flexbox vcenter">
<h2>Brain &amp; body size | neaderthals as compared to humans</h2>

<img data-src="images/images_7b.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="brain-body-size-neaderthals-as-compared-to-humans-2" class="slide level2 flexbox vcenter">
<h2>Brain &amp; body size | neaderthals as compared to humans</h2>

<img data-src="images/images_7b.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="ancova" class="slide level2 flexbox vcenter">
<h2>ANCOVA</h2>
<ul>
<li>Analysis of covariance - mixture of regression and ANOVA</li>
<li>Response is still a normally distributed continuous variable</li>
<li>One or more continuous predictor variables (covariates)</li>
<li>Sometimes the covariates are of biological interest</li>
<li>Most often we want to remove unexplained variance</li>
<li>In this way they are similar to a blocking variable in ANOVA</li>
<li>Operationally, ANCOVA is regular ANOVA in which the group and overall means are replaced by group and overall relationships</li>
</ul>
</section>
<section id="ancova-adjusting-for-the-covariate" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Adjusting for the covariate</h2>

<img data-src="images/images_7b.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="ancova-adjusting-for-the-covariate-1" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Adjusting for the covariate</h2>

<img data-src="images/images_7b.009.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="ancova-linear-model-with-two-covariates" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Linear model with two covariates</h2>

<img data-src="images/images_7b.010.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="ancova-factor-and-covariate-hypothesis-tests" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Factor and covariate hypothesis tests</h2>

<img data-src="images/images_7b.011.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="ancova-f-ratio-tests" class="slide level2 flexbox vcenter">
<h2>ANCOVA | F ratio tests</h2>

<img data-src="images/images_7b.012.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="ancova-assumptions" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Assumptions</h2>
<ul>
<li>The residuals are normally distributed</li>
<li>The residuals show homoscedasticity of variance</li>
<li>The residuals are independent of one another</li>
<li>The relationship between the response variable and each covariate is linear</li>
<li>Homogeneity of slopes among the groups</li>
<li>Similar covariate ranges among the groups</li>
</ul>
</section>
<section id="ancova-heterogeneous-slopes" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Heterogeneous slopes</h2>

<img data-src="images/images_7b.013.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="ancova-heterogeneous-slopes-1" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Heterogeneous slopes</h2>
<ul>
<li>Problem - adjusting to a mean is difficult or impossible if the slopes are different</li>
<li>In essence, the samples for the groups come from two different populations</li>
<li>A test for homogeneity of slopes can be performed</li>
<li>The assumption is tested by looking for a significant interaction term between the categorical response variables and the covariate(s)</li>
</ul>
</section>
<section id="ancova-non-overlapping-range-of-the-covariate" class="slide level2 flexbox vcenter">
<h2>ANCOVA | Non-overlapping range of the covariate</h2>

<img data-src="images/images_7b.014.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="r-interlude-ancova" class="slide level2">
<h2>R INTERLUDE | ANCOVA</h2>
<ul>
<li>Impacts of sexual activity on male fruitfly longevity</li>
<li>Data from Partridge and Faraquhar (1981)</li>
<li>Longevity of male measured in response to access to
<ul>
<li>no females</li>
<li>one virgin</li>
<li>eight virgins</li>
<li>one mated</li>
<li>eight mated</li>
</ul></li>
<li>The male fruit flies also varied in size</li>
<li>The males were assigned randomly to each of the treatment levels, and then measured thorax length as a covariate</li>
</ul>
</section>
<section id="r-interlude-ancova-1" class="slide level2 smaller">
<h2>R INTERLUDE | ANCOVA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a></a>longevity_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'longevity.csv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">','</span>)</span>
<span id="cb55-2"><a></a><span class="fu">head</span>(longevity_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Variables</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a></a>long <span class="ot">&lt;-</span> longevity_data<span class="sc">$</span>LONGEVITY</span>
<span id="cb56-2"><a></a>treat <span class="ot">&lt;-</span> longevity_data<span class="sc">$</span>TREATMENT</span>
<span id="cb56-3"><a></a>thorax <span class="ot">&lt;-</span> longevity_data<span class="sc">$</span>THORAX</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>check to see if the covariate should be included</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a></a><span class="fu">boxplot</span>(long <span class="sc">~</span> treat)</span>
<span id="cb57-2"><a></a><span class="fu">plot</span>(long <span class="sc">~</span> thorax)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-ancova-2" class="slide level2 smaller">
<h2>R INTERLUDE | ANCOVA</h2>
<ul>
<li>assess assumptions of normality and homogeneity of variance</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a></a><span class="fu">plot</span>(<span class="fu">aov</span>(long <span class="sc">~</span> thorax <span class="sc">+</span> treat ), <span class="at">which =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>†ry it again with a transformed response variable</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a></a><span class="fu">plot</span>(<span class="fu">aov</span>(<span class="fu">log10</span>(long) <span class="sc">~</span> thorax <span class="sc">+</span> treat ), <span class="at">which =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>visually assess linearity, homogenetiy of slopes and covariate range equality</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a></a><span class="fu">library</span>(lattice)</span>
<span id="cb60-2"><a></a><span class="fu">print</span>(<span class="fu">xyplot</span>(<span class="fu">log10</span>(long) <span class="sc">~</span> thorax <span class="sc">|</span> treat, <span class="at">type =</span> <span class="fu">c</span>(<span class="st">"r"</span>, <span class="st">"p"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-ancova-3" class="slide level2 smaller">
<h2>R INTERLUDE | ANCOVA</h2>
<ul>
<li>formally test homogenetiy of slopes by testing the interaction term</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a></a><span class="fu">anova</span>(<span class="fu">aov</span>(<span class="fu">log10</span>(long) <span class="sc">~</span> thorax<span class="sc">*</span>treat))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>formally test covariate range disparity by modeling the effect of the treatments on the covariate</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a></a><span class="fu">anova</span>(<span class="fu">aov</span>(thorax <span class="sc">~</span> treat))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>FINALLY, set up contrasts, fit the additive model and visualize the results (pg. 459 and 460 of your Logan book)</li>
<li>Summarize the trends in a nice plot (pg. 461 of your Logan book)</li>
</ul>
</section></section>
<section>
<section id="design-principles-for-planning-a-good-experiment" class="title-slide slide level1 center">
<h1>Design principles for planning a good experiment</h1>

</section>
<section id="what-is-an-experimental-study" class="slide level2">
<h2>What is an experimental study?</h2>
<ul>
<li>In an experimental study the researcher assigns treatments to units</li>
<li>In an observational study nature does the assigning of treatments to units</li>
<li>The crucial advantage of experiments derives from the random assignment of treatments to units</li>
<li>Random assignment, or randomization, minimizes the influence of confounding variables</li>
</ul>
</section>
<section id="mount-everest-example" class="slide level2 vcenter flexbox">
<h2>Mount Everest example</h2>
<p>Survival of climbers of Mount Everest is higher for individuals taking supplemental oxygen than those who don’t.</p>
<p><strong>Why?</strong></p>
</section>
<section id="mount-everest-example-1" class="slide level2">
<h2>Mount Everest example</h2>
<ul>
<li>One possibility is that supplemental oxygen (explanatory variable) really does cause higher survival (response variable).</li>
<li>The other is that the two variables are associated because other variables affect both supplemental oxygen and survival.</li>
<li>Use of supplemental oxygen might be a benign indicator of a greater overall preparedness of the climbers that use it.</li>
<li>Variables (like preparedness) that distort the causal relationship between the measured variables of interest (oxygen use and survival) are called <strong>confounding variables</strong></li>
<li>They are correlated with the variable of interest, and therefore preventing a decision about cause and effect.</li>
<li>With random assignment, no confounding variables will be associated with treatment except by chance.</li>
</ul>
</section>
<section id="clinical-trials" class="slide level2">
<h2>Clinical Trials</h2>
<ul>
<li>The gold standard of experimental designs is the <strong>clinical trial</strong></li>
<li>Experimental design in all areas of biology have been informed by procedures used in clinical trials</li>
<li>A clinical trial is an experimental study in which two or more treatments are assigned to human subjects</li>
<li>The design of clinical trials has been refined because the cost of making a mistake with human subjects is so high</li>
<li>Experiments on nonhuman subjects are simply called “laboratory experiments”or “field experiments”</li>
</ul>
</section>
<section id="example-of-a-clinical-trial" class="slide level2 smaller">
<h2>Example of a clinical trial</h2>
<ul>
<li>Transmission of the HIV-1 virus via sex workers contributes to the rapid spread of AIDS in Africa</li>
<li>The spermicide nonoxynol-9 had shown <em>in vitro</em> activity against HIV-1, which motivated a clinical trial by van Damme et al.&nbsp;(2002).</li>
<li>They tested whether a vaginal gel containing the chemical would reduce the risk of acquiring the disease by female sex workers.</li>
<li>Data were gathered on a volunteer sample of 765 HIV-free sex-workers in six clinics in Asia and Africa.</li>
<li>Two gel treatments were assigned randomly to women at each clinic.</li>
<li>One gel contained nonoxynol-9 and the other a placebo.</li>
<li>Neither the subjects nor the researchers making observations at the clinics knew who received the treatment and who got the placebo.</li>
</ul>
</section>
<section id="example-of-a-clinical-trial-1" class="slide level2 flexbox vcenter">
<h2>Example of a clinical trial</h2>
<p><br></p>

<img data-src="images/images_6a.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="design-components-of-a-clinical-trial" class="slide level2 smaller">
<h2>Design components of a clinical trial</h2>
<p>The goal of experimental design is to eliminate bias and to reduce sampling error when estimating and testing effects of one variable on another.</p>
<ul>
<li>To reduce bias, the experiment included:
<ul>
<li><strong>Simultaneous control group</strong>: study included both the treatment of interest and a control group (the women receiving the placebo).</li>
<li><strong>Randomization</strong>: treatments were randomly assigned to women at each clinic.</li>
<li><strong>Blinding</strong>: neither the subjects nor the clinicians knew which women were assigned which treatment.</li>
</ul></li>
</ul>
</section>
<section id="design-components-of-a-clinical-trial-1" class="slide level2 smaller">
<h2>Design components of a clinical trial</h2>
<ul>
<li>To reduce the effects of sampling error, the experiment included:
<ul>
<li><strong>Replication</strong>: study was carried out on multiple independent subjects.</li>
<li><strong>Balance</strong>: number of women was nearly equal in the two groups at every clinic.</li>
<li><strong>Blocking</strong>: subjects were grouped according to the clinic they attended, yielding multiple repetitions of the same experiment in different settings (“blocks”).</li>
</ul></li>
</ul>
</section>
<section id="simultaneous-control-group" class="slide level2 smaller">
<h2>Simultaneous control group</h2>
<ul>
<li>In clinical trials either a placebo or the currently accepted treatment should be provided.</li>
<li>In experiments requiring intrusive methods to administer treatment, such as
<ul>
<li>injections</li>
<li>surgery</li>
<li>restraint</li>
<li>confinement</li>
</ul></li>
<li>the control subjects should be perturbed in the same way as the other subjects, except for the treatment itself, as far as ethical considerations permit.</li>
</ul>
</section>
<section id="simultaneous-control-group-1" class="slide level2 smaller">
<h2>Simultaneous control group</h2>
<ul>
<li>The “sham operation”, in which surgery is carried out without the experimental treatment itself, is an example.</li>
<li>In field experiments, applying a treatment of interest may physically disturb the plots receiving it and the surrounding areas, perhaps by trampling the ground by the researchers.</li>
<li>Ideally, the same disturbance should be applied to the control plots.</li>
</ul>
</section>
<section id="randomization" class="slide level2">
<h2>Randomization</h2>
<ul>
<li>The researcher should randomize assignment of treatments to units or subjects</li>
<li>Chance rather than conscious or unconscious decision determines which units end up receiving the treatment and which the control</li>
<li>A completely randomized design is one in which treatments are assigned to all units by randomization</li>
</ul>
</section>
<section id="randomization-1" class="slide level2">
<h2>Randomization</h2>
<ul>
<li>Randomization breaks the association between possible confounding variables and the explanatory variable</li>
<li>Randomization doesn’t eliminate the variation contributed by confounding variables, only their correlation with treatment</li>
<li>Randomization ensures that variation from confounding variables is similar between the different treatment groups.</li>
</ul>
</section>
<section id="randomization-2" class="slide level2">
<h2>Randomization</h2>
<ul>
<li>Randomization should be carried out using a random process:
<ul>
<li>List all n subjects, one per row, in a computer spreadsheet.</li>
<li>Use the computer to give each individual a random number.</li>
<li>Assign treatment A to those subjects receiving the lowest numbers and treatment B to those with the highest numbers.</li>
</ul></li>
<li>Other ways of assigning treatments to subjects are almost always inferior because they do not eliminate the effects of confounding variables.</li>
<li>“Haphazard” assignment, in which the researcher chooses a treatment while trying to make it random, has repeatedly been shown to be non-random and prone to bias.</li>
</ul>
</section>
<section id="blinding" class="slide level2">
<h2>Blinding</h2>
<ul>
<li>Blinding is the process of concealing information from participants (sometimes including researchers) about which subjects receive which treatment.</li>
<li>Blinding prevents subjects and researchers from changing their behavior, consciously or unconsciously, as a result of knowing which treatment they were receiving or administering.</li>
<li>For example, studies showing that acupuncture has a significant effect on back pain are limited to those without blinding (Ernst and White 1998).</li>
</ul>
</section>
<section id="blinding-1" class="slide level2 smaller">
<h2>Blinding</h2>
<ul>
<li>In a single-blind experiment, the subjects are unaware of the treatment that they have been assigned.</li>
<li>Treatments must be indistinguishable to subjects, which prevents them from responding differently according to knowledge of treatment.</li>
<li>Blinding can also be a concern in non-human studies where animals respond to stimuli</li>
</ul>
</section>
<section id="blinding-2" class="slide level2 smaller">
<h2>Blinding</h2>
<ul>
<li>In a double-blind experiment the researchers administering the treatments and measuring the response are also unaware of which subjects are receiving which treatments
<ul>
<li>Researchers sometimes have pet hypotheses, and they might treat experimental subjects in different ways depending on their hopes for the outcome</li>
<li>Many response variables are difficult to measure and require some subjective interpretation, which makes the results prone to a bias</li>
<li>Researchers are naturally more interested in the treated subjects than the control subjects, and this increased attention can itself result in improved response</li>
</ul></li>
</ul>
</section>
<section id="blinding-3" class="slide level2">
<h2>Blinding</h2>
<ul>
<li>Reviews of medical studies have revealed that studies carried out without double- blinding exaggerated treatment effects by 16% on average compared with studies carried out with double-blinding (Jüni et al.&nbsp;2001).</li>
<li>Experiments on non–human subjects are also prone to bias from lack of blinding.</li>
</ul>
</section>
<section id="blinding-4" class="slide level2">
<h2>Blinding</h2>
<ul>
<li>Bebarta et al.(2003) reviewed 290 two-treatment experiments carried out on animals or on cell lines. The odds of detecting a positive effect of treatment were more than threefold higher in studies without blinding than in studies with blinding.</li>
<li>Blinding can be incorporated into experiments on nonhuman subjects using coded tags that identify the subject to a “blind” observer without revealing the treatment (and who measures units from different treatments in random order).</li>
</ul>
</section>
<section id="replication" class="slide level2">
<h2>Replication</h2>
<ul>
<li>The goal of experiments is to estimate and test treatment effects against the background of variation between individuals (“noise”) caused by other variables</li>
<li>One way to reduce noise is to make the experimental conditions constant</li>
<li>In field experiments, however, highly constant experimental conditions might not be feasible nor desirable</li>
<li>By limiting the conditions of an experiment, we also limit the generality of the results</li>
<li>Another way to make treatment effects stand out is to include extreme treatments and to replicate the data.</li>
</ul>
</section>
<section id="replication-1" class="slide level2 smaller">
<h2>Replication</h2>
<ul>
<li>Replication is the assignment of each treatment to multiple, independent experimental units.</li>
<li>Without replication, we would not know whether response differences were due to the treatments or just chance differences between the treatments caused by other factors.</li>
<li>Studies that use more units (i.e.&nbsp;that have larger sample sizes) will have smaller standard errors and a higher probability of getting the correct answer from a hypothesis test.</li>
<li>Larger samples mean more information, and more information means better estimates and more powerful tests.</li>
</ul>
</section>
<section id="replication-2" class="slide level2 smaller">
<h2>Replication</h2>
<ul>
<li>Replication is not about the number of plants or animals used, but the number of independent units in the experiment. An “experimental unit” is the independent unit to which treatments are assigned.</li>
<li>The figure shows three experimental designs used to compare plant growth under two temperature treatments (indicated by the shading of the pots). The first two designs are un-replicated.</li>
</ul>
</section>
<section id="pseudoreplication" class="slide level2 smaller">
<h2>Pseudoreplication</h2>

<img data-src="images/images_6a.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="balance" class="slide level2">
<h2>Balance</h2>
<ul>
<li>A study design is balanced if all treatments have the same sample size.</li>
<li>Conversely, a design is unbalanced if there are unequal sample sizes between treatments.</li>
<li>Balance is a second way to reduce the influence of sampling error on estimation and hypothesis testing.</li>
<li>To appreciate this, look again at the equation for the standard error of the difference between two treatment means.</li>
</ul>

<img data-src="images/images_6a.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="balance-1" class="slide level2">
<h2>Balance</h2>
<ul>
<li>For a fixed total number of experimental units, n1 + n2, the standard error is smallest when n1 and n2 are equal.</li>
<li>Balance has other benefits. For example, ANOVA is more robust to departures from the assumption of equal variances when designs are balanced or nearly so.</li>
</ul>
</section>
<section id="blocking" class="slide level2">
<h2>Blocking</h2>
<ul>
<li>Blocking is the grouping of experimental units that have similar properties. Within each block, treatments are randomly assigned to experimental units.</li>
<li>Blocking essentially repeats the same, completely randomized experiment multiple times, once for each block.</li>
<li>Differences between treatments are only evaluated within blocks, and in this way the component of variation arising from differences between blocks is discarded.</li>
</ul>

<img data-src="images/images_6a.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="blocking-paired-designs" class="slide level2">
<h2>Blocking | Paired designs</h2>
<ul>
<li>For example, consider the design choices for a two-treatment experiment to investigate the effect of clear cutting on salamander density.</li>
<li>In the completely randomized (“two-sample”) design we take a random sample of forest plots from the population and then randomly assign each plot to either the clear-cut treatment or the no clear-cut treatment.</li>
<li>In the paired design we take a random sample of forest plots and clear-cut a randomly chosen half of each plot, leaving the other half untouched.</li>
</ul>
</section>
<section id="blocking-paired-designs-1" class="slide level2">
<h2>Blocking | Paired designs</h2>
<ul>
<li>In the paired design, measurements on adjacent plot-halves are not independent. This is because they are likely to be similar in soil, water, sunlight, and other conditions that affect the number of salamanders.</li>
<li>As a result, we must analyze paired data differently than when every plot is independent of all the others, as in the case of the two-sample design.</li>
<li>Paired design is usually more powerful than completely randomized design because it controls for a lot of the extraneous variation between plots or sampling units that sometimes obscures the effects we are looking for.</li>
</ul>
</section>
<section id="blocking-paired-designs-2" class="slide level2 smaller">
<h2>Blocking | Paired designs</h2>

<img data-src="images/images_6a.009.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="blocking-randomized-complete-block-design" class="slide level2">
<h2>Blocking | Randomized complete block design</h2>
<ul>
<li>RCB design is analogous to the paired design, but may have more than two treatments. Each treatment is applied once to every block.</li>
<li>As in the paired design, treatment effects in a randomized block design are measured by differences between treatments exclusively within blocks.</li>
<li>By accounting for some sources of sampling variation blocking can make differences between treatments stand out.</li>
<li>Blocking is worthwhile if units within blocks are relatively homogeneous, apart from treatment effects, and units belonging to different blocks vary because of environmental or other differences.</li>
</ul>
</section>
<section id="what-if-you-cant-do-experiments" class="slide level2">
<h2>What if you can’t do experiments?</h2>
<ul>
<li>Experimental studies are not always feasible, in which case we must fall back upon observational studies.</li>
<li>The best observational studies incorporate as many of the features of good experimental design as possible to minimize bias (e.g., blinding) and the impact of sampling error (e.g., replication, balance, blocking, and even extreme treatments) except for one: randomization.</li>
<li>Randomization is out of the question, because in an observational study the researcher does not assign treatments to subjects.</li>
<li>Two strategies are used to limit the effects of confounding variables on a difference between treatments in a controlled observational study: matching; and adjusting for known confounding variables (covariates).</li>
</ul>
</section></section>
<section>
<section id="how-to-present-your-statistical-results" class="title-slide slide level1 center">
<h1>How to present your statistical results</h1>

</section>
<section id="style-of-a-results-section" class="slide level2 flexbox vcenter">
<h2>Style of a results section</h2>
<ul>
<li>Write the text of the Results section concisely and objectively.</li>
<li>The passive voice will likely dominate here, but use the active voice as much as possible.</li>
<li>Use the past tense.</li>
<li>Avoid repetitive paragraph structures. Do not interpret the data here.</li>
</ul>
</section>
<section id="function-of-a-results-section" class="slide level2 flexbox vcenter">
<h2>Function of a results section</h2>
<ul>
<li><p>The function is to objectively present your key results, without interpretation, in an orderly and logical sequence using both text and illustrative materials (Tables and Figures).</p></li>
<li><p>The results section always begins with text, reporting the key results and referring to figures and tables as you proceed.</p></li>
<li><p>The text of the Results section should be crafted to follow this sequence and highlight the evidence needed to answer the questions/hypotheses you investigated.</p></li>
<li><p>Important negative results should be reported, too. Authors usually write the text of the results section based upon the sequence of Tables and Figures.</p></li>
</ul>
</section>
<section id="summaries-of-the-statistical-analyses" class="slide level2 smaller flexbox vcenter">
<h2>Summaries of the statistical analyses</h2>
<p>May appear either in the text (usually parenthetically) or in the relevant Tables or Figures (in the legend or as footnotes to the Table or Figure). Each Table and Figure must be referenced in the text portion of the results, and you must tell the reader what the key result(s) is that each Table or Figure conveys.</p>
<ul>
<li>Tables and Figures are assigned numbers separately and in the sequence that you will refer to them from the text.
<ul>
<li>The first Table you refer to is Table 1, the next Table 2 and so forth.</li>
<li>Similarly, the first Figure is Figure 1, the next Figure 2, etc.</li>
</ul></li>
<li>Each Table or Figure must include a brief description of the results being presented and other necessary information in a legend.
<ul>
<li>Table legends go above the Table; tables are read from top to bottom.</li>
<li>Figure legends go below the figure; figures are usually viewed from bottom to top.</li>
</ul></li>
<li>When referring to a Figure from the text, “Figure” is abbreviated as Fig.,e.g., (Fig. 1. Table is never abbreviated, e.g., Table 1.</li>
</ul>
</section>
<section id="example" class="slide level2 flexbox vcenter">
<h2>Example</h2>
<p>For example, suppose you asked the question, “Is the average height of male students the same as female students in a pool of randomly selected Biology majors?” You would first collect height data from large random samples of male and female students. You would then calculate the descriptive statistics for those samples (mean, SD, n, range, etc) and plot these numbers. Suppose you found that male Biology majors are, on average, 12.5 cm taller than female majors; this is the answer to the question. Notice that the outcome of a statistical analysis is not a key result, but rather an analytical tool that helps us understand what is our key result.</p>
</section>
<section id="differences-directionality-and-magnitude" class="slide level2 smaller flexbox vcenter">
<h2>Differences, directionality, and magnitude</h2>
<ul>
<li><p>Report your results so as to provide as much information as possible to the reader about the nature of differences or relationships.</p></li>
<li><p>For example, if you are testing for differences among groups, and you find a significant difference, it is not sufficient to simply report that “groups A and B were significantly different”. How are they different? How much are they different?</p></li>
<li><p>It is much more informative to say something like, “Group A individuals were 23% larger than those in Group B”, or, “Group B pups gained weight at twice the rate of Group A pups.”</p></li>
<li><p>Report the direction of differences (greater, larger, smaller, etc) and the magnitude of differences (% difference, how many times, etc.) whenever possible.</p></li>
</ul>
</section>
<section id="statistical-results-in-text" class="slide level2 smaller flexbox vcenter">
<h2>Statistical results in text</h2>
<ul>
<li><p>Statistical test summaries (test name, p-value) are usually reported parenthetically in conjunction with the biological results they support. This parenthetical reference should include the statistical test used, the value, degrees of freedom and the level of significance.</p></li>
<li><p>For example, if you found that the mean height of male Biology majors was significantly larger than that of female Biology majors, you might report this result (in blue) and your statistical conclusion (shown in red) as follows:</p>
<ul>
<li>“Males (180.5 ± 5.1 cm; n=34) averaged 12.5 cm taller than females (168 ± 7.6 cm; n=34) in the pool of Biology majors (two-sample t-test, t = 5.78, 33 d.f., p &lt; 0.001).”</li>
</ul></li>
<li><p>If the summary statistics are shown in a figure, the sentence above need not report them specifically, but must include a reference to the figure where they may be seen:</p>
<ul>
<li>“Males averaged 12.5 cm taller than females in the pool of Biology majors (two-sample t-test, t = 5.78, 33 d.f., p &lt; 0.001; Fig. 1).”</li>
</ul></li>
</ul>
</section>
<section id="statistical-results-in-text-1" class="slide level2 flexbox vcenter">
<h2>Statistical results in text</h2>
<ul>
<li>Always enter the appropriate units when reporting data or summary statistics.
<ul>
<li>for an individual value you would write, “the mean length was 10 cm”, or, “the maximum time was 140 min.”</li>
<li>When including a measure of variability, place the unit after the error value, e.g., “…was 10 ± 2.3 m”.</li>
<li>Likewise place the unit after the last in a series of numbers all having the same unit. For example: “lengths of 5, 10, 15, and 20 m”, or “no differences were observed after 2, 4, 6, or 8 min. of incubation”.</li>
</ul></li>
</ul>

<div class="quarto-auto-generated-content">
<div class="footer footer-default">

</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'slow',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>