<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.30">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Bill Cresko">

<title>Week 3 Computational Tools for Bioengineers – Comp Tools Bioengineers F25</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-de070a7b0ab54f8780927367ac907214.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-f7fe251563e67c05764dc4775ff48cfc.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">Comp Tools Bioengineers F25</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Syllabus.html"> 
<span class="menu-text">Syllabus</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Schedule.html"> 
<span class="menu-text">Schedule</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Lectures.html"> 
<span class="menu-text">Lectures</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Homework.html"> 
<span class="menu-text">Homeworks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Software.html"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Policies.html"> 
<span class="menu-text">Policies</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Resources.html"> 
<span class="menu-text">Resources</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#goals-for-today" id="toc-goals-for-today" class="nav-link active" data-scroll-target="#goals-for-today">Goals for today</a></li>
  <li><a href="#the-command-line" id="toc-the-command-line" class="nav-link" data-scroll-target="#the-command-line">The Command Line</a>
  <ul class="collapse">
  <li><a href="#making-new-files" id="toc-making-new-files" class="nav-link" data-scroll-target="#making-new-files">Making new files</a></li>
  <li><a href="#things-to-keep-in-mind" id="toc-things-to-keep-in-mind" class="nav-link" data-scroll-target="#things-to-keep-in-mind">Things to keep in mind</a></li>
  <li><a href="#practice-making-files-and-directories" id="toc-practice-making-files-and-directories" class="nav-link" data-scroll-target="#practice-making-files-and-directories">Practice making files and directories</a></li>
  </ul></li>
  <li><a href="#reading-and-writing-files-in-unix" id="toc-reading-and-writing-files-in-unix" class="nav-link" data-scroll-target="#reading-and-writing-files-in-unix">Reading and Writing Files in Unix</a>
  <ul class="collapse">
  <li><a href="#commands-for-reading-files" id="toc-commands-for-reading-files" class="nav-link" data-scroll-target="#commands-for-reading-files">Commands for Reading Files</a></li>
  <li><a href="#input-redirection" id="toc-input-redirection" class="nav-link" data-scroll-target="#input-redirection">Input Redirection</a></li>
  <li><a href="#output-redirection" id="toc-output-redirection" class="nav-link" data-scroll-target="#output-redirection">Output Redirection</a></li>
  <li><a href="#unix-three-standard-streams" id="toc-unix-three-standard-streams" class="nav-link" data-scroll-target="#unix-three-standard-streams">Unix Three Standard Streams</a></li>
  <li><a href="#visualizing-data-streams" id="toc-visualizing-data-streams" class="nav-link" data-scroll-target="#visualizing-data-streams">Visualizing Data Streams</a></li>
  <li><a href="#basic-stream-redirection" id="toc-basic-stream-redirection" class="nav-link" data-scroll-target="#basic-stream-redirection">Basic Stream Redirection</a></li>
  <li><a href="#bioinformatics-example" id="toc-bioinformatics-example" class="nav-link" data-scroll-target="#bioinformatics-example">Bioinformatics Example</a></li>
  </ul></li>
  <li><a href="#unix-pipes" id="toc-unix-pipes" class="nav-link" data-scroll-target="#unix-pipes">Unix Pipes</a>
  <ul class="collapse">
  <li><a href="#what-are-pipes" id="toc-what-are-pipes" class="nav-link" data-scroll-target="#what-are-pipes">What are Pipes?</a></li>
  <li><a href="#common-pipe-patterns" id="toc-common-pipe-patterns" class="nav-link" data-scroll-target="#common-pipe-patterns">Common Pipe Patterns</a></li>
  <li><a href="#advanced-pipe-examples" id="toc-advanced-pipe-examples" class="nav-link" data-scroll-target="#advanced-pipe-examples">Advanced Pipe Examples</a></li>
  <li><a href="#tips-for-working-with-files-and-pipes" id="toc-tips-for-working-with-files-and-pipes" class="nav-link" data-scroll-target="#tips-for-working-with-files-and-pipes">Tips for Working with Files and Pipes</a></li>
  </ul></li>
  <li><a href="#working-with-large-genomic-data-files" id="toc-working-with-large-genomic-data-files" class="nav-link" data-scroll-target="#working-with-large-genomic-data-files">Working with Large Genomic Data Files</a>
  <ul class="collapse">
  <li><a href="#human-chromosome-data" id="toc-human-chromosome-data" class="nav-link" data-scroll-target="#human-chromosome-data">Human Chromosome Data</a></li>
  <li><a href="#downloading-genomic-data" id="toc-downloading-genomic-data" class="nav-link" data-scroll-target="#downloading-genomic-data">Downloading Genomic Data</a></li>
  <li><a href="#examining-large-files-efficiently" id="toc-examining-large-files-efficiently" class="nav-link" data-scroll-target="#examining-large-files-efficiently">Examining Large Files Efficiently</a></li>
  <li><a href="#creating-a-subset-for-analysis" id="toc-creating-a-subset-for-analysis" class="nav-link" data-scroll-target="#creating-a-subset-for-analysis">Creating a Subset for Analysis</a></li>
  <li><a href="#processing-fasta-files-with-pipes" id="toc-processing-fasta-files-with-pipes" class="nav-link" data-scroll-target="#processing-fasta-files-with-pipes">Processing FASTA Files with Pipes</a></li>
  <li><a href="#finding-patterns-in-genomic-data" id="toc-finding-patterns-in-genomic-data" class="nav-link" data-scroll-target="#finding-patterns-in-genomic-data">Finding Patterns in Genomic Data</a></li>
  <li><a href="#combining-downloads-with-processing" id="toc-combining-downloads-with-processing" class="nav-link" data-scroll-target="#combining-downloads-with-processing">Combining Downloads with Processing</a></li>
  <li><a href="#key-takeaways" id="toc-key-takeaways" class="nav-link" data-scroll-target="#key-takeaways">Key Takeaways</a></li>
  <li><a href="#practice-exercise-genomic-data-pipeline" id="toc-practice-exercise-genomic-data-pipeline" class="nav-link" data-scroll-target="#practice-exercise-genomic-data-pipeline">Practice Exercise: Genomic Data Pipeline</a></li>
  </ul></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="Week3_BioE_Comp_Tools_2025.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Week 3 Computational Tools for Bioengineers</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Bill Cresko </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.1     ✔ stringr   1.5.2
✔ ggplot2   4.0.0     ✔ tibble    3.3.0
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.1.0     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_minimal</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="goals-for-today" class="level2">
<h2 class="anchored" data-anchor-id="goals-for-today">Goals for today</h2>
<ul>
<li>Finish advanced Unix</li>
<li>Work with human chromosome 1 sequence data</li>
<li>Tidy Data</li>
<li>R and RStudio</li>
</ul>
</section>
<section id="the-command-line" class="level1">
<h1>The Command Line</h1>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/w1_code.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
<section id="making-new-files" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="making-new-files">Making new files</h2>
<ul>
<li>Make new folders: <code>mkdir</code></li>
<li>Make new files: <code>nano</code>, <code>touch</code></li>
<li>Rename files: <code>mv</code></li>
<li>Move files: <code>mv</code></li>
<li>Copy files: <code>cp</code></li>
<li>Delete files: <code>rm</code></li>
<li>Delete directories: <code>rmdir</code></li>
<li>Examining file length: <code>wc</code></li>
<li>Reading files: <code>cat</code></li>
<li>Looking at beginning or end: <code>head</code> or <code>tail</code></li>
</ul>
</section>
<section id="things-to-keep-in-mind" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="things-to-keep-in-mind">Things to keep in mind</h2>
<ul>
<li>The shell trusts you
<ul>
<li>It will delete files you say to delete</li>
<li>It will override files if you name 2 things the same</li>
</ul></li>
<li>Naming conventions
<ul>
<li>Avoid spaces</li>
<li>Don’t start with a –</li>
<li>Stick to letters, numbers, . , -, and _</li>
</ul></li>
<li>Use appropriate file extensions in file names
<ul>
<li>Some software expect files with certain extensions (.fasta, .txt, etc.)</li>
</ul></li>
</ul>
</section>
<section id="practice-making-files-and-directories" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="practice-making-files-and-directories">Practice making files and directories</h2>
<ul>
<li>Make a new directory called whatever you’d like.</li>
<li>Add a file named <code>Practice.txt</code> to the directory and add some text to it</li>
<li>Read the contents of the file and get its length</li>
<li>Rename the file to <code>Super_practice.txt</code></li>
<li>Move the file to a new folder named <code>Testing</code></li>
<li>Make a copy of the file named <code>Super_practice_copy.txt</code></li>
<li>Read the contents of the file and get its length to make sure it’s the same as <code>Super_practice.txt</code></li>
<li>Delete the original <code>Super_practice.txt</code></li>
</ul>
</section>
</section>
<section id="reading-and-writing-files-in-unix" class="level1">
<h1>Reading and Writing Files in Unix</h1>
<section id="commands-for-reading-files" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="commands-for-reading-files">Commands for Reading Files</h2>
<ul>
<li><strong><code>cat</code></strong> = concatenate and display files
<ul>
<li>Shows entire file at once</li>
<li>Good for small files</li>
</ul></li>
<li><strong><code>less</code></strong> = view file page by page
<ul>
<li>Navigate with <code>space</code> (forward), <code>b</code> (back), <code>q</code> (quit)</li>
<li>Search with <code>/pattern</code></li>
</ul></li>
<li><strong><code>more</code></strong> = similar to less but simpler
<ul>
<li>Space to go forward, <code>q</code> to quit</li>
</ul></li>
<li><strong><code>head</code></strong> = display first 10 lines (default)
<ul>
<li><code>head -n 20 file.txt</code> shows first 20 lines</li>
</ul></li>
<li><strong><code>tail</code></strong> = display last 10 lines (default)
<ul>
<li><code>tail -n 20 file.txt</code> shows last 20 lines</li>
<li><code>tail -f file.txt</code> follows file as it grows (useful for logs)</li>
</ul></li>
</ul>
</section>
<section id="input-redirection" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="input-redirection">Input Redirection</h2>
<ul>
<li><strong><code>&lt;</code></strong> = redirect input from a file
<ul>
<li><code>command &lt; input.txt</code></li>
<li>Sends contents of file as input to command</li>
</ul></li>
<li><strong><code>&lt;&lt;</code></strong> = here document (multi-line input)
<ul>
<li>Allows you to provide multi-line input directly in the terminal</li>
</ul></li>
<li>Example uses:
<ul>
<li><code>wc -l &lt; data.txt</code> counts lines in data.txt</li>
<li><code>sort &lt; names.txt</code> sorts contents of names.txt</li>
</ul></li>
</ul>
</section>
<section id="output-redirection" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="output-redirection">Output Redirection</h2>
<ul>
<li><strong><code>&gt;</code></strong> = redirect output to a file (overwrites)
<ul>
<li><code>ls -l &gt; filelist.txt</code></li>
<li>Creates new file or overwrites existing</li>
</ul></li>
<li><strong><code>&gt;&gt;</code></strong> = append output to a file
<ul>
<li><code>echo "new line" &gt;&gt; existing.txt</code></li>
<li>Adds to the end of existing file</li>
</ul></li>
<li><strong><code>2&gt;</code></strong> = redirect error messages
<ul>
<li><code>command 2&gt; errors.txt</code></li>
<li>Captures error messages separately</li>
</ul></li>
<li><strong><code>&amp;&gt;</code></strong> = redirect both output and errors
<ul>
<li><code>command &amp;&gt; all_output.txt</code></li>
</ul></li>
</ul>
</section>
<section id="unix-three-standard-streams" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="unix-three-standard-streams">Unix Three Standard Streams</h2>
<ul>
<li><strong>stdin (0)</strong> = Standard Input
<ul>
<li>Where programs read input from</li>
<li>Default: keyboard</li>
</ul></li>
<li><strong>stdout (1)</strong> = Standard Output
<ul>
<li>Where programs write normal output</li>
<li>Default: terminal screen</li>
</ul></li>
<li><strong>stderr (2)</strong> = Standard Error
<ul>
<li>Where programs write error messages</li>
<li>Default: terminal screen (same as stdout)</li>
</ul></li>
</ul>
</section>
<section id="visualizing-data-streams" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="visualizing-data-streams">Visualizing Data Streams</h2>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Every Unix program has these three streams</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>         <span class="ex">┌──────────────┐</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="ex">stdin</span> ───<span class="op">&gt;</span>│              │───<span class="op">&gt;</span> stdout</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>    <span class="kw">(</span><span class="ex">0</span><span class="kw">)</span>  <span class="ex">│</span>   Program    │      <span class="er">(</span><span class="ex">1</span><span class="kw">)</span></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>         <span class="ex">│</span>              │───<span class="op">&gt;</span> stderr</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>         <span class="ex">└──────────────┘</span>      <span class="er">(</span><span class="ex">2</span><span class="kw">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li>File descriptor numbers: 0, 1, 2</li>
<li>Can redirect each stream independently</li>
<li>Programs don’t know if streams are redirected</li>
<li>This abstraction is key to Unix philosophy</li>
</ul>
</section>
<section id="basic-stream-redirection" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="basic-stream-redirection">Basic Stream Redirection</h2>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Redirect stdout to file (equivalent ways)</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> <span class="op">&gt;</span> files.txt</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> <span class="dv">1</span><span class="op">&gt;</span> files.txt      <span class="co"># Explicitly using fd 1</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Redirect stderr to file</span></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> /nonexistent <span class="dv">2</span><span class="op">&gt;</span> errors.txt</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Redirect both stdout and stderr to same file</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-l</span> /nonexistent <span class="op">&amp;&gt;</span> all_output.txt</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="bioinformatics-example" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="bioinformatics-example">Bioinformatics Example</h2>
<p>Processing sequence data with stream control:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># FASTQ processing pipeline with error handling</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">zcat</span> reads.fastq.gz <span class="dv">2</span><span class="op">&gt;</span> unzip_errors.log <span class="kw">|</span> </span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="ex">fastqc</span> stdin <span class="dv">2</span><span class="op">&gt;</span> qc_errors.log <span class="kw">|</span> </span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="ex">trimmomatic</span> <span class="dv">2</span><span class="op">&gt;</span> trim_errors.log <span class="kw">|</span> </span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="ex">bowtie2</span> <span class="at">-x</span> genome <span class="at">-</span> <span class="dv">2</span><span class="op">&gt;</span> alignment_stats.txt <span class="kw">|</span> </span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> sort <span class="dv">2</span><span class="op">&gt;</span> sort_errors.log <span class="kw">|</span> </span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a><span class="ex">samtools</span> index <span class="at">-</span> <span class="dv">2</span><span class="op">&gt;</span> index_errors.log</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Check all error logs at once</span></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> <span class="pp">*</span>_errors.log <span class="kw">|</span> <span class="fu">grep</span> <span class="at">-E</span> <span class="st">"ERROR|WARNING"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="unix-pipes" class="level1 flexbox vcenter">
<h1 class="flexbox vcenter">Unix Pipes</h1>
<section id="what-are-pipes" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="what-are-pipes">What are Pipes?</h2>
<ul>
<li><strong><code>|</code></strong> = the pipe operator
<ul>
<li>Takes output from one command as input to another</li>
<li>Chains commands together</li>
<li>No intermediate files needed!</li>
</ul></li>
<li>Basic syntax:
<ul>
<li><code>command1 | command2 | command3</code></li>
</ul></li>
<li>Power of Unix philosophy:
<ul>
<li>Small tools that do one thing well</li>
<li>Combine them to do complex tasks</li>
</ul></li>
</ul>
</section>
<section id="common-pipe-patterns" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="common-pipe-patterns">Common Pipe Patterns</h2>
<ul>
<li><strong>Counting patterns:</strong>
<ul>
<li><code>grep "pattern" file.txt | wc -l</code></li>
<li>Count lines matching a pattern</li>
</ul></li>
<li><strong>Sorting and uniqueness:</strong>
<ul>
<li><code>cat file.txt | sort | uniq</code></li>
<li>Sort lines and remove duplicates</li>
</ul></li>
<li><strong>Finding top/bottom items:</strong>
<ul>
<li><code>sort data.txt | head -5</code></li>
<li>Get top 5 after sorting</li>
</ul></li>
<li><strong>Filtering and processing:</strong>
<ul>
<li><code>ls -l | grep ".txt" | awk '{print $5, $9}'</code></li>
<li>List txt files with sizes</li>
</ul></li>
</ul>
</section>
<section id="advanced-pipe-examples" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="advanced-pipe-examples">Advanced Pipe Examples</h2>
<ul>
<li><strong>Multi-step data processing:</strong>
<ul>
<li><code>cat data.csv | cut -d',' -f2 | sort | uniq -c | sort -rn</code></li>
<li>Extract column 2, count unique values, sort by frequency</li>
</ul></li>
<li><strong>Complex transformations:</strong>
<ul>
<li><code>find . -name "*.txt" | xargs wc -l | sort -n</code></li>
<li>Find all txt files and sort by line count</li>
</ul></li>
<li><strong>Real-time monitoring:</strong>
<ul>
<li><code>tail -f logfile.txt | grep ERROR</code></li>
<li>Watch log file for errors in real-time</li>
</ul></li>
</ul>
</section>
<section id="tips-for-working-with-files-and-pipes" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="tips-for-working-with-files-and-pipes">Tips for Working with Files and Pipes</h2>
<ul>
<li>Test pipes step by step
<ul>
<li>Build complex pipes incrementally</li>
<li>Check output at each stage</li>
</ul></li>
<li>Use <code>less</code> for large files instead of <code>cat</code></li>
<li>Remember: <code>&gt;</code> overwrites, <code>&gt;&gt;</code> appends</li>
<li>Pipe efficiency:
<ul>
<li>Filter early in the pipeline</li>
<li>Reduces data passed between commands</li>
</ul></li>
<li>Save intermediate results when debugging complex pipes # Unix Data Streams</li>
</ul>
</section>
</section>
<section id="working-with-large-genomic-data-files" class="level1">
<h1>Working with Large Genomic Data Files</h1>
<section id="human-chromosome-data" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="human-chromosome-data">Human Chromosome Data</h2>
<ul>
<li>Genomic data files can be massive (human genome ~6 billion base pairs)</li>
<li>FASTA format is standard for sequence data</li>
<li>Perfect use case for Unix pipes and redirection</li>
<li>Let’s work with human genome assembly version 38 (~3.1 GB)</li>
</ul>
</section>
<section id="downloading-genomic-data" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="downloading-genomic-data">Downloading Genomic Data</h2>
<p>Download version 38 of the human genome from NCBI:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Using `wget` to download human genome</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/<span class="dt">\</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_genomic.fna.gz</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Or using `curl` and save with specific name</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-o</span> human_genome.fa.gz https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/<span class="dt">\</span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_genomic.fna.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><code>wget</code> and <code>curl</code> are both tools for downloading files</li>
<li>The backslash allows long URLs to span multiple lines</li>
<li><code>.gz</code> extension means the file is compressed</li>
</ul>
</section>
<section id="examining-large-files-efficiently" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="examining-large-files-efficiently">Examining Large Files Efficiently</h2>
<p>Don’t use <code>cat</code> on huge files! Use these approaches:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Decompress and look at first 10 lines</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> <span class="at">-c</span> human_genome.fa.gz <span class="kw">|</span> <span class="fu">head</span> <span class="at">-10</span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Check file size before and after decompression</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> human_genome.fa.gz</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a><span class="fu">gunzip</span> human_genome.fa.gz</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="at">-lh</span> human_genome.fa</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Count sequences in FASTA file (lines starting with &gt;)</span></span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="st">"^&gt;"</span> human_genome.fa.gz <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># View sequence headers only</span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="st">"^&gt;"</span> human_genome.fa.gz <span class="kw">|</span> <span class="fu">less</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="creating-a-subset-for-analysis" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="creating-a-subset-for-analysis">Creating a Subset for Analysis</h2>
<p>Make a smaller working file for testing:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract first 1MB of sequence for testing</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span> <span class="at">-n</span> 20000 human_genome.fa.gz <span class="op">&gt;</span> human_genome_subset.fa</span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Or extract specific scaffold/contig by name</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a><span class="fu">awk</span> <span class="st">'/^&gt;chr21:1000000-2000000/,/^&gt;/'</span> human_genome.fa <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span> <span class="at">-n</span> <span class="at">-1</span> <span class="op">&gt;</span> region_of_interest.fa</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a random sample of sequences</span></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="st">"^&gt;"</span> human_genome.fa <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">shuf</span> <span class="at">-n</span> 10 <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">while</span> <span class="bu">read</span> <span class="va">header</span><span class="kw">;</span> <span class="cf">do</span></span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">grep</span> <span class="at">-A</span> 1000 <span class="st">"</span><span class="va">$header</span><span class="st">"</span> human_genome.fa</span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  <span class="cf">done</span> <span class="op">&gt;</span> random_sample.fa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="processing-fasta-files-with-pipes" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="processing-fasta-files-with-pipes">Processing FASTA Files with Pipes</h2>
<p>Extract specific regions or sequences:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract just the sequence (no headers) and count bases</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> human_genome.fa <span class="kw">|</span> <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-c</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Count each type of nucleotide</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> human_genome.fa <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">fold</span> <span class="at">-w1</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sort</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">uniq</span> <span class="at">-c</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract a specific gene region (lines 1000-2000)</span></span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a><span class="fu">sed</span> <span class="at">-n</span> <span class="st">'1000,2000p'</span> human_genome.fa <span class="op">&gt;</span> gene_region.fa</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<ul>
<li><code>grep -v</code> inverts the match (excludes lines)</li>
<li><code>tr -d '\n'</code> removes newlines</li>
<li><code>fold -w1</code> puts each character on its own line</li>
</ul>
</section>
<section id="finding-patterns-in-genomic-data" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="finding-patterns-in-genomic-data">Finding Patterns in Genomic Data</h2>
<p>Search for specific sequences and motifs:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all instances of a restriction enzyme site (EcoRI: GAATTC)</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> human_genome.fa <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="at">-o</span> <span class="st">"GAATTC"</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">wc</span> <span class="at">-l</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract 100 bases around each EcoRI site</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> human_genome.fa <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="at">-oE</span> <span class="st">".{100}GAATTC.{100}"</span> <span class="op">&gt;</span> ecori_contexts.txt</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Find all start codons (ATG) and their positions</span></span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a><span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> human_genome.fa <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="at">-b</span> <span class="at">-o</span> <span class="st">"ATG"</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span> <span class="at">-20</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="combining-downloads-with-processing" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="combining-downloads-with-processing">Combining Downloads with Processing</h2>
<p>Stream processing without saving intermediate files:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Download, decompress, and process in one pipeline</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">-s</span> https://url/to/genome.fa.gz <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gunzip</span> <span class="at">-c</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cut</span> <span class="at">-c1-1000000</span> <span class="op">&gt;</span> first_megabase.txt</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Download and immediately analyze GC content</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">-O</span> <span class="at">-</span> https://url/to/genome.fa.gz <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gunzip</span> <span class="at">-c</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grep</span> <span class="at">-v</span> <span class="st">"^&gt;"</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="at">-d</span> <span class="st">'\n'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tr</span> <span class="st">'ATGC'</span> <span class="st">'0011'</span> <span class="kw">|</span> <span class="dt">\</span></span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">awk</span> <span class="st">'{s+=gsub(/1/,"")} END {print "GC%:", s/length*100}'</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="notes">
<ul>
<li>The <code>-O -</code> flag tells wget to output to stdout</li>
<li>This approach saves disk space and time</li>
</ul>
</div>
</section>
<section id="key-takeaways" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="key-takeaways">Key Takeaways</h2>
<ul>
<li><strong>Real bioinformatics often involves gigabytes of data - these Unix tools scale beautifully!</strong></li>
<li><strong>Never load entire genome files into memory</strong> - use streaming</li>
<li><strong>Combine tools</strong> - each does one thing well</li>
<li><strong>Test on subsets first</strong> - extract small portions for development</li>
<li><strong>Document your pipelines</strong> - save commands in scripts</li>
<li><strong>Use compression</strong> - work with .gz files directly when possible</li>
<li><strong>Think in streams</strong> - data flows through pipes without intermediate files</li>
</ul>
</section>
<section id="practice-exercise-genomic-data-pipeline" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="practice-exercise-genomic-data-pipeline">Practice Exercise: Genomic Data Pipeline</h2>
<p>Build a complete analysis pipeline:</p>
<ol type="1">
<li>Download a bacterial genome (smaller, ~5 MB)</li>
<li>Count the total number of genes (hint: count headers)</li>
<li>Calculate the genome size in base pairs</li>
<li>Find the 10 most common 6-base sequences</li>
<li>Extract all genes with “ribosomal” in the header</li>
<li>Save your pipeline as a shell script for reproducibility</li>
</ol>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>