<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.30">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Bill Cresko">

<title>Week 7b - Statistics for Bioengineering – BioEngineering Statistics Spring 2025</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<script src="../site_libs/quarto-html/quarto.js" type="module"></script>
<script src="../site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-de070a7b0ab54f8780927367ac907214.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-f7fe251563e67c05764dc4775ff48cfc.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../styles.css">
</head>

<body class="nav-fixed quarto-light">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../index.html">
    <span class="navbar-title">BioEngineering Statistics Spring 2025</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" role="menu" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../index.html"> 
<span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../about.html"> 
<span class="menu-text">About</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Syllabus.html"> 
<span class="menu-text">Syllabus</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Policies.html"> 
<span class="menu-text">Policies</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Schedule.html"> 
<span class="menu-text">Schedule</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Lectures.html"> 
<span class="menu-text">Lectures</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Homework.html"> 
<span class="menu-text">Homeworks</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Software.html"> 
<span class="menu-text">Software</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="../Resources.html"> 
<span class="menu-text">Resources</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
            <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#this-week" id="toc-this-week" class="nav-link active" data-scroll-target="#this-week">This week</a></li>
  <li><a href="#multifactor-anova" id="toc-multifactor-anova" class="nav-link" data-scroll-target="#multifactor-anova">Multifactor ANOVA</a>
  <ul class="collapse">
  <li><a href="#multifactor-anova-1" id="toc-multifactor-anova-1" class="nav-link" data-scroll-target="#multifactor-anova-1">Multifactor ANOVA</a></li>
  <li><a href="#multifactor-anova-key-difference-between-nested-and-factorial-designs" id="toc-multifactor-anova-key-difference-between-nested-and-factorial-designs" class="nav-link" data-scroll-target="#multifactor-anova-key-difference-between-nested-and-factorial-designs">Multifactor ANOVA | Key difference between nested and factorial designs</a></li>
  </ul></li>
  <li><a href="#factorial-designs" id="toc-factorial-designs" class="nav-link" data-scroll-target="#factorial-designs">Factorial Designs</a>
  <ul class="collapse">
  <li><a href="#multifactor-anova-2" id="toc-multifactor-anova-2" class="nav-link" data-scroll-target="#multifactor-anova-2">Multifactor ANOVA</a></li>
  <li><a href="#multifactor-anova-3" id="toc-multifactor-anova-3" class="nav-link" data-scroll-target="#multifactor-anova-3">Multifactor ANOVA</a></li>
  <li><a href="#two-factor-factorial-designs" id="toc-two-factor-factorial-designs" class="nav-link" data-scroll-target="#two-factor-factorial-designs">Two Factor Factorial Designs</a></li>
  <li><a href="#three-factor-factorial-designs" id="toc-three-factor-factorial-designs" class="nav-link" data-scroll-target="#three-factor-factorial-designs">Three Factor Factorial Designs</a></li>
  <li><a href="#factorial-anova-both-main-effects-fixed" id="toc-factorial-anova-both-main-effects-fixed" class="nav-link" data-scroll-target="#factorial-anova-both-main-effects-fixed">Factorial ANOVA | both main effects fixed</a></li>
  <li><a href="#factorial-designs-number-of-replicates" id="toc-factorial-designs-number-of-replicates" class="nav-link" data-scroll-target="#factorial-designs-number-of-replicates">Factorial Designs | Number of Replicates</a></li>
  <li><a href="#factorial-anova-both-main-effects-random" id="toc-factorial-anova-both-main-effects-random" class="nav-link" data-scroll-target="#factorial-anova-both-main-effects-random">Factorial ANOVA | both main effects random</a></li>
  <li><a href="#the-mean-squares-for-a-factorial-model" id="toc-the-mean-squares-for-a-factorial-model" class="nav-link" data-scroll-target="#the-mean-squares-for-a-factorial-model">The mean squares for a factorial model</a></li>
  <li><a href="#the-f-ratios-for-a-factorial-model" id="toc-the-f-ratios-for-a-factorial-model" class="nav-link" data-scroll-target="#the-f-ratios-for-a-factorial-model">The F-ratios for a factorial model</a></li>
  <li><a href="#interpretation-significant-main-and-interaction-effects" id="toc-interpretation-significant-main-and-interaction-effects" class="nav-link" data-scroll-target="#interpretation-significant-main-and-interaction-effects">Interpretation | significant main and interaction effects</a></li>
  <li><a href="#interaction-plots" id="toc-interaction-plots" class="nav-link" data-scroll-target="#interaction-plots">Interaction plots</a></li>
  <li><a href="#r-interlude-2-by-2-fixed-effect-factorial-anova" id="toc-r-interlude-2-by-2-fixed-effect-factorial-anova" class="nav-link" data-scroll-target="#r-interlude-2-by-2-fixed-effect-factorial-anova">R INTERLUDE | 2-by-2 fixed effect factorial ANOVA</a></li>
  <li><a href="#r-interlude-2-by-2-fixed-effect-factorial-anova-1" id="toc-r-interlude-2-by-2-fixed-effect-factorial-anova-1" class="nav-link" data-scroll-target="#r-interlude-2-by-2-fixed-effect-factorial-anova-1">R INTERLUDE | 2-by-2 fixed effect factorial ANOVA</a></li>
  <li><a href="#by-2-fixed-effect-boxplots" id="toc-by-2-fixed-effect-boxplots" class="nav-link" data-scroll-target="#by-2-fixed-effect-boxplots">2-by-2 fixed effect boxplots</a></li>
  <li><a href="#by-2-fixed-effect-boxplots-1" id="toc-by-2-fixed-effect-boxplots-1" class="nav-link" data-scroll-target="#by-2-fixed-effect-boxplots-1">2-by-2 fixed effect boxplots</a></li>
  <li><a href="#by-2-fixed-effect-boxplots-2" id="toc-by-2-fixed-effect-boxplots-2" class="nav-link" data-scroll-target="#by-2-fixed-effect-boxplots-2">2-by-2 fixed effect boxplots</a></li>
  <li><a href="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing" id="toc-fit-the-factorial-linear-model-different-ways-to-do-the-same-thing" class="nav-link" data-scroll-target="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing">Fit the factorial linear model | different ways to do the same thing</a></li>
  <li><a href="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-1" id="toc-fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-1" class="nav-link" data-scroll-target="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-1">Fit the factorial linear model | different ways to do the same thing</a></li>
  <li><a href="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-2" id="toc-fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-2" class="nav-link" data-scroll-target="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-2">Fit the factorial linear model | different ways to do the same thing</a></li>
  <li><a href="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-3" id="toc-fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-3" class="nav-link" data-scroll-target="#fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-3">Fit the factorial linear model | different ways to do the same thing</a></li>
  <li><a href="#interaction-plots-1" id="toc-interaction-plots-1" class="nav-link" data-scroll-target="#interaction-plots-1">Interaction Plots</a></li>
  <li><a href="#interaction-plots-2" id="toc-interaction-plots-2" class="nav-link" data-scroll-target="#interaction-plots-2">Interaction Plots</a></li>
  </ul></li>
  <li><a href="#means-tests-for-multifactorial-anovas" id="toc-means-tests-for-multifactorial-anovas" class="nav-link" data-scroll-target="#means-tests-for-multifactorial-anovas">Means tests for multifactorial ANOVAs</a>
  <ul class="collapse">
  <li><a href="#means-tests-factor-level-combinations-in-multi-factor-anova" id="toc-means-tests-factor-level-combinations-in-multi-factor-anova" class="nav-link" data-scroll-target="#means-tests-factor-level-combinations-in-multi-factor-anova">Means tests | factor level combinations in multi-factor ANOVA</a></li>
  <li><a href="#planned-a-priori-contrasts" id="toc-planned-a-priori-contrasts" class="nav-link" data-scroll-target="#planned-a-priori-contrasts">Planned (a priori) contrasts</a></li>
  <li><a href="#r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction" id="toc-r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction" class="nav-link" data-scroll-target="#r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction">R INTERLUDE | 2x2 Fixed-Effects Factorial ANOVA contrasts &amp; interaction</a></li>
  <li><a href="#r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction-1" id="toc-r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction-1" class="nav-link" data-scroll-target="#r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction-1">R INTERLUDE | 2x2 Fixed-Effects Factorial ANOVA contrasts &amp; interaction</a></li>
  </ul></li>
  <li><a href="#nested-anova" id="toc-nested-anova" class="nav-link" data-scroll-target="#nested-anova">Nested ANOVA</a>
  <ul class="collapse">
  <li><a href="#nested-anova-walking-stick-example" id="toc-nested-anova-walking-stick-example" class="nav-link" data-scroll-target="#nested-anova-walking-stick-example">Nested ANOVA | Walking stick example</a></li>
  <li><a href="#nested-anova-walking-stick-example-1" id="toc-nested-anova-walking-stick-example-1" class="nav-link" data-scroll-target="#nested-anova-walking-stick-example-1">Nested ANOVA | Walking stick example</a></li>
  <li><a href="#nested-anova-walking-stick-example-2" id="toc-nested-anova-walking-stick-example-2" class="nav-link" data-scroll-target="#nested-anova-walking-stick-example-2">Nested ANOVA | Walking stick example</a></li>
  <li><a href="#nested-anova-anova-table-of-results" id="toc-nested-anova-anova-table-of-results" class="nav-link" data-scroll-target="#nested-anova-anova-table-of-results">Nested ANOVA | ANOVA Table of Results</a></li>
  <li><a href="#nesting-logic" id="toc-nesting-logic" class="nav-link" data-scroll-target="#nesting-logic">Nesting Logic</a></li>
  <li><a href="#nesting-equations" id="toc-nesting-equations" class="nav-link" data-scroll-target="#nesting-equations">Nesting equations</a></li>
  <li><a href="#nesting-hypothesis-tests" id="toc-nesting-hypothesis-tests" class="nav-link" data-scroll-target="#nesting-hypothesis-tests">Nesting hypothesis tests</a></li>
  <li><a href="#nesting-ms-calculations" id="toc-nesting-ms-calculations" class="nav-link" data-scroll-target="#nesting-ms-calculations">Nesting MS calculations</a></li>
  <li><a href="#nested-anova-table-of-results" id="toc-nested-anova-table-of-results" class="nav-link" data-scroll-target="#nested-anova-table-of-results">Nested ANOVA table of results</a></li>
  <li><a href="#r-interlude-nested-anova" id="toc-r-interlude-nested-anova" class="nav-link" data-scroll-target="#r-interlude-nested-anova">R INTERLUDE | Nested ANOVA</a></li>
  <li><a href="#r-interlude-nested-anova-1" id="toc-r-interlude-nested-anova-1" class="nav-link" data-scroll-target="#r-interlude-nested-anova-1">R INTERLUDE | Nested ANOVA</a></li>
  <li><a href="#r-interlude-nested-anova-2" id="toc-r-interlude-nested-anova-2" class="nav-link" data-scroll-target="#r-interlude-nested-anova-2">R INTERLUDE | Nested ANOVA</a></li>
  <li><a href="#r-interlude-nested-anova-3" id="toc-r-interlude-nested-anova-3" class="nav-link" data-scroll-target="#r-interlude-nested-anova-3">R INTERLUDE | Nested ANOVA</a></li>
  </ul></li>
  <li><a href="#design-principles-for-planning-a-good-experiment" id="toc-design-principles-for-planning-a-good-experiment" class="nav-link" data-scroll-target="#design-principles-for-planning-a-good-experiment">Design principles for planning a good experiment</a>
  <ul class="collapse">
  <li><a href="#what-is-an-experimental-study" id="toc-what-is-an-experimental-study" class="nav-link" data-scroll-target="#what-is-an-experimental-study">What is an experimental study?</a></li>
  <li><a href="#mount-everest-example" id="toc-mount-everest-example" class="nav-link" data-scroll-target="#mount-everest-example">Mount Everest example</a></li>
  <li><a href="#mount-everest-example-1" id="toc-mount-everest-example-1" class="nav-link" data-scroll-target="#mount-everest-example-1">Mount Everest example</a></li>
  <li><a href="#clinical-trials" id="toc-clinical-trials" class="nav-link" data-scroll-target="#clinical-trials">Clinical Trials</a></li>
  <li><a href="#example-of-a-clinical-trial" id="toc-example-of-a-clinical-trial" class="nav-link" data-scroll-target="#example-of-a-clinical-trial">Example of a clinical trial</a></li>
  <li><a href="#example-of-a-clinical-trial-1" id="toc-example-of-a-clinical-trial-1" class="nav-link" data-scroll-target="#example-of-a-clinical-trial-1">Example of a clinical trial</a></li>
  <li><a href="#design-components-of-a-clinical-trial" id="toc-design-components-of-a-clinical-trial" class="nav-link" data-scroll-target="#design-components-of-a-clinical-trial">Design components of a clinical trial</a></li>
  <li><a href="#design-components-of-a-clinical-trial-1" id="toc-design-components-of-a-clinical-trial-1" class="nav-link" data-scroll-target="#design-components-of-a-clinical-trial-1">Design components of a clinical trial</a></li>
  <li><a href="#simultaneous-control-group" id="toc-simultaneous-control-group" class="nav-link" data-scroll-target="#simultaneous-control-group">1. Simultaneous control group</a></li>
  <li><a href="#simultaneous-control-group-1" id="toc-simultaneous-control-group-1" class="nav-link" data-scroll-target="#simultaneous-control-group-1">1. Simultaneous control group</a></li>
  <li><a href="#randomization" id="toc-randomization" class="nav-link" data-scroll-target="#randomization">2. Randomization</a></li>
  <li><a href="#randomization-1" id="toc-randomization-1" class="nav-link" data-scroll-target="#randomization-1">2. Randomization</a></li>
  <li><a href="#randomization-2" id="toc-randomization-2" class="nav-link" data-scroll-target="#randomization-2">2. Randomization</a></li>
  <li><a href="#randomization-types" id="toc-randomization-types" class="nav-link" data-scroll-target="#randomization-types">2. Randomization Types</a></li>
  <li><a href="#random-sampling-approaches" id="toc-random-sampling-approaches" class="nav-link" data-scroll-target="#random-sampling-approaches">2. Random sampling approaches</a></li>
  <li><a href="#blinding" id="toc-blinding" class="nav-link" data-scroll-target="#blinding">3. Blinding</a></li>
  <li><a href="#blinding-1" id="toc-blinding-1" class="nav-link" data-scroll-target="#blinding-1">3. Blinding</a></li>
  <li><a href="#blinding-2" id="toc-blinding-2" class="nav-link" data-scroll-target="#blinding-2">3. Blinding</a></li>
  <li><a href="#blinding-3" id="toc-blinding-3" class="nav-link" data-scroll-target="#blinding-3">3. Blinding</a></li>
  <li><a href="#blinding-4" id="toc-blinding-4" class="nav-link" data-scroll-target="#blinding-4">3. Blinding</a></li>
  <li><a href="#replication" id="toc-replication" class="nav-link" data-scroll-target="#replication">4. Replication</a></li>
  <li><a href="#replication-1" id="toc-replication-1" class="nav-link" data-scroll-target="#replication-1">4. Replication</a></li>
  <li><a href="#replication-2" id="toc-replication-2" class="nav-link" data-scroll-target="#replication-2">4. Replication</a></li>
  <li><a href="#pseudoreplication" id="toc-pseudoreplication" class="nav-link" data-scroll-target="#pseudoreplication">4. Pseudoreplication</a></li>
  <li><a href="#balance" id="toc-balance" class="nav-link" data-scroll-target="#balance">5. Balance</a></li>
  <li><a href="#balance-1" id="toc-balance-1" class="nav-link" data-scroll-target="#balance-1">5. Balance</a></li>
  <li><a href="#blocking" id="toc-blocking" class="nav-link" data-scroll-target="#blocking">6. Blocking</a></li>
  <li><a href="#blocking-paired-designs" id="toc-blocking-paired-designs" class="nav-link" data-scroll-target="#blocking-paired-designs">6. Blocking | Paired designs</a></li>
  <li><a href="#blocking-paired-designs-1" id="toc-blocking-paired-designs-1" class="nav-link" data-scroll-target="#blocking-paired-designs-1">6. Blocking | Paired designs</a></li>
  <li><a href="#blocking-paired-designs-2" id="toc-blocking-paired-designs-2" class="nav-link" data-scroll-target="#blocking-paired-designs-2">6. Blocking | Paired designs</a></li>
  <li><a href="#blocking-randomized-complete-block-design" id="toc-blocking-randomized-complete-block-design" class="nav-link" data-scroll-target="#blocking-randomized-complete-block-design">6. Blocking | Randomized complete block design</a></li>
  <li><a href="#what-if-you-cant-do-experiments" id="toc-what-if-you-cant-do-experiments" class="nav-link" data-scroll-target="#what-if-you-cant-do-experiments">What if you can’t do experiments?</a></li>
  </ul></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="Week7b.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Week 7b - Statistics for Bioengineering</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Bill Cresko </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(gt)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="this-week" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="this-week">This week</h2>
<ul>
<li><p>Multifactor ANOVA</p>
<ul>
<li>Factorial</li>
<li>Nested</li>
</ul></li>
<li><p>Key principles of experimental design</p></li>
<li><p>Generalized Linear Mixed Models (GLMMs)</p></li>
</ul>
</section>
<section id="multifactor-anova" class="level1">
<h1>Multifactor ANOVA</h1>
<section id="multifactor-anova-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="multifactor-anova-1">Multifactor ANOVA</h2>
<ul>
<li>The <strong>factorial ANOVA</strong> design is the most common experimental design used to investigate more than one treatment variable
<ul>
<li>In a factorial design every combination of treatments from two (or more) treatment variables is investigated.</li>
<li>The main purpose of a factorial design is to evaluate possible interactions between variables.</li>
<li>An <strong>interaction</strong> between two explanatory variables means that the effect of one variable on the response depends on the state of a second variable.</li>
</ul></li>
<li><strong>Nested ANOVA</strong> or nested design
<ul>
<li>factors might be hierarchical - in other words nested - within one another</li>
<li>The sources of variance are therefore hierarchical too</li>
<li>A common situation is where sampling of observations occurs within groups</li>
</ul></li>
</ul>
</section>
<section id="multifactor-anova-key-difference-between-nested-and-factorial-designs" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="multifactor-anova-key-difference-between-nested-and-factorial-designs">Multifactor ANOVA | Key difference between nested and factorial designs</h2>
<ul>
<li><strong>Factorial</strong> designs are
<ul>
<li>all pairwise combinations,</li>
<li>and often involve all combinations of factor levels</li>
<li>when each factor is fixed interactions can be assessed</li>
</ul></li>
<li><strong>Nested</strong> designs are hierarchical
<ul>
<li>often contain sub-replicates that are random, uncontrolled, nuisance effects</li>
<li>but the nested factors can be of interest too</li>
</ul></li>
<li>Completely nested designs therefore <strong>have no interaction terms</strong>, whereas factorial designs do</li>
<li><strong>Mixed models</strong> can have a combination of fixed and random factors that are more complicated</li>
</ul>
</section>
</section>
<section id="factorial-designs" class="level1">
<h1>Factorial Designs</h1>
<section id="multifactor-anova-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="multifactor-anova-2">Multifactor ANOVA</h2>
<ul>
<li>Relyae (2003) looked at how a moderate dose (1.6mg/L) of a commonly used pesticide, carbaryl (Sevin), affected bullfrog tadpole survival.</li>
<li>In particular, the experiment asked how the effect of carbaryl depended on whether a native predator, the red-spotted newt, was also present.</li>
<li>The experiment was carried out in 10-L tubs (experimental units), each containing 10 tadpoles.</li>
<li>The four combinations of pesticide treatment (carbaryl vs.&nbsp;water only) and predator treatment (present or absent) were randomly assigned to tubs.</li>
<li>The results showed that survival was high except when pesticide was applied together with the predator.</li>
<li>Thus, the two treatments, predation and pesticide, seem to have <strong>interacted</strong>.</li>
</ul>
</section>
<section id="multifactor-anova-3" class="level2">
<h2 class="anchored" data-anchor-id="multifactor-anova-3">Multifactor ANOVA</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.012.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="two-factor-factorial-designs" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="two-factor-factorial-designs">Two Factor Factorial Designs</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.013.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="three-factor-factorial-designs" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="three-factor-factorial-designs">Three Factor Factorial Designs</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.014.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="factorial-anova-both-main-effects-fixed" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="factorial-anova-both-main-effects-fixed">Factorial ANOVA | both main effects fixed</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.017.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="factorial-designs-number-of-replicates" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="factorial-designs-number-of-replicates">Factorial Designs | Number of Replicates</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.015.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="factorial-anova-both-main-effects-random" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="factorial-anova-both-main-effects-random">Factorial ANOVA | both main effects random</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.018.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="the-mean-squares-for-a-factorial-model" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="the-mean-squares-for-a-factorial-model">The mean squares for a factorial model</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.019.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="the-f-ratios-for-a-factorial-model" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="the-f-ratios-for-a-factorial-model">The F-ratios for a factorial model</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.020.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="interpretation-significant-main-and-interaction-effects" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="interpretation-significant-main-and-interaction-effects">Interpretation | significant main and interaction effects</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.021.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="interaction-plots" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="interaction-plots">Interaction plots</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.022.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:75.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="r-interlude-2-by-2-fixed-effect-factorial-anova" class="level2">
<h2 class="anchored" data-anchor-id="r-interlude-2-by-2-fixed-effect-factorial-anova">R INTERLUDE | 2-by-2 fixed effect factorial ANOVA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>rna_data <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"Stickle_RNAseq.tsv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(rna_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>  Individual Genotype   Microbiota Geno.Micro Gene1 Gene2 Gene3 Gene4 Gene5
1        AC1        A conventional      convA    16    15    71   153     2
2        AC2        A conventional      convA     7     6    72   123     1
3        AC3        A conventional      convA     9     8    22   178     0
4        AC4        A conventional      convA     8    18    15   128     2
5        AC5        A conventional      convA    11    19    51   210     1
6        AC6        A conventional      convA    19    20    31   148     2
  Gene6 Gene7 Gene8 Gene9 Gene10 Gene11 Gene12 Gene13 Gene14 Gene15 Gene16
1    11     9    11   540    104    866   1004     39     31      1     88
2     4    11    17   374     84    837    908     38     29     20    100
3    17    13    24   410    119    883   1261     31    105      3      5
4    14    11    25   434     63    844    900     33      1     17    116
5     9    11    15   390     84    843   1358     26    116      3     59
6    14    13    18   541     95    815   1401     34     80     10     70
  Gene17 Gene18 Gene19 Gene20 Gene21 Gene22 Gene23 Gene24 Gene25 Gene26 Gene27
1     85   1429     16     21    166      8      4      5    136    178      2
2     51   1564     17     10    177     10      9      9    102    173     12
3     25   1656     16     37    179      3     16      8    203    221     13
4    102   1587     17     23    174     16     11      6    190    170     11
5     53   1471     17     21    177      4     17      7    201    162      8
6     40   1517     16     23    173     12     10      9    116    210      5
  Gene28 Gene29 Gene30 Gene31 Gene32 Gene33 Gene34 Gene35 Gene36 Gene37 Gene38
1      5      4    825    172    196   1887     17     34    111    139     11
2      1      4    836    204    137   1800     11     44    255    120      7
3      1      9    661    341     48   2401     23     18      7    120      0
4      3      4    905    193    202   2782     17     26     15    145      9
5      3      5    931    220    261   2200     17      2    225    114     18
6      2     10    686    221     58   2285     15     53     54    107     10
  Gene39 Gene40 Gene41 Gene42 Gene43 Gene44 Gene45 Gene46 Gene47 Gene48 Gene49
1     21      4     95    294    133      8      8    143      8    112     15
2     12      5     78    303    116      9      2    322     11     95     14
3      5      9     96    401    188      7     14      9     17      8     16
4     28      5     86    344    146      7      9    176     13    135     16
5     17     10     86    337    159      8      2    123     13    142     16
6     16      6    100    277    109      8      3    369      7    114     15
  Gene50 Gene51 Gene52 Gene53 Gene54 Gene55 Gene56 Gene57 Gene58 Gene59 Gene60
1     23      2    141      4     15      9    303    112      4      2      9
2      8      0    154      5     27      9    633    219      6      3      8
3     23      2    167      8      1     14      0      2     72     49      8
4      7      3    156      4      6     16    726     85    110     23      9
5     23      2    144      6     22     14     34    346     23     62      9
6     19      3    132      7      1      6    189    125      8     19      9
  Gene61 Gene62 Gene63 Gene64 Gene65 Gene66 Gene67 Gene68 Gene69 Gene70 Gene71
1      9      8     23  12310      3    225    287      1    211    675     20
2     11     12     38  10732      4    175    285      0    178    469     23
3     25     17     31   7924      2    115    283      1    216    328     63
4     36     16     23  10259      3    112    287      1    174     88     18
5      5     11     23   9190      5    288    286      1    173    429     36
6      7     18     39  10743      3    146    282      0    203    331     26
  Gene72 Gene73 Gene74 Gene75 Gene76 Gene77 Gene78 Gene79 Gene80 Gene81 Gene82
1     35     13   8781      9     85      0      0      4     74    268    119
2     59     16   8585      9     81      0      2      2    106    149    133
3      1     13  11252      5    118      0      1      7    302     89    151
4     75     12   7089      4     87      0      1      4     98    115    126
5     49     12  10185      8     89      0      0      2    311    234    105
6     40     13   8259      8     85      0      2      4    110    217    134
  Gene83 Gene84 Gene85 Gene86 Gene87 Gene88 Gene89 Gene90 Gene91 Gene92 Gene93
1     56      1    275    118    157   1126    248     15    155     66     28
2     49      0    326    115    160   1100    250     24    147    118     34
3     37      3    247    105    221    801    195      1    151     11     37
4     42      0    233    110    169    918    245      4    148    135     31
5     47      1    330    114    193    732    253     16    148     22     32
6     40      1    191    112    179   1000    222     20    149      7     32
  Gene94 Gene95 Gene96 Gene97 Gene98 Gene99 Gene100 Gene101 Gene102 Gene103
1     50    224    131    276     12    108     937      17       1     148
2     52    270    139    237     16     98     974      18       9     130
3     90    162    142    463      0    147    1202       0       2     186
4     81    166    138    266      7    105    1048       9       9     164
5     27    314    131    316      2    145     982       7       2     154
6     72    195    134    353     18    125     729      23       5     142
  Gene104 Gene105 Gene106 Gene107 Gene108 Gene109 Gene110 Gene111 Gene112
1     385       3      49      17      15      16      77      46      37
2     269      14      95      11      10      17      55      26      20
3     188       9       3      29      21       8      23      10       0
4     364       9      46      35      12      19      49      31      35
5     176       4      72      28      11      15       2      30       9
6     427      29      62      15      20      10      16      60      25
  Gene113 Gene114 Gene115 Gene116 Gene117 Gene118 Gene119 Gene120 Gene121
1       1      21       8       1     853      37     240     776       4
2       2      20      14       2     746      43     211     724       2
3       1       8       6       1     247      60     294     185       2
4       1      25       8       2     568      55     236     465       2
5       0       9       6       2     511      53     222     675       4
6       2       9      13       1    1066      45     316    1085       2
  Gene122 Gene123 Gene124 Gene125 Gene126 Gene127 Gene128 Gene129 Gene130
1       3      24     303      40      89     142     141    1582     557
2       2      24     444      24      69     100     132    1511     609
3       3      76     145      41     120     203     255    1885     163
4       3      14     338      26     125     181     202    1744     612
5       3      77     515      19      81     201     218    2121     266
6       3      15     313      40      39      83       2    1527     481
  Gene131 Gene132 Gene133 Gene134 Gene135 Gene136 Gene137 Gene138 Gene139
1     351      55    1016      50     243      20     104     158       5
2     322      41    1441      45     250      15      88     206       6
3     507      84     907      94      47      22      62     315       5
4     366     100    1162      74     194      19      47     353       5
5     474      37    1432      96     274      22     105     110       6
6     353      69    1520      10     281      21      54     169       4
  Gene140 Gene141 Gene142 Gene143 Gene144 Gene145 Gene146 Gene147 Gene148
1       1     252      22      40     104       8      26     404     634
2       4     254      18     118      88       3      31     407     808
3       0     397      88       0      26       1      25     389     769
4       1     491      26      20     114       1      29     398     722
5       1     361      42      23      76       7      30     396     733
6       1     423       1      15     127       2      33     397     810
  Gene149 Gene150 Gene151 Gene152 Gene153 Gene154 Gene155 Gene156 Gene157
1      40      13       8      35     622      36      30      65     220
2      20      17       4      78     531      41      26      41     193
3      22      31       3       0     469      84       5      79     207
4      36      28       4      17     374      14      31      55     224
5      10      18       8       4     510     116      32      40     205
6      31      25       1      90     425       6      22      43     190
  Gene158 Gene159 Gene160 Gene161 Gene162 Gene163 Gene164 Gene165 Gene166
1       9     517      61      21     102     205      35     159     420
2      10     413      81      16      81     202      28     210     216
3       3     555      54      38     130     341      19     204      38
4       7     570      83       3     109     258      36     183     216
5       7     518      58      12      63     273      29     212     324
6       3     504      44      19     100     375      21     179     183
  Gene167 Gene168 Gene169 Gene170 Gene171 Gene172 Gene173 Gene174 Gene175
1     103     915       1     331       2     239     891    1823     489
2      83     854       0     269       0     236     750    1707     465
3      95    1084       3     369       0      87     318     581     677
4      95    1111       1     432       0      72    1042     150     608
5      83     938       1     354       0     318    1102    3281     585
6      82     960       1     335       0     156     810     448     505
  Gene176 Gene177 Gene178 Gene179 Gene180 Gene181 Gene182 Gene183 Gene184
1      44     138      94      28      71     148     276     597       2
2      99     180      98      27      80     108     349     581       5
3       2     272      27       7       3     172     302     603       5
4      78     195     123      10      56     113     324     590       6
5      95     234      75      31      91     123     267     584       4
6      61     170      84      11      37     205     261     605       3
  Gene185 Gene186 Gene187 Gene188 Gene189 Gene190 Gene191 Gene192 Gene193
1      32     157     111      90      15     186      76    3133      16
2      46     161      84      96      14     216      86    2941      28
3       6      81     162     107       1     892     526    3985      11
4      20     126      15     103       4      41     189    4552      20
5      58     155     185      94      12     248     635    3488      26
6      56      94      93      90      15    1013     310    3643      14
  Gene194 Gene195 Gene196 Gene197 Gene198 Gene199 Gene200 Gene201 Gene202
1     110     869      27      58     196      17    1431      74      58
2     114     820      48      45     194      17    1332      65      85
3     308     928       5      19     219      13    3909      82      56
4     234     820      13      47     206      18    3193      76      62
5     370     884      33      28     193      14    3491      82      67
6     182     793       3      35     205      16    2670      73      72
  Gene203 Gene204 Gene205 Gene206 Gene207 Gene208 Gene209 Gene210 Gene211
1       5      23       8       2      13     175       3    1054       4
2       9      23       2       1      12     154       1    1041       4
3       2       6       1       0      14     211       0      86      23
4       0      12       5       0      13     196       0     580      12
5       2      20       4       0      13     203       0     555       5
6       3      17       3       2      13     184       0    1069       3
  Gene212 Gene213 Gene214 Gene215 Gene216 Gene217 Gene218 Gene219 Gene220
1    2226       2      83      53     160      15       3     250       8
2    1952       6      27      82     139      19       5     239       9
3     519       3      54      13     231      30      14     252       8
4    1592       4      43      16     138      22       5     251       9
5    1248       8      96      59     132      34       2     249       9
6    1422       5      64      91     294      30       9     245       8
  Gene221 Gene222 Gene223 Gene224 Gene225 Gene226 Gene227 Gene228 Gene229
1     319    1678     293    3756     162      35     151     201      38
2     303    1447     230    2859     161      20      77     188      37
3     310    1430      73    3078     186      58      16     221       2
4     326    1542     262    3511     172      26      20     225      79
5     309    1665     204    3014     188       8     149     186      31
6     314    1467     276    2215     183      56     243     219      22
  Gene230 Gene231 Gene232 Gene233 Gene234 Gene235 Gene236 Gene237 Gene238
1      40     578    1770     247      74     141     105     239      58
2      21     553    1616     256     101      88      82     220     117
3       3     696     438     225      59       3     148     130       9
4      32     637    1553     236      77       3     141     108      64
5       4     489    1830     246      82     165     118     104     118
6       8     697    1401     256      47     152     139     302      52
  Gene239 Gene240 Gene241 Gene242 Gene243 Gene244 Gene245 Gene246 Gene247
1     335     116       4      63     364      31      67    8266       2
2     308      96       4      59     326      74      51    8664       8
3     470     208      23      95     412       3      41    8327      13
4     329     213      11      71     343      56      34    8332      13
5     380     122       4      71     402      34      66    8552       5
6     334      86      12      84     330      18      62    8886      12
  Gene248 Gene249 Gene250 Gene251 Gene252 Gene253 Gene254 Gene255 Gene256
1      39      51      13      78       2     144      93      17     627
2      48      47      24      84       6     194     106      20     611
3       8      56      19      59       3     434     176      84     853
4      29      50      22      87       4     279     152       6     615
5      67      53      31      62       5     132     123      32     779
6      25      50      11      85       2     135     203      32     825
  Gene257 Gene258 Gene259 Gene260 Gene261 Gene262 Gene263 Gene264 Gene265
1      30      14     419     102     675      46     120      23      21
2      38      17     523     113     607     111     101       8      20
3       7       2     757      94     399       1       7      60      13
4      55       5     609     103     502      57     143       3      19
5      26       3     280      81     463     130      19      41       5
6       7       7     746      96     534      44      48       7      15
  Gene266 Gene267 Gene268 Gene269 Gene270 Gene271 Gene272 Gene273 Gene274
1      16       1      62     709       3     141      11     108      69
2       9       2      54     661       2     163      17     100     110
3       5       1      52     601      22     152       6      98      30
4       7       1      69     646       7     130       8     105      58
5       2       1      69     589      16     151      11      97     167
6      15       2      57     694       8     144      15     102      67
  Gene275 Gene276 Gene277 Gene278 Gene279 Gene280 Gene281 Gene282 Gene283
1     110     216      26      39       5    2571      69     377    4921
2      98     243      27      44       4    2889      69     448    4836
3     113      72      14      60       1     485     116     495    4370
4     115     193      33      32       2    3963     100     479    4641
5     122     205      16      54       2    1215     121     455    4476
6     102      96      20      47       0    2046      47     411    4622
  Gene284 Gene285 Gene286 Gene287 Gene288 Gene289 Gene290 Gene291 Gene292
1      24     237      71    2764     125     307    1344     437     429
2      19     230      71    2742      94     303    1254     844     601
3      24     262      96    2906     155     388    1477     143     114
4      25     244      57    2606     115     427    1324    1076     522
5      22     254      75    2844      84     311    1404     312      88
6      22     263      98    2831     135     249    1524     576     150
  Gene293 Gene294 Gene295 Gene296 Gene297 Gene298 Gene299 Gene300 Gene301
1      55     157       5      24      16     594       8      55       2
2      64     194      10      22      22     498      18      44       3
3      52     264       6      30      51     785      17       1       2
4      52     280       5      26      14     679      13      37       2
5      56      91      10      26      28     680      11      18       1
6      63     171       5      27       7     845      16       4       2
  Gene302 Gene303 Gene304 Gene305 Gene306 Gene307 Gene308 Gene309 Gene310
1    7380      72       5      49      16      18     167      39     131
2    6331      90       8      43      36      11     183      36     121
3    6978     114      13      46      27      17     208       9      33
4    6661      89      11      52      19      22     211       6      59
5    6519     105      10      48      29      13     153      27      74
6    6528      87      14      53      15      17     164      50      92
  Gene311 Gene312 Gene313 Gene314 Gene315 Gene316 Gene317 Gene318 Gene319
1     655      63     115      11     140     282    1225     109       6
2     296      67      97       8     139     236    1157     114       3
3      55     180     121      19     121     338     273      91       5
4     365      74     114       5     143     291     907     103       8
5     586       5     139       9     135     363    1182     116       3
6     759     123     119      20     135     280     334      93       5
  Gene320 Gene321 Gene322 Gene323 Gene324 Gene325 Gene326 Gene327 Gene328
1     330      57      33      43     306      86      22       4      28
2     316      68      29      59     572      69      22       4      25
3     330      81      30     134     124     164      21       6      33
4     326      66      31      86     123      64      21       4      31
5     331      53      30      50     107     121      22       2      33
6     332      43      32      71     483     110      21       2      33
  Gene329 Gene330 Gene331 Gene332 Gene333 Gene334 Gene335 Gene336 Gene337
1     307      33       8     278      58       9       7      22      14
2     404      24      11     227      56       9      15      36      14
3     488      43      10     406      56       1      15       8       2
4     357      20      13     264      55      10       9      32      13
5     241      18       8     444      56       9      10      18       4
6     402      42      10     513      57       8      16      19      17
  Gene338 Gene339 Gene340 Gene341 Gene342 Gene343 Gene344 Gene345 Gene346
1      12      88       7      45    1896      35       9      19    1042
2      14     103      14      70    1717      27      21      19    2227
3       6      70       6       9    1578      11      31       8     137
4      11     127       4      48    1925      27      20      12    2354
5       9      70       8      21    1684      22      20      19    3050
6       7      88      11       8    1859      17      21      24    1112
  Gene347 Gene348 Gene349 Gene350 Gene351 Gene352 Gene353 Gene354 Gene355
1      30      77     159     164      53     122      36      18      66
2      21     114     147     160      42     132      51      22      58
3      42      25     216     236      46     152      42       6      87
4      13      67     148     223      47     128      57       6      88
5      19      41     125     211      54     125      34      29      56
6      38      81     178     250      44     128      34      17      81
  Gene356 Gene357 Gene358 Gene359 Gene360 Gene361 Gene362 Gene363 Gene364
1      14     267       9     168       2       9      18       0     120
2      22     243       5     382       2       1      21       0     101
3       3     259       9       0      10       6      51       0      45
4      20     258       8      84       3       6      39       0     185
5      10     257       6     325       3       4      27       0      15
6       8     248       5     234       5       3      41       0      64
  Gene365 Gene366 Gene367 Gene368 Gene369 Gene370 Gene371 Gene372 Gene373
1     790      51      98       6    1179     198    2562     346     296
2     747      67      94      12    1324     424    2635     295     306
3     857     259     313       1    2478       3    2814     361     313
4     765     173     190      15    1594      62    2559     334     308
5     845       2      28       1    2649     178    2586     371     292
6     801     351     276       8    1781     301    2571     339     317
  Gene374 Gene375 Gene376 Gene377 Gene378 Gene379 Gene380 Gene381 Gene382
1     612      13     166      21      61      26      27    1155     978
2     613      14     319      22      85      11      39     960     911
3     160      19      10      27      65      32      13    1108     908
4     313      16     282      21      85       2      24    1127     990
5     434      16      77      18      79      17      30    1001     955
6      93       8      57      21      61      38      27     916     938
  Gene383 Gene384 Gene385 Gene386 Gene387 Gene388 Gene389 Gene390 Gene391
1     124       3     267     245      77     128      98     272     282
2     127       1     255     245      68     135      98     512     562
3     144      37     495      52      80      24      96      97      66
4     136      22     152     189      79      85      95     334     643
5     121       1     306     184      73     115      97     334     515
6     117       4     427     325      76     111      97      18      18
  Gene392 Gene393 Gene394 Gene395 Gene396 Gene397 Gene398 Gene399 Gene400
1     208     157     592       5     241      57    1146      13    1720
2     362     233     547      34     442     101    2001      24    2700
3      43      24     692       1      56       7     286      13    1043
4     217     239     599       6     661      98    2298      21    2418
5     120     328     606      13      44      86    1498      16     800
6     126     170     713       9     394      57    1005      20     681
  Gene401 Gene402 Gene403 Gene404 Gene405 Gene406 Gene407 Gene408 Gene409
1    3289     841      44     102     137      51      78    2416     139
2    4943    1245      47     111     152      56      67    2398     139
3    1670     543      79     112     172      64      94    2285     276
4    3322    1161      61     106     151      54      87    2204     134
5    5556     606      41     113     144      48      73    2347     177
6    5801    1017      39     103     178      54      60    2375      41
  Gene410 Gene411 Gene412 Gene413 Gene414 Gene415 Gene416 Gene417 Gene418
1     228     312      14    1683      97      54      24    1101    1568
2     205     315      22    1224    1211      30      28    1015    1556
3     192     383      25     908     107       5      13     324     369
4     207     401      15    1666     481       5      19    1127     903
5     217     345      19    1569     181      38      27     855     700
6     198     278      26    1486      71      63      22    1733     882
  Gene419 Gene420 Gene421 Gene422 Gene423 Gene424 Gene425 Gene426 Gene427
1      19      36      50     787     657      69       3     826      58
2      26      45      39     756     676      79       1     493      52
3      14       5      41     210     826     126       1     231      94
4      19      23      31      87     740      48       4     277      89
5      22      60      43    1166     801     113       4     656      73
6      15      31      39     470     793      98       2     570      59
  Gene428 Gene429 Gene430 Gene431 Gene432 Gene433 Gene434 Gene435 Gene436
1      18     464     135       9     105      59      11      15     174
2      17     431     143      16     142      57      24      23     175
3      23     435     192      10      99     110       0       9     293
4      18     456     234       5     147      11      12      12      88
5      17     443     139      17     109       1      20      19     175
6      25     439     179       4     125      73       8      14     224
  Gene437 Gene438 Gene439 Gene440 Gene441 Gene442 Gene443 Gene444 Gene445
1      32      89      47      22      50     164     136       1     435
2      30      79      34      19      44     113     124       0     533
3      40       5       8      38      47      45     155       0     476
4      27      98      31      15      45     110     130       1     513
5      36      60      32      19      45     120     135       1     513
6      31      26      33      39      45     186     163       0     392
  Gene446 Gene447 Gene448 Gene449 Gene450 Gene451 Gene452 Gene453 Gene454
1      27     130    1719     370      26      42       1      19      46
2      33     141     858     325      26      54       1      10      41
3      99     177     344     145       6       1       0      21      27
4      72     179     501     146      23      13       0      12      38
5      35     174     992     342      29      18       0      10      17
6      29     180    1274     233      16      60       0      21      31
  Gene455 Gene456 Gene457 Gene458 Gene459 Gene460 Gene461 Gene462 Gene463
1       2       1      54      22     112     472     660      32      17
2       0       2      39      11     110     411     618      50      20
3      35      16      23      13     114     315     843      83      57
4       7      12      47      21     110     486     748      79      49
5       1      21      18      16     112     419     668      34      32
6      14      19      16      16     112     451     712      67      32
  Gene464 Gene465 Gene466 Gene467 Gene468 Gene469 Gene470 Gene471 Gene472
1      42     104     937     162     116     605      34      14      66
2      21      62     481     135     106     590      32      12      42
3      55       9     321     239       3     564      23       2      11
4      35       1     581     229      63     599      40      15      45
5      60      34     579     177     117     575      29      12      78
6      28     127     457     243     203     588      26       7      57
  Gene473 Gene474 Gene475 Gene476 Gene477 Gene478 Gene479 Gene480 Gene481
1      32    1073     138    1258     163      20       6      53     762
2      35     970     171    1193     199      11       9      56     740
3       7     209     178    2135     131       9      17     128     231
4       3     287     172     982     106      17       5     153     238
5      26     611     194    1686     104       3      11       4      76
6      14     577     167    1685     197      20      10     137     560
  Gene482 Gene483 Gene484 Gene485 Gene486 Gene487 Gene488 Gene489 Gene490
1     992    1082      15     423      69     730      58     218    1675
2     902     926      12     203      87     838      78     224    1728
3     201     212       2      32     326     710      66     277    1808
4    1061     678       8       1      72     825      69     280    1812
5     826    1473      18     334       8     794      82     282    1788
6      59    1122       7     198     420     643      68     262    1818
  Gene491 Gene492 Gene493 Gene494 Gene495 Gene496 Gene497 Gene498 Gene499
1     535    1937     166       3    2405     275    2008    1681      65
2     549    1528     177       3    2262     273    1959    1654      70
3     494    3667     237       7      67       6     534     439       5
4     532    3312     223       3    2556     196    1169     374      14
5     507    2511     171       2    1933     250    1330     922      29
6     478    1160     208       3    1158       1    2213      18      14
  Gene500 Gene501 Gene502 Gene503 Gene504 Gene505 Gene506 Gene507 Gene508
1       4       1      20     205      23     176      47      14      22
2       6      10      31     180      12     380      92      45      32
3       7       5      38       7      18      11       5     123     148
4       8       8      38      22      15     131      59      50      56
5       6       4      21      90      25     312       4      71      83
6       3       1      31     274      28     240       4      25      86
  Gene509 Gene510 Gene511 Gene512 Gene513 Gene514 Gene515 Gene516 Gene517
1      24     137     143     455      23       7       1       5     499
2      53     147     113     406      31       4       9      12     505
3     313     197     242     401       0      11       1      10    1154
4      58      80      80     481       4       9       2       6     472
5     154     198     141     415      31       9       3       7     742
6     332     143     127     385       5      11       8      10     661
  Gene518 Gene519 Gene520 Gene521 Gene522 Gene523 Gene524 Gene525 Gene526
1     112    1030    2644       8      76      33       2      55    3567
2     109     838    2676       9      67      51       2      36    2726
3      73     226    2727      11      21      62       2      37    1033
4      76    1538    2696       9      42      41       2      37    2523
5      99     996    2667      11      26      45       2      46    1669
6     130     298    2706       6      49      48       2      60    4378
  Gene527 Gene528 Gene529 Gene530 Gene531 Gene532 Gene533 Gene534 Gene535
1     292     312     290      64       0      57     574     144      56
2     310     323     275      41       8      67     488     140      85
3     300     283     381      90      10     131     967      37     113
4     293     283     339     117      11     111     492      82      95
5     311     317     299      16       1      92     431     117      56
6     301     337     189      16      11       5     682     128      51
  Gene536 Gene537 Gene538 Gene539 Gene540 Gene541 Gene542 Gene543 Gene544
1     231      84     101    1252      43       2    1026      89      62
2     219      63     120    1278      47       8     734      56      80
3      59      23     106    1260      23      17     633      51     111
4     348      26     106    1270      34       6     760      78      97
5      73      76     113    1261      45       5    1030      46      42
6     197     128     107    1260      45       9     643      57     109
  Gene545 Gene546 Gene547 Gene548 Gene549 Gene550 Gene551 Gene552 Gene553
1     175     324      82      27      53      53      17       6     313
2     179     285      50      33      58      29      19       3     415
3     170     394     355      16      48      51       6      47     305
4     175     384     133      26      47      48       6      11     311
5     175     246     288      23      58      55       9      19     324
6     177     295     218      20      51      32      18      32     323
  Gene554 Gene555 Gene556 Gene557 Gene558 Gene559 Gene560 Gene561 Gene562
1       7     655      17     299      19     207      15      96      13
2      17     606      19     263      19     153       9     114      25
3      59     744      27     388      21     217      11      88       7
4      20     716      27     185      18     221      11      81      28
5       9     792      24     316      19     190      14      84      11
6      52     751      17     299      20     210       9     110       0
  Gene563 Gene564 Gene565 Gene566 Gene567 Gene568 Gene569 Gene570 Gene571
1      33     158      51      28      17       1       5      19      23
2      36     141      52      25       1       1       5      20      17
3      68     183      73      49      13       1       2      22     152
4      54     162      59      22       9       1       4      20      60
5      45     186      48      14       1       1       6      19     165
6      62     201      66      34      13       1       4      21      40
  Gene572 Gene573 Gene574 Gene575 Gene576 Gene577 Gene578 Gene579 Gene580
1      52       0      42     189     252      55     639      41      52
2      88       2      63     189     239      77     668      48      41
3     142      10       5     150     296      98     228      48      14
4     185       5      42     163     272      66     609      45      22
5      48       1       9     168     297      80     409      52      14
6     112       2      48     205     299      83     458      47      41
  Gene581 Gene582 Gene583 Gene584 Gene585 Gene586 Gene587 Gene588 Gene589
1     310       9     332    1458     173      45      55     147     129
2     308       8     164    1879     325      55      46     147     116
3      95      10      96    1285       2      82       0       0     155
4     298       8     428    1606     280      76      48      46     147
5     247       9     101    1082     109      47      65     111     150
6      79       8     305    1487     343      54      71      37     133
  Gene590 Gene591 Gene592 Gene593 Gene594 Gene595 Gene596 Gene597 Gene598
1      30       0       0       2       1      14      22      11       1
2     770       4       2       2       2      22      14      24       6
3      24       1       0       1       0      22      17      13       1
4     593       4       1       1       1      24      22      32       7
5     244       5       1       2       2      17      17      21       0
6     544       3       0       1       1      21      21      21       2
  Gene599 Gene600
1      24      69
2      82      88
3      25      70
4      37      87
5       4      66
6      25      80</code></pre>
</div>
</div>
</section>
<section id="r-interlude-2-by-2-fixed-effect-factorial-anova-1" class="level2">
<h2 class="anchored" data-anchor-id="r-interlude-2-by-2-fixed-effect-factorial-anova-1">R INTERLUDE | 2-by-2 fixed effect factorial ANOVA</h2>
<ul>
<li>continuous response variable and two main effect categorical variables</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> rna_data<span class="sc">$</span>Gene100</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>microbiota <span class="ot">&lt;-</span> rna_data<span class="sc">$</span>Microbiota</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">&lt;-</span> rna_data<span class="sc">$</span>Genotype</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota)</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> genotype)</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="by-2-fixed-effect-boxplots" class="level2">
<h2 class="anchored" data-anchor-id="by-2-fixed-effect-boxplots">2-by-2 fixed effect boxplots</h2>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week7b_files/figure-html/unnamed-chunk-14-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="by-2-fixed-effect-boxplots-1" class="level2">
<h2 class="anchored" data-anchor-id="by-2-fixed-effect-boxplots-1">2-by-2 fixed effect boxplots</h2>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week7b_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="by-2-fixed-effect-boxplots-2" class="level2">
<h2 class="anchored" data-anchor-id="by-2-fixed-effect-boxplots-2">2-by-2 fixed effect boxplots</h2>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week7b_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing" class="level2">
<h2 class="anchored" data-anchor-id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing">Fit the factorial linear model | different ways to do the same thing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>rna_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(gene <span class="sc">~</span> microbiota <span class="sc">+</span> genotype <span class="sc">+</span> microbiota<span class="sc">:</span>genotype)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>rna_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>rna_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(gene <span class="sc">~</span> microbiota <span class="sc">+</span> genotype <span class="sc">+</span> microbiota<span class="sc">:</span>genotype)</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>rna_lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Examine the fitted model diagnostics and the ANOVA results table</li>
</ul>
</section>
<section id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-1" class="level2">
<h2 class="anchored" data-anchor-id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-1">Fit the factorial linear model | different ways to do the same thing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(rna_aov)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rna_aov)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(rna_aov)</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(rna_lm)</span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rna_alm)</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(rna_alm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-2" class="level2">
<h2 class="anchored" data-anchor-id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-2">Fit the factorial linear model | different ways to do the same thing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(rna_aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Response: gene
                    Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
microbiota           1  243763  243763  6.8179   0.01087 *  
genotype             1    4234    4234  0.1184   0.73170    
microbiota:genotype  1 1452066 1452066 40.6133 1.299e-08 ***
Residuals           76 2717264   35753                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">anova</span>(rna_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Analysis of Variance Table

Response: gene
                    Df  Sum Sq Mean Sq F value    Pr(&gt;F)    
microbiota           1  243763  243763  6.8179   0.01087 *  
genotype             1    4234    4234  0.1184   0.73170    
microbiota:genotype  1 1452066 1452066 40.6133 1.299e-08 ***
Residuals           76 2717264   35753                      
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
</section>
<section id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-3" class="level2">
<h2 class="anchored" data-anchor-id="fit-the-factorial-linear-model-different-ways-to-do-the-same-thing-3">Fit the factorial linear model | different ways to do the same thing</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rna_aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                    Df  Sum Sq Mean Sq F value  Pr(&gt;F)    
microbiota           1  243763  243763   6.818  0.0109 *  
genotype             1    4234    4234   0.118  0.7317    
microbiota:genotype  1 1452066 1452066  40.613 1.3e-08 ***
Residuals           76 2717264   35753                    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rna_lm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = gene ~ microbiota * genotype)

Residuals:
    Min      1Q  Median      3Q     Max 
-336.00 -131.85   19.52  144.00  507.00 

Coefficients:
                         Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)               1004.95      42.28  23.768  &lt; 2e-16 ***
microbiotamono             159.05      59.79   2.660  0.00953 ** 
genotypeB                  254.90      59.79   4.263 5.73e-05 ***
microbiotamono:genotypeB  -538.90      84.56  -6.373 1.30e-08 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 189.1 on 76 degrees of freedom
Multiple R-squared:  0.3849,    Adjusted R-squared:  0.3606 
F-statistic: 15.85 on 3 and 76 DF,  p-value: 4.254e-08</code></pre>
</div>
</div>
</section>
<section id="interaction-plots-1" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="interaction-plots-1">Interaction Plots</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>summary_df <span class="ot">&lt;-</span> rna_data <span class="sc">|&gt;</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genotype, Microbiota) <span class="sc">|&gt;</span></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_response =</span> <span class="fu">mean</span>(Gene100), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>summary_df <span class="ot">&lt;-</span> rna_data <span class="sc">|&gt;</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Genotype, Microbiota) <span class="sc">|&gt;</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">mean_response =</span> <span class="fu">mean</span>(Gene100),</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>            <span class="at">se =</span> <span class="fu">sd</span>(Gene100) <span class="sc">/</span> <span class="fu">sqrt</span>(<span class="fu">n</span>()), <span class="at">.groups =</span> <span class="st">"drop"</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a><span class="co"># ggplot interaction plot</span></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(summary_df, <span class="fu">aes</span>(<span class="at">x =</span> Genotype, <span class="at">y =</span> mean_response, <span class="at">group =</span> Microbiota, <span class="at">color =</span> Microbiota)) <span class="sc">+</span></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="fu">aes</span>(<span class="at">ymin =</span> mean_response <span class="sc">-</span> se, <span class="at">ymax =</span> mean_response <span class="sc">+</span> se), <span class="at">width =</span> <span class="fl">0.2</span>) <span class="sc">+</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Interaction Plot with SE"</span>,</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Genotype"</span>,</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Mean Expression of Gene100"</span>,</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Microbiota"</span>) <span class="sc">+</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="interaction-plots-2" class="level2">
<h2 class="anchored" data-anchor-id="interaction-plots-2">Interaction Plots</h2>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week7b_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="means-tests-for-multifactorial-anovas" class="level1">
<h1>Means tests for multifactorial ANOVAs</h1>
<section id="means-tests-factor-level-combinations-in-multi-factor-anova" class="level2">
<h2 class="anchored" data-anchor-id="means-tests-factor-level-combinations-in-multi-factor-anova">Means tests | factor level combinations in multi-factor ANOVA</h2>
<ul>
<li>The F-ratio test for a single-factor ANOVA tests for any difference among groups.</li>
<li>If we want to understand specific differences, we need further “contrasts”.</li>
<li>Unplanned comparisons (post hoc)</li>
<li>Planned comparisons (a priori)</li>
<li>Now we need to make ‘pseudo-factors’ that combine our levels of interest</li>
</ul>
</section>
<section id="planned-a-priori-contrasts" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="planned-a-priori-contrasts">Planned (a priori) contrasts</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7b.002.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction">R INTERLUDE | 2x2 Fixed-Effects Factorial ANOVA contrasts &amp; interaction</h2>
<p>continuous response and two main effect variables</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>rna_data <span class="ot">&lt;-</span><span class="fu">read.table</span>(<span class="st">"Stickle_RNAseq.tsv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">"</span><span class="sc">\t</span><span class="st">"</span>)</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>gene <span class="ot">&lt;-</span> rna_data<span class="sc">$</span>Gene80</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>microbiota <span class="ot">&lt;-</span> rna_data<span class="sc">$</span>Microbiota</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>genotype <span class="ot">&lt;-</span> rna_data<span class="sc">$</span>Genotype</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>make new “pseudo factor,” combining genotype and microbiota</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>gxm <span class="ot">&lt;-</span> <span class="fu">interaction</span>(genotype,microbiota)</span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">levels</span>(gxm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "A.conventional" "B.conventional" "A.mono"         "B.mono"        </code></pre>
</div>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> gxm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Week7b_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>specify the following 2 contrasts</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">contrasts</span>(gxm) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">c</span>(<span class="dv">2</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="sc">-</span><span class="dv">1</span>), <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="sc">-</span><span class="dv">1</span>, <span class="dv">3</span>, <span class="sc">-</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="r-interlude-2x2-fixed-effects-factorial-anova-contrasts-interaction-1">R INTERLUDE | 2x2 Fixed-Effects Factorial ANOVA contrasts &amp; interaction</h2>
<p>Fit the factorial linear model</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>rna_aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(gene <span class="sc">~</span> gxm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Examine the ANOVA table, using supplied contrasts. Figure out the appropriate titles to give them.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(rna_aov, <span class="at">split =</span> <span class="fu">list</span>(<span class="at">gxm =</span> <span class="fu">list</span>(<span class="st">'xxx'</span><span class="ot">=</span><span class="dv">1</span>,<span class="st">'xxx'</span><span class="ot">=</span><span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>What does the contrast summary tell you about the nature of the interaction?</p>
</section>
</section>
<section id="nested-anova" class="level1">
<h1>Nested ANOVA</h1>
<section id="nested-anova-walking-stick-example" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="nested-anova-walking-stick-example">Nested ANOVA | Walking stick example</h2>
<ul>
<li>Example 1: Study of “repeatability” (simple nested design)</li>
<li>The walking stick, <em>Timema cristinae</em>, is a wingless herbivorous insect on plants in chaparral habitats of California.</li>
<li>Nosil and Crespi (2006) measured individuals using digital photographs.</li>
<li>To evaluate measurement repeatability they took two separate photographs of each specimen.</li>
<li>After measuring traits on one set of photographs, they repeated the measurements on the second set.</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.001.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nested-anova-walking-stick-example-1" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nested-anova-walking-stick-example-1">Nested ANOVA | Walking stick example</h2>
<p>Each pair of dots represents the two measurements</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.002.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nested-anova-walking-stick-example-2" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nested-anova-walking-stick-example-2">Nested ANOVA | Walking stick example</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.003.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nested-anova-anova-table-of-results" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nested-anova-anova-table-of-results">Nested ANOVA | ANOVA Table of Results</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.004.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nesting-logic" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nesting-logic">Nesting Logic</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.005.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nesting-equations" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nesting-equations">Nesting equations</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.006.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nesting-hypothesis-tests" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nesting-hypothesis-tests">Nesting hypothesis tests</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.007.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nesting-ms-calculations" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nesting-ms-calculations">Nesting MS calculations</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.008.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="nested-anova-table-of-results" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="nested-anova-table-of-results">Nested ANOVA table of results</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.009.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="r-interlude-nested-anova" class="level2">
<h2 class="anchored" data-anchor-id="r-interlude-nested-anova">R INTERLUDE | Nested ANOVA</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_7a.010.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="r-interlude-nested-anova-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="r-interlude-nested-anova-1">R INTERLUDE | Nested ANOVA</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>andrew_data <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'andrew.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span>‘\t<span class="st">')</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="st">head(andrew_data)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>There are four variables: ‘TREAT’, ‘PATCH’, ‘QUAD’ and ‘ALGAE’</li>
<li>The main effect factor is TREAT</li>
<li>Make a simplified factor called TREAT2, in which 0% and 33% are a level called “low” and 66% and 100% are “high”</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>andrew_data<span class="sc">$</span>TREAT2 <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(“low”,<span class="dv">40</span>),<span class="fu">rep</span>(“high”,<span class="dv">40</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>The nested factor is PATCH - also need to turn this into a factor</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>andrew_data<span class="sc">$</span>PATCH <span class="ot">&lt;-</span> <span class="fu">factor</span>(andrew_data<span class="sc">$</span>PATCH)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="r-interlude-nested-anova-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="r-interlude-nested-anova-2">R INTERLUDE | Nested ANOVA</h2>
<ul>
<li>In this case, our response variable is ALGAE</li>
<li>Look at the distribution of ALGAE for the two levels of TREAT2 using boxplots based on the patch means, which are the replicates in this case.</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>andrew.agg <span class="ot">&lt;-</span> <span class="fu">with</span>(andrew_data, <span class="fu">aggregate</span>(<span class="fu">data.frame</span>(ALGAE), </span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>                  <span class="at">by =</span> <span class="fu">list</span>(<span class="at">TREAT2=</span>TREAT2, <span class="at">PATCH=</span>PATCH), mean)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>andrew.agg <span class="ot">&lt;-</span> <span class="fu">gsummary</span>(andrew_data, <span class="at">groups=</span>andrew_data<span class="sc">$</span>PATCH)</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(ALGAE <span class="sc">~</span> TREAT2, andrew.agg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Evaluate assumptions based on the boxplots</li>
<li>Is the design balanced (equal numbers of sub-replicates per PATCH)?</li>
</ul>
</section>
<section id="r-interlude-nested-anova-3" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="r-interlude-nested-anova-3">R INTERLUDE | Nested ANOVA</h2>
<ul>
<li>Run the nested ANOVA:</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>nested.aov <span class="ot">&lt;-</span> <span class="fu">aov</span>(ALGAE <span class="sc">~</span> TREAT2 <span class="sc">+</span> <span class="fu">Error</span>(PATCH), <span class="at">data=</span>andrew_data)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(nested.aov)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Do we detect an effect of TREAT2 (high vs low sea urchin density)?</li>
<li>Estimate variance components to assess relative contributions of the random factors</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(nlme)</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a><span class="fu">VarCorr</span>(<span class="fu">lme</span>(ALGAE <span class="sc">~</span> <span class="dv">1</span>, <span class="at">random =</span> <span class="sc">~</span><span class="dv">1</span> <span class="sc">|</span> TREAT2<span class="sc">/</span>PATCH, andrew_data))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Calculate the % of variation due to between-treatment differences vs.&nbsp;due to among patches within treatment differences.</li>
<li>What do these variance component estimates tell us???</li>
</ul>
</section>
</section>
<section id="design-principles-for-planning-a-good-experiment" class="level1">
<h1>Design principles for planning a good experiment</h1>
<section id="what-is-an-experimental-study" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="what-is-an-experimental-study">What is an experimental study?</h2>
<ul>
<li>In an <strong>experimental study</strong> the researcher assigns treatments to units</li>
<li>In an <strong>observational study</strong> nature does the assigning of treatments to units</li>
<li>The crucial advantage of experiments derives from <strong>the random assignment of treatments to units</strong></li>
<li>Randomization <strong>minimizes the influence of confounding variables</strong></li>
<li>Can infer cause and effect more easily</li>
</ul>
</section>
<section id="mount-everest-example" class="level2 vcenter flexbox">
<h2 class="vcenter flexbox anchored" data-anchor-id="mount-everest-example">Mount Everest example</h2>
<p>Survival of climbers of Mount Everest is higher for individuals taking supplemental oxygen than those who don’t.</p>
<p><strong>Why?</strong></p>
</section>
<section id="mount-everest-example-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="mount-everest-example-1">Mount Everest example</h2>
<ul>
<li>One possibility is that supplemental oxygen (explanatory variable) really does cause higher survival (response variable).</li>
<li>The other is that the two variables are associated because other variables affect both supplemental oxygen and survival.</li>
<li>Use of supplemental oxygen might be a benign indicator of a greater overall preparedness of the climbers that use it.</li>
<li>Variables (like preparedness) that distort the causal relationship between the measured variables of interest (oxygen use and survival) are called <strong>confounding variables</strong></li>
<li>They are correlated with the variable of interest, and therefore preventing a decision about cause and effect.</li>
<li>With random assignment, no confounding variables will be associated with treatment except by chance.</li>
</ul>
</section>
<section id="clinical-trials" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="clinical-trials">Clinical Trials</h2>
<ul>
<li>The gold standard of experimental designs is the <strong>clinical trial</strong></li>
<li>Experimental design in all areas of biology have been informed by procedures used in clinical trials</li>
<li>A clinical trial is an experimental study in which two or more treatments are assigned to human subjects</li>
<li>The design of clinical trials has been refined because the cost of making a mistake with human subjects is so high</li>
<li>Experiments on nonhuman subjects are simply called “laboratory experiments”or “field experiments”</li>
</ul>
</section>
<section id="example-of-a-clinical-trial" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="example-of-a-clinical-trial">Example of a clinical trial</h2>
<ul>
<li>Transmission of the HIV-1 virus via sex workers contributes to the rapid spread of AIDS in Africa</li>
<li>The spermicide nonoxynol-9 had shown <em>in vitro</em> activity against HIV-1, which motivated a clinical trial by van Damme et al.&nbsp;(2002).</li>
<li>They tested whether a vaginal gel containing the chemical would reduce the risk of acquiring the disease by female sex workers.</li>
<li>Data were gathered on a volunteer sample of 765 HIV-free sex-workers in six clinics in Asia and Africa.</li>
<li>Two gel treatments were assigned randomly to women at each clinic.</li>
<li>One gel contained nonoxynol-9 and the other a placebo.</li>
<li>Neither the subjects nor the researchers making observations at the clinics knew who received the treatment and who got the placebo.</li>
</ul>
</section>
<section id="example-of-a-clinical-trial-1" class="level2 flexbox vcenter">
<h2 class="flexbox vcenter anchored" data-anchor-id="example-of-a-clinical-trial-1">Example of a clinical trial</h2>
<p><br></p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_6a.005.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="design-components-of-a-clinical-trial" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="design-components-of-a-clinical-trial">Design components of a clinical trial</h2>
<p>The goal of experimental design is to eliminate bias and to reduce sampling error when estimating and testing effects of one variable on another.</p>
<ul>
<li>To reduce bias, the experiment included:
<ul>
<li><strong>Simultaneous control group</strong>: study included both the treatment of interest and a control group (the women receiving the placebo).</li>
<li><strong>Randomization</strong>: treatments were randomly assigned to women at each clinic.</li>
<li><strong>Blinding</strong>: neither the subjects nor the clinicians knew which women were assigned which treatment.</li>
</ul></li>
</ul>
</section>
<section id="design-components-of-a-clinical-trial-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="design-components-of-a-clinical-trial-1">Design components of a clinical trial</h2>
<ul>
<li>To reduce the effects of sampling error, the experiment included:
<ul>
<li><strong>Replication</strong>: study was carried out on multiple independent subjects.</li>
<li><strong>Balance</strong>: number of women was nearly equal in the two groups at every clinic.</li>
<li><strong>Blocking</strong>: subjects were grouped according to the clinic they attended, yielding multiple repetitions of the same experiment in different settings (“blocks”).</li>
</ul></li>
</ul>
</section>
<section id="simultaneous-control-group" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="simultaneous-control-group">1. Simultaneous control group</h2>
<ul>
<li>In clinical trials either a placebo or the currently accepted treatment should be provided.</li>
<li>In experiments requiring intrusive methods to administer treatment, such as
<ul>
<li>injections</li>
<li>surgery</li>
<li>restraint</li>
<li>confinement</li>
</ul></li>
<li>the control subjects should be perturbed in the same way as the other subjects, except for the treatment itself, as far as ethical considerations permit.</li>
</ul>
</section>
<section id="simultaneous-control-group-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="simultaneous-control-group-1">1. Simultaneous control group</h2>
<ul>
<li>The “sham operation”, in which surgery is carried out without the experimental treatment itself, is an example.</li>
<li>In field experiments, applying a treatment of interest may physically disturb the plots receiving it and the surrounding areas, perhaps by trampling the ground by the researchers.</li>
<li>Ideally, the same disturbance should be applied to the control plots.</li>
</ul>
</section>
<section id="randomization" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="randomization">2. Randomization</h2>
<ul>
<li>The researcher should randomize assignment of treatments to units or subjects</li>
<li>Chance rather than conscious or unconscious decision determines which units end up receiving the treatment and which the control</li>
<li>A completely randomized design is one in which treatments are assigned to all units by randomization</li>
</ul>
</section>
<section id="randomization-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="randomization-1">2. Randomization</h2>
<ul>
<li>Randomization breaks the association between possible confounding variables and the explanatory variable</li>
<li>Randomization doesn’t eliminate the variation contributed by confounding variables, only their correlation with treatment</li>
<li>Randomization ensures that variation from confounding variables is similar between the different treatment groups.</li>
</ul>
</section>
<section id="randomization-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="randomization-2">2. Randomization</h2>
<ul>
<li>Randomization should be carried out using a random process:
<ul>
<li>List all n subjects, one per row, in a computer spreadsheet.</li>
<li>Use the computer to give each individual a random number.</li>
<li>Assign treatment A to those subjects receiving the lowest numbers and treatment B to those with the highest numbers.</li>
</ul></li>
<li>Other ways of assigning treatments to subjects are almost always inferior because they do not eliminate the effects of confounding variables.</li>
<li>“Haphazard” assignment, in which the researcher chooses a treatment while trying to make it random, has repeatedly been shown to be non-random and prone to bias.</li>
</ul>
</section>
<section id="randomization-types" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="randomization-types">2. Randomization Types</h2>
<ul>
<li><p><strong>Completely randomized design</strong> - all subjects are placed to treatment or control with equal probability</p></li>
<li><p><strong>Randomized block</strong> - first broken into groups (e.g.&nbsp;age or gender) then assigned to treatment groups at random</p></li>
<li><p><strong>Matched pair design</strong> - subjects are paired by similarity before being randomly assigned to treatment groups</p></li>
</ul>
</section>
<section id="random-sampling-approaches" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="random-sampling-approaches">2. Random sampling approaches</h2>
<ul>
<li><strong>simple random sample</strong> - every sample has equal probability of being chosen</li>
<li><strong>stratified sample</strong> - divided into groups then a simple random sample are taken from each</li>
<li><strong>cluster sample</strong> - divided into similar group, usually naturally occured, a simple random samples of clusters is then taken and very member of the cluster is included in the sample</li>
<li><strong>multistage sampling</strong> - combines the above. First clusters are random sampled. Second random samples are take from each. Then process is repeated</li>
<li><strong>systematic sample</strong> - members of a sample are chosen in a pre-determined ways. e.g Choose every 20th person coming into a store</li>
</ul>
</section>
<section id="blinding" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blinding">3. Blinding</h2>
<ul>
<li><strong>Blinding</strong> is the process of concealing information from participants (sometimes including researchers) about which subjects receive which treatment.</li>
<li>Blinding prevents <strong>subjects</strong> and <strong>researchers</strong> from changing their behavior, consciously or unconsciously, as a result of knowing which treatment they were receiving or administering.</li>
<li>For example, studies showing that acupuncture has a significant effect on back pain are limited to those without blinding (Ernst and White 1998).</li>
</ul>
</section>
<section id="blinding-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blinding-1">3. Blinding</h2>
<ul>
<li>In a <strong>single-blind experiment</strong>, the subjects are unaware of the treatment that they have been assigned.</li>
<li>Treatments must be indistinguishable to subjects, which prevents them from responding differently according to knowledge of treatment.</li>
<li>Blinding can also be a concern in non-human studies where animals respond to stimuli</li>
</ul>
</section>
<section id="blinding-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blinding-2">3. Blinding</h2>
<ul>
<li>In a <strong>double-blind experiment</strong> the researchers administering the treatments and measuring the response are also unaware of which subjects are receiving which treatments
<ul>
<li>Researchers sometimes have pet hypotheses, and they might treat experimental subjects in different ways depending on their hopes for the outcome</li>
<li>Many response variables are difficult to measure and require some subjective interpretation, which makes the results prone to a bias</li>
<li>Researchers are naturally more interested in the treated subjects than the control subjects, and this increased attention can itself result in improved response</li>
</ul></li>
</ul>
</section>
<section id="blinding-3" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blinding-3">3. Blinding</h2>
<ul>
<li>Reviews of medical studies have revealed that studies carried out without double- blinding exaggerated treatment effects by 16% on average compared with studies carried out with double-blinding (Jüni et al.&nbsp;2001).</li>
<li>Experiments on non–human subjects are also prone to bias from lack of blinding.</li>
</ul>
</section>
<section id="blinding-4" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blinding-4">3. Blinding</h2>
<ul>
<li>Bebarta et al.(2003) reviewed 290 two-treatment experiments carried out on animals or on cell lines. The odds of detecting a positive effect of treatment were more than threefold higher in studies without blinding than in studies with blinding.</li>
<li>Blinding can be incorporated into experiments on nonhuman subjects using coded tags that identify the subject to a “blind” observer without revealing the treatment (and who measures units from different treatments in random order).</li>
</ul>
</section>
<section id="replication" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="replication">4. Replication</h2>
<ul>
<li>The goal of experiments is to estimate and test treatment effects against the background of variation between individuals (“noise”) caused by other variables</li>
<li>One way to reduce noise is to make the experimental conditions constant</li>
<li>In field experiments, however, highly constant experimental conditions might not be feasible nor desirable</li>
<li>By limiting the conditions of an experiment, we also limit the generality of the results</li>
<li>Another way to make treatment effects stand out is to include extreme treatments and to replicate the data.</li>
</ul>
</section>
<section id="replication-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="replication-1">4. Replication</h2>
<ul>
<li>Replication is the assignment of each treatment to multiple, independent experimental units.</li>
<li>Without replication, we would not know whether response differences were due to the treatments or just chance differences between the treatments caused by other factors.</li>
<li>Studies that use more units (i.e.&nbsp;that have larger sample sizes) will have smaller standard errors and a higher probability of getting the correct answer from a hypothesis test.</li>
<li>Larger samples mean more information, and more information means better estimates and more powerful tests.</li>
</ul>
</section>
<section id="replication-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="replication-2">4. Replication</h2>
<ul>
<li>Replication is not about the number of plants or animals used, but the number of independent units in the experiment. An “experimental unit” is the independent unit to which treatments are assigned.</li>
<li>The figure shows three experimental designs used to compare plant growth under two temperature treatments (indicated by the shading of the pots). The first two designs are un-replicated.</li>
</ul>
</section>
<section id="pseudoreplication" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="pseudoreplication">4. Pseudoreplication</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_6a.006.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="balance" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="balance">5. Balance</h2>
<ul>
<li>A study design is balanced if all treatments have the same sample size.</li>
<li>Conversely, a design is unbalanced if there are unequal sample sizes between treatments.</li>
<li>Balance is a second way to reduce the influence of sampling error on estimation and hypothesis testing.</li>
<li>To appreciate this, look again at the equation for the standard error of the difference between two treatment means.</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_6a.007.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="balance-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="balance-1">5. Balance</h2>
<ul>
<li>For a fixed total number of experimental units, n1 + n2, the standard error is smallest when n1 and n2 are equal.</li>
<li>Balance has other benefits. For example, ANOVA is more robust to departures from the assumption of equal variances when designs are balanced or nearly so.</li>
</ul>
</section>
<section id="blocking" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blocking">6. Blocking</h2>
<ul>
<li>Blocking is the grouping of experimental units that have similar properties. Within each block, treatments are randomly assigned to experimental units.</li>
<li>Blocking essentially repeats the same, completely randomized experiment multiple times, once for each block.</li>
<li>Differences between treatments are only evaluated within blocks, and in this way the component of variation arising from differences between blocks is discarded.</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_6a.008.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="blocking-paired-designs" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blocking-paired-designs">6. Blocking | Paired designs</h2>
<ul>
<li>For example, consider the design choices for a two-treatment experiment to investigate the effect of clear cutting on salamander density.</li>
<li>In the completely randomized (“two-sample”) design we take a random sample of forest plots from the population and then randomly assign each plot to either the clear-cut treatment or the no clear-cut treatment.</li>
<li>In the paired design we take a random sample of forest plots and clear-cut a randomly chosen half of each plot, leaving the other half untouched.</li>
</ul>
</section>
<section id="blocking-paired-designs-1" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blocking-paired-designs-1">6. Blocking | Paired designs</h2>
<ul>
<li>In the paired design, measurements on adjacent plot-halves are not independent. This is because they are likely to be similar in soil, water, sunlight, and other conditions that affect the number of salamanders.</li>
<li>As a result, we must analyze paired data differently than when every plot is independent of all the others, as in the case of the two-sample design.</li>
<li>Paired design is usually more powerful than completely randomized design because it controls for a lot of the extraneous variation between plots or sampling units that sometimes obscures the effects we are looking for.</li>
</ul>
</section>
<section id="blocking-paired-designs-2" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blocking-paired-designs-2">6. Blocking | Paired designs</h2>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/images_6a.009.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="blocking-randomized-complete-block-design" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="blocking-randomized-complete-block-design">6. Blocking | Randomized complete block design</h2>
<ul>
<li>RCB design is analogous to the paired design, but may have more than two treatments. Each treatment is applied once to every block.</li>
<li>As in the paired design, treatment effects in a randomized block design are measured by differences between treatments exclusively within blocks.</li>
<li>By accounting for some sources of sampling variation blocking can make differences between treatments stand out.</li>
<li>Blocking is worthwhile if units within blocks are relatively homogeneous, apart from treatment effects, and units belonging to different blocks vary because of environmental or other differences.</li>
</ul>
</section>
<section id="what-if-you-cant-do-experiments" class="level2 smaller">
<h2 class="smaller anchored" data-anchor-id="what-if-you-cant-do-experiments">What if you can’t do experiments?</h2>
<ul>
<li>Experimental studies are not always feasible, in which case we must fall back upon observational studies.</li>
<li>The best observational studies incorporate as many of the features of good experimental design as possible to minimize bias (e.g., blinding) and the impact of sampling error (e.g., replication, balance, blocking, and even extreme treatments) except for one: randomization.</li>
<li>Randomization is out of the question, because in an observational study the researcher does not assign treatments to subjects.</li>
<li>Two strategies are used to limit the effects of confounding variables on a difference between treatments in a controlled observational study: matching; and adjusting for known confounding variables (covariates).</li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
</div> <!-- /content -->




</body></html>