<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.5.57">

  <meta name="author" content="Bill Cresko">
  <title>BioEngineering Statistics Spring 2025 – Week 3a - Statistics for Bioengineering</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-titled.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-titled) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-titled.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-titled .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-titled .callout-title  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-titled.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-titled.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-title {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-title {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-title {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-titled .callout-body > .callout-content > :last-child {
    padding-bottom: 0.5rem;
    margin-bottom: 0;
  }

  .callout.callout-titled .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-titled) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-title {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-title {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-title {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-title {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-title {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Week 3a - Statistics for Bioengineering</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Bill Cresko 
</div>
</div>
</div>

</section>
<section id="what-type-of-plot-do-i-use-for-each-data-type" class="slide level2">
<h2>What type of plot do I use for each data type?</h2>

<img data-src="images/Chart_flow_chart.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Flow chart to determine what type of data visualization and which ggplot geom to use</p></section>
<section id="three-dimensional-data" class="slide level2">
<h2>Three dimensional data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a><span class="fu">set.seed</span>(<span class="dv">345</span>) <span class="co"># make the example reproducible</span></span>
<span id="cb1-2"><a></a>a <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">10</span>,<span class="dv">10</span>)</span>
<span id="cb1-3"><a></a>b <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">5</span>,<span class="dv">5</span>)</span>
<span id="cb1-4"><a></a>c <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">1</span>,<span class="dv">1</span>)</span>
<span id="cb1-5"><a></a>d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(a,b,c)</span>
<span id="cb1-6"><a></a></span>
<span id="cb1-7"><a></a><span class="fu">library</span>(MASS)</span>
<span id="cb1-8"><a></a>DENS <span class="ot">&lt;-</span> <span class="fu">kde2d</span>(d<span class="sc">$</span>a,d<span class="sc">$</span>b)</span>
<span id="cb1-9"><a></a><span class="fu">contour</span>(DENS)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-2-1.png" width="960" class="r-stretch"></section>
<section id="three-dimensional-data-1" class="slide level2">
<h2>Three dimensional data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a></a><span class="fu">set.seed</span>(<span class="dv">345</span>) <span class="co"># make the example reproducible</span></span>
<span id="cb2-2"><a></a>a <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">10</span>,<span class="dv">10</span>)</span>
<span id="cb2-3"><a></a>b <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">5</span>,<span class="dv">5</span>)</span>
<span id="cb2-4"><a></a>c <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">1</span>,<span class="dv">1</span>)</span>
<span id="cb2-5"><a></a>d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(a,b,c)</span>
<span id="cb2-6"><a></a></span>
<span id="cb2-7"><a></a><span class="fu">library</span>(MASS)</span>
<span id="cb2-8"><a></a>DENS <span class="ot">&lt;-</span> <span class="fu">kde2d</span>(d<span class="sc">$</span>a,d<span class="sc">$</span>b)</span>
<span id="cb2-9"><a></a><span class="fu">filled.contour</span>(DENS,<span class="at">plot.axes =</span> {</span>
<span id="cb2-10"><a></a>  <span class="fu">axis</span>(<span class="dv">1</span>)</span>
<span id="cb2-11"><a></a>  <span class="fu">axis</span>(<span class="dv">2</span>)</span>
<span id="cb2-12"><a></a><span class="fu">contour</span>(DENS,<span class="at">add =</span> <span class="cn">TRUE</span>)})</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-3-1.png" width="960" class="r-stretch"></section>
<section id="three-dimensional-data-2" class="slide level2">
<h2>Three dimensional data</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a></a><span class="fu">set.seed</span>(<span class="dv">345</span>) <span class="co"># make the example reproducible</span></span>
<span id="cb3-2"><a></a>a <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">10</span>,<span class="dv">10</span>)</span>
<span id="cb3-3"><a></a>b <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">5</span>,<span class="dv">5</span>)</span>
<span id="cb3-4"><a></a>c <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">100</span>,<span class="dv">1</span>,<span class="dv">1</span>)</span>
<span id="cb3-5"><a></a>d <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(a,b,c)</span>
<span id="cb3-6"><a></a></span>
<span id="cb3-7"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-8"><a></a><span class="fu">ggplot</span>(d,<span class="fu">aes</span>(<span class="at">x=</span>a,<span class="at">y=</span>b)) <span class="sc">+</span></span>
<span id="cb3-9"><a></a>  <span class="fu">geom_density2d</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-4-1.png" width="960" class="r-stretch"></section>
<section>
<section id="examples-of-the-good-bad-and-the-ugly-of-graphical-representation-of-data" class="title-slide slide level1 center">
<h1>Examples of the good, bad and the ugly of graphical representation of data</h1>

</section>
<section id="section" class="slide level2 flexbox vcenter">
<h2></h2>
<ul>
<li>Examples of bad graphs and how to improve them.</li>
<li>Courtesy of K.W. Broman<br>
</li>
<li>www.biostat.wisc.edu/~kbroman/topten_worstgraphs/</li>
</ul>
</section>
<section id="ticker-tape-parade" class="slide level2 smaller">
<h2>Ticker tape parade</h2>

<img data-src="images/images_4a.003.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Roeder K 1994 DNA fingerprinting: A review of the controversy with discussion . Statistical Science 9:222T278, Figure 4</p></section>
<section id="a-line-to-no-understanding" class="slide level2 smaller">
<h2>A line to no understanding</h2>

<img data-src="images/images_4a.004.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Epstein MP, Satten GA 2003 Inference on haplotype effects in caseTcontrol studies using unphased genotype data. American Journal of Human Genetics 73:1316T1329, Figure 1</p></section>
<section id="a-hot-cup-o-mixed-messages" class="slide level2 flexbox vcenter">
<h2>A hot cup o’ mixed messages</h2>

<img data-src="images/images_4a.006.jpeg" class="r-stretch"></section>
<section id="i-want-the-biggest-slice-of-the-tfbs-pie" class="slide level2 flexbox vcenter smaller">
<h2>I want the biggest slice of the TFBS pie</h2>

<img data-src="images/images_4a.005.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Cawley S, et al.&nbsp;2004 Unbiased mapping of transcription factor binding sites along human chromosomes 21 and 22 points to widespread regulation of noncoding RNAs. Cell 116:499T509, Figure 1</p></section>
<section id="a-bake-sale-of-pie-charts" class="slide level2 smaller">
<h2>A bake sale of pie charts</h2>

<img data-src="images/images_4a.007.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Bell ML, et al.&nbsp;2007 Spatial and temporal variation in PM2.5 chemical composition in the United States for health effects studies. Environmental Health Perspectives 115:989T995, Figure 3</p></section>
<section id="wack-a-mole" class="slide level2">
<h2>Wack a mole</h2>

<img data-src="images/images_4a.008.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Cotter DJ, et al.&nbsp;2004 Hematocrit was not validated as a surrogate endpoint for survival amoung epoetinTtreated hemodialysis patients. Journal of Clinical Epidemiology 57:1086T1095, Figure 2</p></section>
<section class="slide level2">

</section></section>
<section>
<section id="best-practices" class="title-slide slide level1 center">
<h1>Best Practices</h1>

</section>
<section id="principles-of-effective-display" class="slide level2 flexbox vcenter">
<h2>Principles of effective display</h2>
<blockquote>
<p>“Graphical excellence is that which gives to the viewer the greatest number of ideas in the shortest time with the least ink in the smallest space”</p>
<p>— Edward Tufte</p>
</blockquote>
</section>
<section id="the-best-statistical-graphic-ever-drawn-according-to-edward-tufte" class="slide level2 flexbox vcenter">
<h2>The best statistical graphic ever drawn according to Edward Tufte</h2>

<img data-src="images/images_4a.009.jpeg" class="r-stretch quarto-figure-center"><p class="caption">This map by Charles Joseph Minard portrays the losses suffered by Napoleon’s army in the Russian campaign of 1812</p></section>
<section id="principles-of-effective-display-1" class="slide level2">
<h2>Principles of effective display</h2>
<ul>
<li>Show the data<br>
</li>
<li>Encourage the eye to compare differences</li>
<li>Represent magnitudes honestly and accurately<br>
</li>
<li>Draw graphical elements clearly, minimizing clutter<br>
</li>
<li>Make displays easy to interpret</li>
</ul>
</section>
<section id="above-all-else-show-the-data-tufte-1983" class="slide level2">
<h2>“Above all else show the data” | Tufte 1983</h2>

<img data-src="images/images_4a.010.jpeg" class="r-stretch quarto-figure-center"><p class="caption">The relationship between the numbers of native tropical stingless bees and Africanized honey bees onflowering shrubs in French Guiana. The data have been erased in the left panel. Redrawn from Roubik 1978.</p></section>
<section id="maximize-the-data-to-ink-ratio-within-reason-tufte-1983" class="slide level2">
<h2>“Maximize the data to ink ratio, within reason” | Tufte 1983</h2>
<p>Draw graphical elements clearly, minimizing clutter</p>

<img data-src="images/images_4a.011.jpeg" class="r-stretch quarto-figure-center"><p class="caption">The percentage of adults over 18 with a “body mass index” greater than 25 in different years The Economist 2006 . Body mass index is a measure of weight relative to height.</p></section>
<section id="a-graphic-does-not-distort-if-the-visual-representation-of-the-data-is-consistent-with-the-numerical-representation-tufte-1983" class="slide level2">
<h2>“A graphic does not distort if the visual representation of the data is consistent with the numerical representation” – Tufte 1983</h2>
<p>Represent magnitudes honestly and accurately</p>

<img data-src="images/images_4a.012.jpeg" class="r-stretch quarto-figure-center"><p class="caption">Slow wave sleep in the brain hemispheres of mallard ducks sleeping with one eye open. From Rattenborg et al.&nbsp;1999 Nature</p></section>
<section id="how-fox-news-makes-a-figure-." class="slide level2">
<h2>How Fox News makes a figure ….</h2>

<img data-src="images/images_4a.013.jpeg" class="r-stretch"></section>
<section id="how-fox-news-makes-a-figure-.-1" class="slide level2">
<h2>How Fox News makes a figure ….</h2>

<img data-src="images/images_4a.014.jpeg" class="r-stretch"><p>“Graphical excellence begins with telling the truth about the data” – Tufte 1983</p>
</section>
<section id="r-interlude" class="slide level2">
<h2>R Interlude</h2>
<ul>
<li><p>read in the data set <code>Week1b_Stickle_RNAseq.tsv</code> and assign it to a data object</p></li>
<li><p>try out the command <code>View</code> for the whole data set</p></li>
<li><p>try out the command <code>summary</code> for one of the variables</p></li>
<li><p>load the package <code>tidyverse</code> and try the command <code>glimpse</code></p></li>
<li><p>try making some nice plots of the different types of data</p></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="probability-distributions-and-sampling" class="title-slide slide level1 center">
<h1>Probability, distributions and sampling</h1>

</section>
<section id="stochastic-processes-in-statistics" class="slide level2 smaller">
<h2>Stochastic Processes in Statistics</h2>
<ul>
<li>We often want to know truths about the world, but the best we can do is estimate them</li>
<li>Uncertainty in those estimates is a given.</li>
<li>The process of statistics is largely about quantifying and managing uncertainty.</li>
<li>Random variables are the product of stochastic processes</li>
<li>Expectations can be based on theoretical probability distributions</li>
<li>We are going to start with probability rules and then slowly experience different distributions</li>
</ul>
</section>
<section id="different-flavors-of-inferential-statistics" class="slide level2 smaller">
<h2>Different flavors of inferential statistics</h2>
<ul>
<li><strong>Frequentist Statistics</strong>
<ul>
<li>Classical or standard approaches</li>
<li>Null hypothesis testing<br>
<br>
</li>
</ul></li>
<li><strong>Hierarchical Probabilistic Modeling</strong>
<ul>
<li>Maximum Likelihood</li>
<li>Bayesian Analyses</li>
<li>Machine Learning</li>
</ul></li>
</ul>
</section>
<section id="what-is-probability" class="slide level2 smaller">
<h2>What is probability</h2>
<ul>
<li><strong>Frequency interpretation</strong></li>
</ul>
<p>“Probabilities are understood as mathematically convenient approximations to long run relative frequencies.”</p>
<ul>
<li><strong>Subjective (Bayesian) interpretation</strong></li>
</ul>
<p>“A probability statement expresses the opinion of some individual regarding how certain an event is to occur.”</p>
</section>
<section id="random-variables-probability" class="slide level2 smaller">
<h2>Random variables &amp; probability</h2>
<ul>
<li><p><strong>Probability</strong> is the expression of belief in some future outcome</p></li>
<li><p>A <strong>random variable</strong> can take on different values with different probabilities</p></li>
<li><p>The <strong>sample space</strong> of a random variable is the universe of all possible values</p></li>
</ul>
</section>
<section id="random-variables-probability-1" class="slide level2 smaller">
<h2>Random variables &amp; probability</h2>
<ul>
<li>The <strong>sample space</strong> can be represented by a
<ul>
<li><strong>probability mass distribution</strong> (discrete)</li>
<li><strong>probability density function (PDF)</strong> (continuous)</li>
<li>algebra and calculus are used for each respectively</li>
<li>probabilities of a sample space <strong>always sum to 1.0</strong></li>
</ul></li>
<li>How does it make sense that a sample space will always sum to 1?</li>
</ul>
</section>
<section id="bernoulli-distribution" class="slide level2 smaller">
<h2>Bernoulli distribution</h2>
<ul>
<li><p>Describes the expected outcome of a single event with probability <code>p</code></p></li>
<li><p>Example of flipping of a <strong>fair</strong> coin once</p></li>
</ul>
<p><span class="math display">\[Pr(X=\text{Head}) = \frac{1}{2} = 0.5 = p \]</span></p>
<p><span class="math display">\[Pr(X=\text{Tails}) = \frac{1}{2} = 0.5 = 1 - p = q \]</span></p>
</section>
<section id="bernoulli-distribution-1" class="slide level2 smaller">
<h2>Bernoulli distribution</h2>
<ul>
<li>If the coin isn’t fair then <span class="math inline">\(p \neq 0.5\)</span></li>
<li>However, the probabilities still sum to 1</li>
</ul>
<p><span class="math display">\[ p + (1-p) = 1 \]</span> <span class="math display">\[ p + q = 1 \]</span></p>
<ul>
<li>Same is true for other binary possibilities
<ul>
<li>success or failure</li>
<li>yes or no answers</li>
<li>choosing an allele from a population based upon allele frequencies (Hardy-Weinberg ring any bells??)</li>
</ul></li>
</ul>
</section>
<section id="probability-rules" class="slide level2 smaller">
<h2>Probability rules</h2>
<ul>
<li>Flip a coin twice</li>
<li>Represent the first flip as ‘X’ and the second flip as ‘Y’</li>
</ul>
<p><span class="math display">\[ Pr(\text{X=H and Y=H}) = p*p = p^2 \]</span> <span class="math display">\[ Pr(\text{X=H and Y=T}) = p*q = pq = p^2 \]</span> <span class="math display">\[ Pr(\text{X=T and Y=H}) = q*p = pq \]</span> <span class="math display">\[ Pr(\text{X=T and Y=T}) = q*q = q^2 \]</span></p>
</section>
<section id="probability-rules-1" class="slide level2 smaller">
<h2>Probability rules</h2>
<ul>
<li>Probability that the <code>H</code> and <code>T</code> can occur in any order</li>
</ul>
<p><span class="math display">\[ \text{Pr(X=H) or Pr(X=T)} = p+q=1\]</span></p>
<p><span class="math display">\[ \text{Pr(X=H and Y=T) or Pr(X=T and Y=H)} = \]</span> <span class="math display">\[ (p*q) + (p*q) = 2pq \]</span></p>
<ul>
<li>These are the <strong>‘and’</strong> and <strong>‘or’</strong> rules of probability
<ul>
<li>‘and’ means multiply the probabilities</li>
<li>‘or’ means sum the probabilities</li>
<li>most probability distributions can be built up from these simple rules</li>
</ul></li>
</ul>
</section>
<section id="lets-simulate-some-coin-flips" class="slide level2 smaller">
<h2>Let’s simulate some coin flips</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a></a><span class="co"># tossing a fair coin</span></span>
<span id="cb4-2"><a></a>coin <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"heads"</span>, <span class="st">"tails"</span>)</span>
<span id="cb4-3"><a></a></span>
<span id="cb4-4"><a></a><span class="fu">sample</span>(coin)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "tails" "heads"</code></pre>
</div>
</div>
</section>
<section id="lets-simulate-some-coin-flips-1" class="slide level2 smaller">
<h2>Let’s simulate some coin flips</h2>
<ul>
<li>What happens when we change the probabilities or the sample size? How confident are we that our coin is fair?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a>flips <span class="ot">&lt;-</span> <span class="fu">sample</span>(coin, <span class="at">prob =</span> <span class="fu">c</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>), <span class="at">size=</span><span class="dv">13</span>, <span class="at">replace=</span><span class="cn">TRUE</span>)</span>
<span id="cb6-2"><a></a><span class="fu">barplot</span>(<span class="fu">table</span>(flips))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-6-1.png" style="width:50.0%" class="r-stretch"></section>
<section id="expectation-and-moments-of-distributions" class="slide level2 smaller">
<h2>Expectation and Moments of Distributions</h2>
<ul>
<li>Distributions have <strong>moments</strong> that can be estimated</li>
<li>1st, 2nd, 3rd and 4th moments of a distribution?</li>
<li>The expectation or mean of a random variable X is:</li>
</ul>
<p><span class="math display">\[E[X] = \sum_{\text{all x}}^{}xP(X=x) = \mu\]</span></p>
<ul>
<li>Often we want to know how dispersed the random variable is around its mean.</li>
<li>One measure of dispersion is the variance</li>
<li>There are higher moments of distributions (e.g.&nbsp;skew and kurtosis)</li>
</ul>
<p><span class="math display">\[Var(X) = E[X^2] = \sigma^2\]</span></p>
</section>
<section id="joint-probability" class="slide level2 smaller">
<h2>Joint probability</h2>
<p><span class="math display">\[Pr(X,Y) = Pr(X) * Pr(Y)\]</span></p>
<ul>
<li>Note that this is true for two <strong>independent</strong> events</li>
<li>However, for two non-independent events we also have to take into account their <strong>covariance</strong></li>
<li>To do this we need <strong>conditional probabilities</strong></li>
</ul>
</section>
<section id="conditional-probability" class="slide level2 smaller">
<h2>Conditional probability</h2>
<ul>
<li>For two <strong>independent</strong> variables: Probability of Y, given X, or the probability of X, given Y.</li>
</ul>
<p><span class="math display">\[Pr(Y|X) = Pr(Y)\text{ and }Pr(X|Y) = Pr(X)\]</span></p>
<ul>
<li>For two <strong>non-independent</strong> variables</li>
</ul>
<p><span class="math display">\[Pr(Y|X) \neq Pr(Y)\text{ and }Pr(X|Y) \neq Pr(X)\]</span></p>
<ul>
<li>Variables that are non-independent have a shared variance, which is also known as the <strong>covariance</strong></li>
<li>Covariance standardized to a mean of zero and a unit standard deviation is <strong>correlation</strong></li>
</ul>
</section>
<section id="what-is-likelihood-vs.-probability" class="slide level2 smaller">
<h2>What is Likelihood vs.&nbsp;Probability?</h2>
<ul>
<li><p>The <strong>probability</strong> of an event is the proportion of times that the event would occur if we repeated a random trial over and over again under the same conditions.</p></li>
<li><p>The <strong>likelihood</strong> is a conditional probability of a parameter value given a set of data</p></li>
<li><p>The likelihood of a population parameter <strong>equaling a specific value, given the data</strong></p></li>
</ul>
<p><code>L[parameter|data] = Pr[data|parameter]</code></p>
<ul>
<li><p><strong>Likelihood function</strong> which can have a maximum</p></li>
<li><p>What is a <strong>Bayesian estimate</strong>? - the use of prior distribution to update a posterior distribution</p></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="data-wrangling-and-exploratory-data-analysis-eda" class="title-slide slide level1 center">
<h1>Data wrangling and exploratory data analysis (EDA)</h1>

</section>
<section id="data-cleaning-and-manipulation" class="slide level2">
<h2>Data cleaning and manipulation</h2>
<ul>
<li><p>Here, we differentiate “data cleaning” from “data manipulation”, which is perhaps an arbitrary distinction.</p></li>
<li><p>“Data cleaning” typically refers to altering variable class information, fixing mistakes that could have arisen in the data (e.g., an extra ‘.’ symbol in a numeric value), and things of this nature.</p></li>
<li><p>“Data manipulation”, in my mind, refers to altering the structure of the data in a way that changes the functional structure the data (e.g., an addition of a column, deletion of rows, long/wide formatting change).</p></li>
</ul>
</section>
<section id="tidyverse-family-of-packages" class="slide level2 smaller">
<h2>Tidyverse family of packages</h2>

<img data-src="images/week_3.011.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"><p>Let’s load the tidyverse meta-package and check the output.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="tidyverse-family-of-packages-1" class="slide level2 vcenter flexbox">
<h2>Tidyverse family of packages</h2>
<ul>
<li><p>Hadley Wickham and others have written R packages to modify data</p></li>
<li><p>These packages do many of the same things as base functions in R</p></li>
<li><p>However, they are specifically designed to do them faster and more easily</p></li>
<li><p>Wickham also wrote the package ggplot2 for elegant graphics creations</p></li>
</ul>
</section>
<section id="example-of-a-tibble" class="slide level2 smaller">
<h2>Example of a tibble</h2>

<img data-src="images/week_3.012.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="example-of-a-tibble-1" class="slide level2 smaller">
<h2>Example of a tibble</h2>

<img data-src="images/week_3.013.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="types-of-vectors-of-data" class="slide level2 vcenter">
<h2>Types of vectors of data</h2>
<p><code>int</code> stands for integers</p>
<p><code>db</code>l stands for doubles, or real numbers</p>
<p><code>chr</code> stands for character vectors, or strings</p>
<p><code>dttm</code> stands for date-times (a date + a time)</p>
<p><code>lgl</code> stands for logical, vectors that contain only TRUE or FALSE</p>
<p><code>fctr</code> stands for factors, which R uses to represent categorical variables with fixed possible values</p>
<p><code>date</code> stands for dates</p>
</section>
<section id="types-of-vectors-of-data-1" class="slide level2 smaller">
<h2>Types of vectors of data</h2>
<ul>
<li>Logical vectors can take only three possible values:
<ul>
<li><code>FALSE</code></li>
<li><code>TRUE</code></li>
<li><code>NA</code> which is ‘not available’.</li>
</ul></li>
<li>Integer and double vectors are known collectively as numeric vectors.
<ul>
<li>In <code>R</code> numbers are doubles by default.</li>
</ul></li>
<li>Integers have one special value: NA, while doubles have four:
<ul>
<li><code>NA</code></li>
<li><code>NaN</code> which is ‘not a number’</li>
<li><code>Inf</code></li>
<li><code>-Inf</code></li>
</ul></li>
</ul>
</section>
<section id="tidyverse-packages-cont." class="slide level2">
<h2>Tidyverse packages (cont.)</h2>
<p>The tidyverse actually comes with a lot more packages than those that are just loaded automatically.<sup>1</sup></p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> [1] "broom"         "conflicted"    "cli"           "dbplyr"       
 [5] "dplyr"         "dtplyr"        "forcats"       "ggplot2"      
 [9] "googledrive"   "googlesheets4" "haven"         "hms"          
[13] "httr"          "jsonlite"      "lubridate"     "magrittr"     
[17] "modelr"        "pillar"        "purrr"         "ragg"         
[21] "readr"         "readxl"        "reprex"        "rlang"        
[25] "rstudioapi"    "rvest"         "stringr"       "tibble"       
[29] "tidyr"         "xml2"          "tidyverse"    </code></pre>
</div>
</div>
<p>We’ll use several of these additional packages during the remainder of this course.</p>
<ul>
<li>E.g. The <strong>lubridate</strong> package for working with dates and the <strong>rvest</strong> package for webscraping.</li>
<li>However, bear in mind that these packages will have to be loaded separately.</li>
</ul>
</section>
<section id="key-dplyr-verbs" class="slide level2">
<h2>Key dplyr verbs</h2>
<p>There are five key dplyr verbs that you need to learn.</p>
<ol type="1">
<li><code>filter</code>: Filter (i.e.&nbsp;subset) observations (rows) based on their values.<br>
</li>
<li><code>select</code>: Select (i.e.&nbsp;subset) variables (columns) by their names:<br>
</li>
<li><code>arrange</code>: Arrange (i.e.&nbsp;reorder) rows based on their values.<br>
</li>
<li><code>mutate</code>: Create new columns.<br>
</li>
<li><code>summarise</code>: Collapse multiple rows into a single summary value.<sup>1</sup></li>
</ol>
</section>
<section id="filter-arrange-select" class="slide level2">
<h2><code>filter()</code>, <code>arrange()</code> &amp; <code>select()</code></h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a><span class="fu">filter</span>(flights, month <span class="sc">==</span> <span class="dv">11</span> <span class="sc">|</span> month <span class="sc">==</span> <span class="dv">12</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a></a><span class="fu">arrange</span>(flights, year, month, day)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a></a><span class="fu">select</span>(flights, year)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>

<img data-src="images/week_3.015.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:60.0%"></section>
<section id="conditionals" class="slide level2 smaller">
<h2>conditionals</h2>
<ul>
<li><p><code>==</code> equals exactly</p></li>
<li><p><code>&lt;, &lt;=</code> is smaller than, is smaller than or equal to</p></li>
<li><p><code>&gt;</code>, &gt;=` is bigger than, is bigger than or equal to</p></li>
<li><p><code>!=</code> not equal to</p></li>
<li><p><code>!</code> NOT operator, to specify things that should be omitted</p></li>
<li><p><code>&amp;</code> AND operator, allows you to chain two conditions which must both be met</p></li>
<li><p><code>\|</code> OR operator, to chains two conditions when at least one should be met</p></li>
<li><p><code>%in%</code> belongs to one of the following (usually followed by a vector of possible values)</p></li>
</ul>
<p>The <code>AND (</code>&amp;<code>) and the OR (</code>|`) operators are also super useful when you want to separate data based on multiple conditions.</p>
</section>
<section id="mutate-transmutate" class="slide level2">
<h2><code>mutate()</code> &amp; <code>transmutate()</code></h2>
<p>This function will add a new variable that is a function of other variable(s)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a></a><span class="fu">mutate</span>(flights_sml,</span>
<span id="cb12-2"><a></a>  <span class="at">gain =</span> arr_delay <span class="sc">-</span> dep_delay,</span>
<span id="cb12-3"><a></a>  <span class="at">hours =</span> air_time <span class="sc">/</span> <span class="dv">60</span>,</span>
<span id="cb12-4"><a></a>  <span class="at">gain_per_hour =</span> gain <span class="sc">/</span> hours</span>
<span id="cb12-5"><a></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="group_by-summarize" class="slide level2 smaller">
<h2><code>group_by( )</code> &amp; <code>summarize( )</code></h2>
<p>This first function allows you to aggregate data by values of categorical variables</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a></a>by_day <span class="ot">&lt;-</span> <span class="fu">group_by</span>(flights, year, month, day)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Once you have done this aggregation, you can then calculate values (in this case the mean) of other variables split by the new aggregated levels of the categorical variable</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a></a><span class="fu">summarise</span>(by_day, <span class="at">delay =</span> <span class="fu">mean</span>(dep_delay, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<ul>
<li>Note - you can get a lot of missing values!</li>
<li>That’s because aggregation functions obey the usual rule of missing values:
<ul>
<li>if there’s any missing value in the input, the output will be a missing value.</li>
<li>fortunately, all aggregation functions have an na.rm argument which removes the missing values prior to computation</li>
</ul></li>
</ul>
</section>
<section id="other-dplyr-goodies" class="slide level2 smaller">
<h2>Other dplyr goodies</h2>
<ul>
<li><p><code>group_by</code> and <code>ungroup</code>: For (un)grouping. - Particularly useful with the <code>summarise</code> and <code>mutate</code> commands, as we’ve already seen.<br>
</p></li>
<li><p><code>slice</code>: Subset rows by position rather than filtering by values. - E.g. <code>starwars %&gt;% slice(c(1, 5))</code><br>
</p></li>
<li><p><code>pull</code>: Extract a column from as a data frame as a vector or scalar. - E.g. <code>starwars %&gt;% filter(gender=="female") %&gt;% pull(height)</code><br>
</p></li>
<li><p><code>count</code> and <code>distinct</code>: Number and isolate unique observations. - E.g. <code>starwars %&gt;% count(species)</code>, or <code>starwars %&gt;% distinct(species)</code><br>
</p></li>
<li><p>You could also use a combination of <code>mutate</code>, <code>group_by</code>, and <code>n()</code>, e.g.&nbsp;<code>starwars %&gt;% group_by(species) %&gt;% mutate(num = n())</code>.</p></li>
</ul>
</section>
<section id="r-interlude-playing-with-tidyverse-functions" class="slide level2 smaller">
<h2>R INTERLUDE | Playing with <code>Tidyverse</code> functions</h2>
<ul>
<li>Step 1 - Read in the <code>Week1b_Stickle_RNAseq.tsv</code> dataset</li>
<li>Step 2 - Make the dataset into a tibble</li>
<li>Step 3 - Select all of the categorical variables and only 4 of the gene count variables and put them into a new data object</li>
<li>Step 4 - Mutate each of the 4 gene expression values by performing a square root transformation making a new variable for each of the original (keep all 8 in the dataset).</li>
<li>Step 5 - Summarize the mean and standard deviation for each of the gene count variables grouped by the ‘sex’ and ‘population’ and ‘treatment’ categorical variables</li>
<li>Step 6 - Create a histogram for one of the original gene expression variables, and one of the derived variables<br>
</li>
<li>Step 7 - Create a box plot for one of the original gene expression variables, and one of the derived variables, split by treatment</li>
<li>Step 8 - Write the final data table to a .csv file and one of the figures to a .pdf file</li>
</ul>
</section>
<section class="slide level2">

</section>
<section class="slide level2">

</section></section>
<section>
<section id="common-probability-distributions-in-genomics-many-of-these-are-thanks-to-sally-otto-at-ubc" class="title-slide slide level1 center">
<h1>Common probability distributions in genomics | Many of these are thanks to Sally Otto at UBC</h1>

</section>
<section id="discrete-probability-distributions" class="slide level2">
<h2>Discrete Probability Distributions</h2>
</section>
<section id="geometric-distribution" class="slide level2">
<h2><strong>Geometric Distribution</strong></h2>
<ul>
<li>If a single event has two possible outcomes the probability of having to observe <code>k</code> trials before the first “one” appears is given by the <strong>geometric distribution</strong></li>
<li>The probability that the first “one” would appear on the first trial is <code>p</code>, but the probability that the first “one” appears on the second trial is <code>(1-p)*p</code></li>
<li>By generalizing this procedure, the probability that there will be <code>k-1</code> failures before the first success is:</li>
</ul>
<p><span class="math display">\[P(X=k)=(1-p)^{k-1}p\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{1}{p}\)</span></li>
<li>variance = <span class="math inline">\(\frac{(1-p)}{p^2}\)</span></li>
</ul>
</section>
<section id="geometric-distribution-1" class="slide level2">
<h2><strong>Geometric Distribution</strong></h2>
<ul>
<li>Example: If the probability of extinction of an endangered population is estimated to be 0.1 every year, what is the expected time until extinction?</li>
</ul>

<img data-src="images/prob.017.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="geometric-distribution-2" class="slide level2 smaller">
<h2><strong>Geometric Distribution</strong></h2>
<ul>
<li>If a single event has two possible outcomes the probability of having to observe <code>k</code> trials before the first “one” appears is given by the <strong>geometric distribution</strong></li>
<li>The probability that the first “one” would appear on the first trial is <code>p</code>, but the probability that the first “one” appears on the second trial is <code>(1-p)*p</code></li>
<li>By generalizing this procedure, the probability that there will be <code>k-1</code> failures before the first success is:</li>
</ul>
<p><span class="math display">\[P(X=k)=(1-p)^{k-1}p\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{1}{p}\)</span></li>
<li>variance = <span class="math inline">\(\frac{(1-p)}{p^2}\)</span></li>
</ul>
</section>
<section id="geometric-distribution-3" class="slide level2 smaller">
<h2><strong>Geometric Distribution</strong></h2>
<ul>
<li>Example: If the probability of extinction of an endangered population is estimated to be 0.1 every year, what is the expected time until extinction?</li>
</ul>

<img data-src="images/prob.017.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="geometric-distribution-4" class="slide level2 smaller">
<h2><strong>Geometric Distribution</strong></h2>
<ul>
<li>The distribution gives the probability of extinction in a given year (requiring that the population did not go extinct in all of the years prior)</li>
<li>If we want to know the probability of the population going exticnt by year 4, we simply add up the probabilities for years 1-3 using “or” rules</li>
</ul>

<img data-src="images/prob.017.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="testing-geometric-distributions" class="slide level2 smaller">
<h2>Testing Geometric Distributions</h2>
<ul>
<li>dgeom gives the density (probability) of an event (p) after (x) failures</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a></a><span class="co">#dgeom(x=20, p=0.1)</span></span>
<span id="cb15-2"><a></a><span class="co"># 0.01215767</span></span>
<span id="cb15-3"><a></a><span class="fu">plot</span>(<span class="fu">dgeom</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,<span class="fl">0.1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-20-1.png" class="quarto-figure quarto-figure-center r-stretch" style="width:40.0%"></section>
<section id="testing-geometric-distributions-1" class="slide level2 smaller">
<h2>Testing Geometric Distributions</h2>
<ul>
<li>pgeom gives the cumulative probability of event (p) in (q) trials</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.890581</code></pre>
</div>
</div>
</section>
<section id="binomial-distribution" class="slide level2">
<h2><strong>Binomial Distribution</strong></h2>
<ul>
<li><p>A <strong>binomial distribution</strong> results from the <strong>combination</strong> of several independent Bernoulli events</p></li>
<li><p><strong>Example</strong></p>
<ul>
<li>Pretend that you flip 20 fair coins
<ul>
<li>or collect alleles from a heterozygote</li>
</ul></li>
<li>Now repeat that process and record the number of heads</li>
<li>We expect that most of the time we will get approximately 10 heads</li>
<li>Sometimes we get many fewer heads or many more heads</li>
</ul></li>
</ul>
</section>
<section id="binomial-distribution-1" class="slide level2">
<h2>Binomial Distribution</h2>
<ul>
<li>The distribution of probabilities for each combination of outcomes is</li>
</ul>
<p><span class="math display">\[\large f(k) = {n \choose k} p^{k} (1-p)^{n-k}\]</span></p>
<ul>
<li><code>n</code> is the total number of trials</li>
<li><code>k</code> is the number of successes</li>
<li><code>p</code> is the probability of success</li>
<li><code>q</code> is the probability of not success</li>
<li>For binomial as with the Bernoulli <code>p = 1-q</code></li>
</ul>
</section>
<section id="binomial-probability-distribution" class="slide level2">
<h2>Binomial Probability Distribution</h2>

<img data-src="images/week_2.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="binomial-distribution-2" class="slide level2 smaller">
<h2><strong>Binomial Distribution</strong></h2>
<ul>
<li><p>A <strong>binomial distribution</strong> results from the <strong>combination</strong> of several independent Bernoulli events</p></li>
<li><p><strong>Example</strong></p>
<ul>
<li>Pretend that you flip 20 fair coins
<ul>
<li>or collect alleles from a heterozygote</li>
</ul></li>
<li>Now repeat that process and record the number of heads</li>
<li>We expect that most of the time we will get approximately 10 heads</li>
<li>Sometimes we get many fewer heads or many more heads</li>
</ul></li>
</ul>
</section>
<section id="binomial-distribution-3" class="slide level2 smaller">
<h2>Binomial Distribution</h2>
<ul>
<li>The distribution of probabilities for each combination of outcomes is</li>
</ul>
<p><span class="math display">\[\large f(k) = {n \choose k} p^{k} (1-p)^{n-k}\]</span></p>
<ul>
<li><code>n</code> is the total number of trials</li>
<li><code>k</code> is the number of successes</li>
<li><code>p</code> is the probability of success</li>
<li><code>q</code> is the probability of not success</li>
<li>For binomial as with the Bernoulli <code>p = 1-q</code></li>
</ul>
</section>
<section id="binomial-probability-distribution-1" class="slide level2 smaller">
<h2>Binomial Probability Distribution</h2>

<img data-src="images/week_2.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="testing-binomial-distributions" class="slide level2 smaller">
<h2>Testing Binomial Distributions</h2>
<ul>
<li>dbinom gives the density (probability) of number successes (x) in number trials (size), with (prob) probability between 0-1</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a></a><span class="co"># dbinom(x=5, size=10, p=0.5)</span></span>
<span id="cb17-2"><a></a><span class="co"># 0.246</span></span>
<span id="cb17-3"><a></a><span class="fu">plot</span>(<span class="fu">dbinom</span>(<span class="at">x=</span><span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>, <span class="at">size=</span><span class="dv">10</span>, <span class="at">p=</span><span class="fl">0.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-24-1.png" class="quarto-figure quarto-figure-center r-stretch" style="width:40.0%"></section>
<section id="testing-binomial-distributions-1" class="slide level2 smaller">
<h2>Testing Binomial Distributions</h2>
<ul>
<li>pbinom gives the cumulative probability of reaching at least (q) number of successes after (size) number of trials</li>
</ul>

<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-25-1.png" width="960" class="r-stretch"></section>
<section id="negative-binomial-distribution" class="slide level2">
<h2><strong>Negative Binomial Distribution</strong></h2>
<ul>
<li>Extension of the geometric distribution describing the waiting time until <code>r</code> “ones” have appeared.</li>
<li>Generalizes the geometric distribution</li>
<li>Probability of the <span class="math inline">\(r^{th}\)</span> “one” appearing on the <span class="math inline">\(k^{th}\)</span> trial:</li>
</ul>
<p><span class="math display">\[P(X=k)=(\frac{k-1}{r-1})p^{r-1}(1-p)^{k-r}p\]</span></p>
<p><br></p>
<p>which simplifies to</p>
<p><span class="math display">\[P(X=k)=(\frac{k-1}{r-1})p^{r}(1-p)^{k-r}\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{r}{p}\)</span></li>
<li>variance = <span class="math inline">\(r(1-p)/p^2\)</span></li>
</ul>
</section>
<section id="negative-binomial-distribution-1" class="slide level2">
<h2><strong>Negative Binomial Distribution</strong></h2>
<ul>
<li>Example: If a predator must capture 10 prey before it can grow large enough to reproduce</li>
<li>What would the mean age of onset of reproduction be if the probability of capturing a prey on any given day is 0.1?</li>
<li>Notice that the variance is quite high (~1000) and that the distribution looks quite skewed</li>
</ul>

<img data-src="images/prob.018.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="negative-binomial-distribution-2" class="slide level2 smaller">
<h2><strong>Negative Binomial Distribution</strong></h2>
<ul>
<li>Extension of the geometric distribution describing the waiting time until <code>r</code> “successes” have happened.</li>
<li>Generalizes the geometric distribution</li>
<li>Probability of the <span class="math inline">\(r^{th}\)</span> “success” appearing on the <span class="math inline">\(k^{th}\)</span> trial</li>
<li>mean = <span class="math inline">\(\frac{r}{p}\)</span></li>
<li>variance = <span class="math inline">\(r(1-p)/p^2\)</span></li>
</ul>
</section>
<section id="negative-binomial-distribution-3" class="slide level2 smaller">
<h2><strong>Negative Binomial Distribution</strong></h2>
<ul>
<li>Example: If a predator must capture 10 prey before it can grow large enough to reproduce</li>
<li>What would the mean age of onset of reproduction be if the probability of capturing a prey on any given day is 0.1?</li>
<li>Notice that the variance is quite high (~1000) and that the distribution looks quite skewed</li>
</ul>

<img data-src="images/prob.018.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="poisson-probability-distribution" class="slide level2">
<h2><strong>Poisson Probability Distribution</strong></h2>
<ul>
<li><p>Another common situation in biology is when each trial is discrete, but the number of observations of each outcome is observed/counted</p></li>
<li><p>Some examples are</p>
<ul>
<li>counts of snails in several plots of land</li>
<li>observations of the firing of a neuron in a unit of time</li>
<li>count of genes in a genome binned to units of 500 AA</li>
</ul></li>
<li><p>Just like before you have ‘successes’, but</p>
<ul>
<li>now you count them for each replicate</li>
<li>the replicates now are units of area or time</li>
<li>the values can now range from 0 to a large number</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-1" class="slide level2">
<h2>Poisson Probability Distribution</h2>
<p><br></p>
<ul>
<li>For example, you can examine 1000 genes
<ul>
<li>count the number of base pairs in the coding region of each gene</li>
<li>what is the probability of observing a gene with ‘r’ bp?</li>
</ul></li>
<li><code>Pr(Y=r)</code> is the probability that the number of occurrences of an event <code>y</code> equals a count <code>r</code> in the total number of trials</li>
</ul>
<p><br></p>
<p><span class="math display">\[Pr(Y=r) = \frac{e^{-\mu}\mu^r}{r!}\]</span></p>
</section>
<section id="poisson-probability-distribution-2" class="slide level2 smaller">
<h2>Poisson Probability Distribution</h2>
<ul>
<li>Note that this is a single parameter function because <span class="math inline">\(\mu = \sigma^2\)</span></li>
<li>The two together are often just represented by <span class="math inline">\(\lambda\)</span></li>
</ul>
<p><span class="math display">\[Pr(y=r) = \frac{e^{-\lambda}\lambda^r}{r!}\]</span></p>
<ul>
<li>This means that for a variable that is truly Poisson distributed:
<ul>
<li>the mean and variance should be equal to one another</li>
<li>variables that are approximately Poisson distributed but have a larger variance than mean are often called ‘overdispersed’</li>
<li>quite common in RNA-seq and microbiome data</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-gene-length-by-bins-of-500-nucleotides" class="slide level2 smaller">
<h2>Poisson Probability Distribution | gene length by bins of 500 nucleotides</h2>

<img data-src="images/week_2.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="poisson-probability-distribution-increasing-parameter-values-of-lambda" class="slide level2 smaller">
<h2>Poisson Probability Distribution | increasing parameter values of <span class="math inline">\(\lambda\)</span></h2>

<img data-src="images/week_2.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="poisson-probability-distribution-3" class="slide level2 smaller">
<h2><strong>Poisson Probability Distribution</strong></h2>
<ul>
<li><p>Another common situation in biology is when each trial is discrete, but the number of observations of each outcome is observed/counted</p></li>
<li><p>Some examples are</p>
<ul>
<li>counts of snails in several plots of land</li>
<li>observations of the firing of a neuron in a unit of time</li>
<li>count of genes in a genome binned to units of 500 AA</li>
</ul></li>
<li><p>Just like before you have ‘successes’, but</p>
<ul>
<li>now you count them for each replicate</li>
<li>the replicates now are units of area or time</li>
<li>the values can now range from 0 to a large number</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-4" class="slide level2 smaller">
<h2>Poisson Probability Distribution</h2>
<ul>
<li>For example, you can examine 1000 genes
<ul>
<li>count the number of base pairs in the coding region of each gene</li>
<li>what is the probability of observing a gene with ‘r’ bp?</li>
</ul></li>
<li><code>Pr(Y=r)</code> is the probability that the number of occurrences of an event <code>y</code> equals a count <code>r</code> in the total number of trials</li>
</ul>
</section>
<section id="poisson-probability-distribution-5" class="slide level2 smaller">
<h2>Poisson Probability Distribution</h2>
<ul>
<li>Note that this is a single parameter function because <span class="math inline">\(\mu = \sigma^2\)</span></li>
<li>The two together are often just represented by <span class="math inline">\(\lambda\)</span></li>
</ul>
<p><span class="math display">\[Pr(y=r) = \frac{e^{-\lambda}\lambda^r}{r!}\]</span></p>
<ul>
<li>This means that for a variable that is truly Poisson distributed:
<ul>
<li>the mean and variance should be equal to one another</li>
<li>variables that are approximately Poisson distributed but have a larger variance than mean are often called ‘overdispersed’</li>
<li>quite common in RNA-seq and microbiome data</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-gene-length-by-bins-of-500-nucleotides-1" class="slide level2 smaller">
<h2>Poisson Probability Distribution | gene length by bins of 500 nucleotides</h2>

<img data-src="images/week_2.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="poisson-probability-distribution-increasing-parameter-values-of-lambda-1" class="slide level2 smaller">
<h2>Poisson Probability Distribution | increasing parameter values of <span class="math inline">\(\lambda\)</span></h2>

<img data-src="images/week_2.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="testing-poisson-distributions" class="slide level2 smaller">
<h2>Testing Poisson Distributions</h2>
<p>Number of counts (x) given a mean and variance of lambda</p>
<div class="cell" data-layout-align="center">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1839397</code></pre>
</div>

</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-32-1.png" class="quarto-figure quarto-figure-center r-stretch" style="width:40.0%"></section>
<section id="poisson-probability-distribution-6" class="slide level2 smaller">
<h2>Poisson Probability Distribution</h2>
<ul>
<li><p>Another common situation in biology is when each trial is discrete but number of observations of each outcome is observed</p></li>
<li><p>Some examples are</p>
<ul>
<li>counts of snails in several plots of land</li>
<li>observations of the firing of a neuron in a unit of time</li>
<li>count of genes in a genome binned to units of 500 amino acids in length</li>
</ul></li>
<li><p>Just like before you have ‘successes’, but</p>
<ul>
<li>now you count them for each replicate</li>
<li>the replicates now are units of area or time</li>
<li>the values can now range from 0 to an arbitrarily large number</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-7" class="slide level2 smaller">
<h2>Poisson Probability Distribution</h2>
<ul>
<li>For example, you can examine 100 plots of land
<ul>
<li>count the number of snails in each plot</li>
<li>what is the probability of observing a plot with ‘r’ snails is</li>
</ul></li>
<li><code>Pr(Y=r)</code> is the probability that the number of occurrences of an event <code>y</code> equals a count <code>r</code> in the total number of trials</li>
</ul>
<p><span class="math display">\[Pr(Y=r) = \frac{e^{-\mu}\mu^r}{r!}\]</span></p>
<ul>
<li>Note that this is a single parameter function because <span class="math inline">\(\mu = \sigma^2\)</span>, and the two together are often just represented by <span class="math inline">\(\lambda\)</span></li>
</ul>
<p><span class="math display">\[Pr(y=r) = \frac{e^{-\lambda}\lambda^r}{r!}\]</span></p>
<ul>
<li>This means that for a variable that is truly Poisson distributed:
<ul>
<li>the mean and variance should be equal to one another, a hypothesis that you can test</li>
<li>variables that are approximately Poisson distributed but have a larger variance than mean are often called ‘overdispersed’</li>
</ul></li>
</ul>
</section>
<section id="poisson-probability-distribution-gene-length-by-bins-of-500-nucleotides-2" class="slide level2 smaller">
<h2>Poisson Probability Distribution | gene length by bins of 500 nucleotides</h2>

<img data-src="images/week_2.004.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="poisson-probability-distribution-increasing-parameter-values-of-lambda-2" class="slide level2 smaller">
<h2>Poisson Probability Distribution | increasing parameter values of <span class="math inline">\(\lambda\)</span></h2>

<img data-src="images/week_2.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="continuous-probability-distributions" class="slide level2">
<h2><strong>Continuous probability distributions</strong></h2>
<p><br></p>
<p>P(observation lies within dx of x) = f(x)dx</p>
<p><span class="math display">\[P(a\leq X \leq b) = \int_{a}^{b} f(x) dx\]</span></p>
<p><br></p>
<p>Remember that the indefinite integral sums to one</p>
<p><span class="math display">\[\int_{-\infty}^{\infty} f(x) dx = 1\]</span></p>
</section>
<section id="continuous-probabilities" class="slide level2">
<h2>Continuous probabilities</h2>
<p><br></p>
<p><code>E[X]</code> may be found by integrating the product of <code>x</code> and the probability density function over all possible values of <code>x</code>:</p>
<p><span class="math display">\[E[X] = \int_{-\infty}^{\infty} xf(x) dx \]</span></p>
<p><br></p>
<p><span class="math inline">\(Var(X) = E[X^2] - (E[X])^2\)</span>, where the expectation of <span class="math inline">\(X^2\)</span> is</p>
<p><span class="math display">\[E[X^2] = \int_{-\infty}^{\infty} x^2f(x) dx \]</span></p>
</section>
<section id="uniform-distribution" class="slide level2">
<h2><strong>Uniform Distribution</strong></h2>
<p><br></p>
<p><span class="math display">\[E[X] = \int_{a}^{b} x\frac{1}{b-a} dx = \frac{(a+b)}{2} \]</span></p>
<p><br></p>

<img data-src="images/prob.019.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="uniform-distribution-1" class="slide level2 smaller">
<h2><strong>Uniform Distribution</strong></h2>
<ul>
<li>Means that the probability is equal for all possible outcomes</li>
<li>Like drawing m&amp;m out of a bag with equal proportions of colors</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1</code></pre>
</div>
</div>
</section>
<section id="uniform-distribution-2" class="slide level2 smaller">
<h2>Uniform Distribution</h2>
<div class="cell">
<div class="cell-output-display">
<div>
<figure>
<p><img data-src="Week3a_files/figure-revealjs/unnamed-chunk-37-1.png" width="960"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="images/prob.019.jpeg" class="quarto-figure quarto-figure-center" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="exponential-distribution" class="slide level2">
<h2>Exponential Distribution</h2>
<p><br></p>
<p><span class="math display">\[f(x)=\lambda e^{-\lambda x}\]</span></p>
<p><br></p>
<ul>
<li><code>E[X]</code> can be found be integrating <span class="math inline">\(xf(x)\)</span> from 0 to infinity</li>
</ul>
<p><br></p>
<ul>
<li>leading to the result that</li>
</ul>
<p><br></p>
<ul>
<li><span class="math inline">\(E[X] = \frac{1}{\lambda}\)</span></li>
<li><span class="math inline">\(E[X^2] = \frac{1}{\lambda^2}\)</span></li>
</ul>
</section>
<section id="exponential-distribution-1" class="slide level2">
<h2>Exponential Distribution</h2>
<ul>
<li>For example, let equal the instantaneous death rate of an individual.</li>
<li>The lifespan of the individual would be described by an exponential distribution (assuming that does not change over time).</li>
</ul>

<img data-src="images/prob.020.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="exponential-distribution-2" class="slide level2 smaller">
<h2>Exponential Distribution</h2>
<ul>
<li>For example, let equal the instantaneous death rate of an individual.</li>
<li>The lifespan of the individual would be described by an exponential distribution (assuming that does not change over time).</li>
</ul>

<img data-src="images/prob.020.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="exponential-distribution-3" class="slide level2 smaller">
<h2>Exponential Distribution</h2>

<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-41-1.png" width="960" class="r-stretch"></section>
<section id="gamma-distribution" class="slide level2">
<h2><strong>Gamma Distribution</strong></h2>
<ul>
<li>The gamma distribution generalizes the exponential distribution.</li>
<li>It describes the waiting time until the <span class="math inline">\(r^{th}\)</span> event for a process that occurs randomly over time at a rate <span class="math inline">\(\lambda\)</span> :</li>
</ul>
<p><br></p>
<p><span class="math display">\[f(x) = \frac{e^{-\lambda x}\lambda x^{r-1}}{(r-1)!}\lambda\]</span></p>
<p><br></p>
<p><span class="math display">\[ Mean =  \frac{r}{\lambda} \]</span> <span class="math display">\[ Variance = \frac{r}{\lambda^2} \]</span></p>
</section>
<section id="gamma-distribution-1" class="slide level2">
<h2>Gamma Distribution</h2>
<p><br></p>
<ul>
<li><strong>Example</strong>: If, in a PCR reaction, DNA polymerase synthesizes new DNA strands at a rate of 1 per millisecond, how long until 1000 new DNA strands are produced?</li>
</ul>
<p><br></p>
<ul>
<li>Assume that DNA synthesis does not deplete the pool of primers or nucleotides in the chamber, so that each event is independent of other events in the PCR chamber.</li>
</ul>
</section>
<section id="gamma-distribution-2" class="slide level2 smaller">
<h2><strong>Gamma Distribution</strong></h2>
<ul>
<li>The gamma distribution generalizes the exponential distribution.</li>
<li>It describes the waiting time until the <span class="math inline">\(r^{th}\)</span> event for a process that occurs randomly over time at a rate <span class="math inline">\(\lambda\)</span></li>
</ul>
</section>
<section id="gamma-distribution-3" class="slide level2 smaller">
<h2>Gamma Distribution</h2>
<ul>
<li><p><strong>Example</strong>: If, in a PCR reaction, DNA polymerase synthesizes new DNA strands at a rate of 1 per millisecond, how long until 1000 new DNA strands are produced?</p></li>
<li><p>Assume that DNA synthesis does not deplete the pool of primers or nucleotides in the chamber, so that each event is independent of other events in the PCR chamber.</p></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="common-probability-distributions" class="title-slide slide level1 center">
<h1>Common probability distributions</h1>

</section>
<section id="r-interlude-binomially-distributed-data" class="slide level2 smaller">
<h2>R INTERLUDE | binomially distributed data</h2>
<ul>
<li>Pretend that you want to create your own binomial distribution
<ul>
<li>you could flip 20 fair coins: 20 independent Bernoulli “trials”</li>
<li>you could then replicate this 100 times: 100 “observations”</li>
<li>for each one of your replicates you record the number of heads</li>
<li>probability of heads (“success”) for any flip of a fair coin is 0.5</li>
</ul></li>
<li>Draw a diagram of what you think the distribution would look like
<ul>
<li>what will the x-axis represent, and what are its values?</li>
<li>what will the y-axis represent, and what are its values?</li>
<li>where will the ‘center of mass’ be?</li>
<li>what parameter determines the value of the center of mass?</li>
<li>what parameter determines the spread of the distribution?</li>
</ul></li>
</ul>
</section>
<section id="r-interlude-binomially-distributed-data-1" class="slide level2 smaller">
<h2>R INTERLUDE | binomially distributed data</h2>
<ul>
<li>Using R, simulate these experimental data
<ul>
<li>using the <span class="math inline">\(rbinom()\)</span> function</li>
<li>look at the arguments to see how they map on to the trials, probability of each trial, and number of replicates</li>
<li>use the <span class="math inline">\(hist()\)</span> function to plot the simulated data.</li>
<li>what do the y- and x-axes represent?</li>
<li>what is the most common outcome, in terms of the number of “successes” per trial? Does this make sense?</li>
</ul></li>
<li>Now just change you script to do the following
<ul>
<li>perform 200 trials for each of the 100 replicates</li>
<li>perform 2000 trials for each of the 100 replicate</li>
<li>how do the distributions change when you go from 20 to 200 to 2000 trials?</li>
</ul></li>
</ul>
</section>
<section id="r-interlude-binomially-distributed-data-2" class="slide level2 smaller">
<h2>R INTERLUDE | binomially distributed data</h2>
<ul>
<li>Note that the binomial function incorporates both the ‘and’ and ‘or’ rules of probability</li>
<li>This part is the probability of each outcome (multiplication)</li>
</ul>
<p><span class="math display">\[\large p^{k} (1-p)^{n-k}\]</span></p>
<p>This part (called the binomial coefficient) is the number of different ways each combination of outcomes can be achieved (summation)</p>
<p><span class="math display">\[\large {n \choose k}\]</span> Together they equal the probability of a specified number of successes</p>
<p><span class="math display">\[\large f(k) = {n \choose k} p^{k} (1-p)^{n-k}\]</span></p>
</section>
<section id="r-interlude-binomially-distributed-data-3" class="slide level2 smaller">
<h2>R INTERLUDE | binomially distributed data</h2>
<ul>
<li>Now perform your experiment using R</li>
<li>Use the <code>rbinom</code> function to replicate what you sketched out for coin flipping</li>
<li>Be sure to check out the other functions in the <code>binom</code> family</li>
<li>Make sure you know how you are mapping the parameeters to values</li>
<li>Take the output and create a histogram</li>
<li>Does it look similar to what you expected?</li>
<li>Now change the values around so that
<ul>
<li>You have more or fewer coin flips per trial</li>
<li>You replicate the process with more or fewer trials</li>
<li>You change the coin from being fair to being slightly biased</li>
<li>You change the coin from being slightly to heavily biased</li>
</ul></li>
</ul>
</section>
<section id="the-gaussian-or-normal-distribution" class="slide level2 smaller">
<h2>The Gaussian or Normal Distribution</h2>
</section>
<section id="log-normal-pdf-continuous-version-of-poisson--ish" class="slide level2 smaller">
<h2>Log-normal PDF | Continuous version of Poisson (-ish)</h2>

<img data-src="images/week_2.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="transformations-to-normalize-data" class="slide level2 smaller vcenter flexbox">
<h2>Transformations to ‘normalize’ data</h2>

<img data-src="images/week_2.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="transformations-to-normalize-data-1" class="slide level2 smaller vcenter flexbox">
<h2>Transformations to ‘normalize’ data</h2>

<img data-src="images/week_2.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="binomial-to-normal-categorical-to-continuous" class="slide level2 smaller vcenter flexbox">
<h2>Binomial to Normal | Categorical to continuous</h2>

<img data-src="images/week_2.009.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="the-normal-aka-gaussian-probability-density-function-pdf" class="slide level2 smaller">
<h2>The Normal (aka Gaussian) | Probability Density Function (PDF)</h2>

<img data-src="images/week_2.011.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="normal-pdf" class="slide level2 smaller">
<h2>Normal PDF</h2>

<img data-src="images/week_2.012.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="normal-pdf-a-function-of-two-parameters" class="slide level2">
<h2>Normal PDF | A function of two parameters</h2>
<h3 class="smaller" id="mu-and-sigma">(<span class="math inline">\(\mu\)</span> and <span class="math inline">\(\sigma\)</span>)</h3>

<img data-src="images/week_2.032.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:40.0%"><p>where <span class="math display">\[\large \pi \approx 3.14159\]</span></p>
<p><span class="math display">\[\large \epsilon \approx 2.71828\]</span></p>
<p>To write that a variable (v) is distributed as a normal distribution with mean <span class="math inline">\(\mu\)</span> and variance <span class="math inline">\(\sigma^2\)</span>, we write the following:</p>
<p><span class="math display">\[\large v \sim \mathcal{N} (\mu,\sigma^2)\]</span></p>
</section>
<section id="normal-pdf-estimates-of-mean-and-variance" class="slide level2 smaller">
<h2>Normal PDF | estimates of mean and variance</h2>
<p>Estimate of the mean from a single sample</p>
<p><span class="math display">\[\Large \bar{x} = \frac{1}{n}\sum_{i=1}^{n}{x_i} \]</span></p>
<p>Estimate of the variance from a single sample</p>
<p><span class="math display">\[\Large s^2 = \frac{1}{n-1}\sum_{i=1}^{n}{(x_i - \bar{x})^2} \]</span></p>
</section>
<section id="normal-pdf-1" class="slide level2 smaller">
<h2>Normal PDF</h2>

<img data-src="images/week_2.010.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="why-is-the-normal-special-in-biology" class="slide level2 smaller">
<h2>Why is the Normal special in biology?</h2>

<img data-src="images/week_2.013.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="why-is-the-normal-special-in-biology-1" class="slide level2 smaller">
<h2>Why is the Normal special in biology?</h2>

<img data-src="images/week_2.015.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="why-is-the-normal-special-in-biology-2" class="slide level2 smaller">
<h2>Why is the Normal special in biology?</h2>

<img data-src="images/week_2.014.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:60.0%"></section>
<section id="parent-offspring-resemblance" class="slide level2 smaller">
<h2>Parent-offspring resemblance</h2>

<img data-src="images/week_2.016.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:45.0%"></section>
<section id="genetic-model-of-complex-traits" class="slide level2 smaller">
<h2>Genetic model of complex traits</h2>

<img data-src="images/week_2.017.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="distribution-of-f_2-genotypes-really-just-binomial-sampling" class="slide level2 smaller">
<h2>Distribution of <span class="math inline">\(F_2\)</span> genotypes | really just binomial sampling</h2>

<img data-src="images/week_2.018.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:70.0%"></section>
<section id="why-else-is-the-normal-special" class="slide level2 smaller">
<h2>Why else is the Normal special?</h2>
<ul>
<li>The normal distribution is immensely useful because of the central limit theorem, which says that the mean of many random variables independently drawn from the same distribution is distributed approximately normally</li>
<li>One can think of numerous situations, such as
<ul>
<li>when multiple genes contribute to a phenotype</li>
<li>or that many factors contribute to a biological process</li>
</ul></li>
<li>In addition, whenever there is variance introduced by stochastic factors the central limit theorem holds</li>
<li>Thus, normal distributions occur throughout genomics</li>
<li>It’s also the basis of the majority of classical statistics</li>
</ul>
</section>
<section id="z-scores-of-normal-variables" class="slide level2 smaller">
<h2>z-scores of normal variables</h2>
<ul>
<li>Often we want to make variables more comparable to one another</li>
<li>For example, consider measuring the leg length of mice and of elephants
<ul>
<li>Which animal has longer legs in absolute terms?</li>
<li>Proportional to their body size?</li>
</ul></li>
<li>A good way to answer these last questions is to use ‘z-scores’</li>
</ul>
</section>
<section id="z-scores-of-normal-variables-1" class="slide level2 smaller">
<h2>z-scores of normal variables</h2>
<ul>
<li>z-scores are standardized to a mean of 0 and a standard deviation of 1</li>
<li>We can modify any normal distribution to have a mean of 0 and a standard deviation of 1</li>
<li>Another term for this is the standard normal distribution</li>
</ul>
<p><span class="math display">\[\huge z_i = \frac{(x_i - \bar{x})}{s}\]</span></p>
</section>
<section id="r-interlude-simulate-a-population-and-sample-it" class="slide level2 smaller">
<h2>R INTERLUDE | Simulate a population and sample it!</h2>
<p>Simulate a population of 10,000 individual values for a variable x:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a></a>x <span class="ot">&lt;-</span> <span class="fu">rnorm</span>(<span class="dv">10000</span>, <span class="at">mean=</span><span class="fl">50.5</span>, <span class="at">sd=</span><span class="fl">5.5</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Take 1000 random samples of size 20, take the mean of each sample, and plot the distribution of these 1000 sample means.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a></a>x_sample_means <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb21-2"><a></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">1000</span>){</span>
<span id="cb21-3"><a></a>x_samp <span class="ot">&lt;-</span> <span class="fu">sample</span>(x, <span class="dv">20</span>, <span class="at">replace=</span><span class="cn">FALSE</span>)</span>
<span id="cb21-4"><a></a>x_sample_means[i] <span class="ot">&lt;-</span> <span class="fu">mean</span>(x_samp)</span>
<span id="cb21-5"><a></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>For one of your samples, use the equation from the previous slide to transform the values to z-scores.</p>
<p>Plot the distribution of the z-scores, and calculate the mean and the standard deviation.</p>
</section>
<section id="r-interlude-simulate-a-population-and-sample-it-1" class="slide level2">
<h2>R INTERLUDE | Simulate a population and sample it!</h2>
<ul>
<li>Now, create a second population (called x.lognorm) by log-transforming all 10,000 values from population “x”</li>
<li>Plot the histogram of these data</li>
<li>Repeat the taking of 1000 samples of size 20</li>
<li>Take the mean of each sample</li>
<li>Plot the distribution of these 1000 sample means from the known lognormal population.</li>
<li>What does the distribution of the sampled means look like?</li>
</ul>
</section>
<section id="more-than-one-variable-bivariate-normal-correlation-and-covariation" class="slide level2 smaller">
<h2>More than one variable | Bivariate normal, correlation and covariation</h2>

<img data-src="images/week_2.019.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="more-than-one-variable-bivariate-normal-correlation-and-covariation-1" class="slide level2 smaller">
<h2>More than one variable | Bivariate normal, correlation and covariation</h2>

<img data-src="images/week_2.020.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="covariance-and-correlation" class="slide level2 vcenter flexbox">
<h2>Covariance and Correlation</h2>

<img data-src="images/week_2.021.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="linear-models" class="slide level2">
<h2>Linear Models</h2>

<img data-src="images/week_2.022.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="show-the-data-anscombes-quartet" class="slide level2 smaller">
<h2>Show the data! | Anscombe’s Quartet</h2>

<img data-src="images/week_2.023.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="show-the-data-anscombes-quartet-1" class="slide level2">
<h2>Show the data! | Anscombe’s Quartet</h2>
<ul>
<li><p>Mean of x in each case 9 (exact)</p></li>
<li><p>Variance of x in each case 11 (exact)</p></li>
<li><p>Mean of y in each case 7.50 (to 2 decimal places)</p></li>
<li><p>Variance of y in each case 4.122 or 4.127 (to 3 decimal places)</p></li>
<li><p>Correlation between x and y in each case 0.816 (to 3 decimal places)</p></li>
<li><p>Linear regression line in each case <span class="math inline">\(y =3.00 + 0.50x\)</span> (to 2 decimal places)</p></li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section id="sampling-and-estimation" class="title-slide slide level1 center">
<h1>Sampling and estimation</h1>

</section>

<section>
<section id="where-do-we-begin" class="title-slide slide level1 center">
<h1>Where do we begin?</h1>
<ul>
<li>A major goal of statistics is to estimate <strong>parameters</strong> of a population so that we can compare them to values that are of practical importance to our understanding of the system, or to compare parameter estimates between and among different populations</li>
<li>This is the first important step before getting to hypothesis testing!</li>
</ul>
</section>
<section id="understanding-populations-and-their-parameters" class="slide level2">
<h2>Understanding Populations and their Parameters</h2>
<ul>
<li>We often think about the samples we are collecting as a part of a larger population</li>
<li>Since we can’t measure every member of that population, we instead use sampling to estimate the parameters of the population as a whole
<ul>
<li>Some common parameters: mean, range, median</li>
<li>If we performed random sampling, we assume that the parameter estimates of our sample are equitable to the true population parameters</li>
</ul></li>
</ul>
</section>
<section id="accuracy-vs.-precision" class="slide level2">
<h2>Accuracy vs.&nbsp;Precision</h2>

<img data-src="images/week_2.024.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:50.0%"></section>
<section id="accuracy-vs.-precision-1" class="slide level2">
<h2>Accuracy vs.&nbsp;Precision</h2>
<ul>
<li><strong>Accuracy</strong> is the closeness of an estimated value to its true value</li>
<li><strong>Precision</strong> is the closeness of repeated estimates to one another</li>
<li>Our goal is to have <strong>unbiased estimates</strong> that are the most precise</li>
<li>We have to <strong>estimate parameters</strong> and <strong>test hypotheses</strong> by taking <strong>samples</strong> that approximate the underlying distribution</li>
<li>The goal of <strong>replication</strong> is to quantify variation at as many levels in a study as possible</li>
<li>The goal of <strong>randomization</strong> is to avoid bias as much as possible</li>
</ul>
</section>
<section id="parameter-estimation" class="slide level2">
<h2>Parameter Estimation</h2>
<ul>
<li><strong>Parametric</strong> (a few special exceptions, like the sample mean and its standard error)</li>
<li><strong>Ordinary Least Squares (OLS)</strong> - optimized procedure that produces one definitive result, easy to use but no estimates of confidence</li>
<li><strong>Resampling</strong> - bootstrapping and randomization</li>
<li><strong>Maximum Likelihood (ML)</strong> - Can provide model-based estimates with confidence, but harder to calculate</li>
<li><strong>Bayesian Approaches</strong> - Incorporates prior information into ML estimation</li>
</ul>
</section>
<section id="sampling-variation-of-a-parameter" class="slide level2 smaller">
<h2>Sampling variation of a parameter</h2>

<img data-src="images/week_2.025.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="estimation" class="slide level2 smaller">
<h2>Estimation</h2>
<ul>
<li>Estimation is the process of inferring a population parameter from sample data</li>
<li>The value of one sample estimate is almost never the same as the population parameter because of random sampling error</li>
<li>Imagine taking multiple samples, each will be different from the true value</li>
<li>Most will be close, but some will be far away</li>
<li>The <strong>expected value</strong> of a very large number of sample estimates is the value of the parameter being estimated</li>
<li>Sampling distribution of an estimate
<ul>
<li>all values we might have obtained from our sample</li>
<li>probabilities of occurrence</li>
</ul></li>
<li>Standard error of an estimate
<ul>
<li>standard deviation of a sampling distribution</li>
<li>measures the precision of the parameter estimate</li>
<li>NO ESTIMATE IS USEFUL WITHOUT IT!</li>
</ul></li>
</ul>
</section>
<section id="estimation-and-confidence-intervals" class="slide level2 vcenter flexbox">
<h2>Estimation and confidence intervals</h2>

<img data-src="images/week_2.025.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="estimation-and-confidence-intervals-1" class="slide level2 vcenter flexbox">
<h2>Estimation and confidence intervals</h2>

<img data-src="images/week_2.026.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="estimation-and-confidence-intervals-2" class="slide level2 vcenter flexbox">
<h2>Estimation and confidence intervals</h2>

<img data-src="images/week_2.027.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="estimation-and-confidence-intervals-3" class="slide level2 vcenter flexbox">
<h2>Estimation and confidence intervals</h2>

<img data-src="images/week_2.028.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="standard-error-of-the-mean-sem" class="slide level2">
<h2>Standard Error of the Mean (SEM)</h2>
<p><span class="math display">\[\huge \sigma_{\bar{x}} \approx s_{\bar{x}} = \frac{s}{\sqrt{n}} \]</span></p>
<ul>
<li>where <span class="math inline">\(s_{\bar{x}}\)</span> is the estimated standard error of the distribution of the mean estimates</li>
<li>which is usually just referred to as the ’standard error of the mean (SEM)</li>
<li>note that this <strong>is not</strong> the standard deviation of the original distribution</li>
<li>importantly, the SEM will go down as the sample size increases</li>
</ul>
</section>
<section id="standard-error" class="slide level2">
<h2>Standard Error</h2>
<ul>
<li><p>Sadly, most other kinds of estimates do not have this amazing property.</p></li>
<li><p>What to do?</p></li>
<li><p>One answer: make your own sampling distribution for the estimate using the “bootstrap”.</p></li>
<li><p>Method invented by Efron (1979).</p></li>
</ul>
</section>
<section id="parameter-estimation-bootstrap-algorithm" class="slide level2">
<h2>Parameter Estimation | Bootstrap Algorithm</h2>
<ul>
<li>Use <code>R</code> to take a random sample of individuals from the original data</li>
<li>Calculate the estimate using the measurements in the bootstrap sample (step 1)</li>
<li>This is the first bootstrap replicate estimate</li>
<li>Repeat steps 1 and 2 a large number of times (1000 times is reasonable)</li>
<li>Calculate the sample standard deviation of all the bootstrap replicate estimates obtained in step 3</li>
<li>The resulting quantity is called the bootstrap standard error</li>
</ul>
</section>
<section id="the-etymology-of-the-term-bootstrap" class="slide level2 smaller">
<h2>The etymology of the term ‘bootstrap’</h2>

<img data-src="images/week_2.030.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:100.0%"></section>
<section id="why-the-bootstrap-is-good" class="slide level2">
<h2>Why the bootstrap is good</h2>
<ul>
<li>Can be applied to almost any sample statistic
<ul>
<li>This includes means, proportions, correlations, regression</li>
</ul></li>
<li>Works when there is no ready formula for a standard error
<ul>
<li>For example the median, trimmed mean, correlation, eigenvalue, etc.</li>
</ul></li>
<li>Is nonparametric, so doesn’t require normally-distributed data</li>
<li>Works for estimates based on complicated sampling procedures or calculations
<ul>
<li>For example, it is used in phylogeny estimation</li>
</ul></li>
</ul>
</section>
<section id="r-interlude-bootstrapping-to-produce-a-confidence-interval" class="slide level2">
<h2>R INTERLUDE | Bootstrapping to produce a confidence interval</h2>
<p>Examine the <code>data_frames.R</code> script</p>
<p>Examine the <code>simple_boostrap.R</code> script</p>
<p>On your own - use R to figure out the bootstrap distribution for other parameters (such as variance).</p>
</section>
<section id="r-interlude-bootstrapping-to-produce-a-confidence-interval-1" class="slide level2 smaller">
<h2>R INTERLUDE | Bootstrapping to produce a confidence interval</h2>
<p><code>x &lt;- c(0.9, 1.2, 1.2, 1.3, 1.4, 1.4, 1.6, 1.6, 2.0, 2.0)</code></p>
<ul>
<li>Use R to make 1000 “pseudo-samples” of size 10 (with replacement), using a for loop as before.</li>
<li>Name the pseudo-sample object “xboot”, and name the means of the xboot samples “z”.</li>
<li>Plot the histogram of the resampled means, and calculate the standard deviation of the sample means (the bootstrap SEM) using the <code>sd()</code> function.</li>
<li>How does it compare with the ordinary standard error of the mean calculated from the original, real sample?</li>
</ul>
<p><code>sd(x)/sqrt(10)</code></p>
<ul>
<li>Now take one of the genes from the <code>GacuRNAseq_Subset.csv</code> data and obtain a bootstrapped estimate of the mean expression level.</li>
</ul>
</section>
<section id="r-interlude-bootstrapping-to-produce-a-confidence-interval-2" class="slide level2">
<h2>R INTERLUDE | Bootstrapping to produce a confidence interval</h2>
<ul>
<li>The 2.5th and 97.5th percentiles of the bootstrap sampling distribution are a passable 95% confidence interval</li>
<li>Note that no transformations or normality assumptions needed</li>
<li>You can use the <code>quantile()</code> function to calculate these</li>
<li><em>On your own</em> - use R to figure out the bootstrap distribution for other parameters (such as variance).</li>
</ul>
</section>
<section id="parameter-estimation-ordinary-least-squares-ols" class="slide level2">
<h2>Parameter Estimation | Ordinary Least Squares (OLS)</h2>
<ul>
<li>Algorithmic approach to parameter estimations</li>
<li>One of the oldest and best developed statistical approaches</li>
<li>Used extensively in linear models (ANOVA and regression)</li>
<li>By itself only produces a single best estimate (No C.I.’s)</li>
<li>Can use resampling approaches to get C.I.’s</li>
<li>Many OLS estimators have been duplicated by ML estimators</li>
</ul>
</section>
<section id="parameter-estimation-ordinary-least-squares-ols-1" class="slide level2 smaller">
<h2>Parameter Estimation | Ordinary Least Squares (OLS)</h2>

<img data-src="images/week_2.031.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="sampling-exercise" class="slide level2">
<h2>Sampling Exercise</h2>
<ul>
<li>Since we are working with simulated data, we can also afford to simulate our sampling!</li>
<li>Try taking a sample from our <code>true_pop</code> dataset and change the sample size, then calculate the mean and range for your sample and see how it compares to the true values.</li>
<li>How many college students are you including in your survey?</li>
</ul>
</section>
<section id="test-it-out" class="slide level2">
<h2>Test it out</h2>
<ul>
<li>Try changing the lambda of the original population, and see how the SD changes</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.24649</code></pre>
</div>
</div>
</section>
<section id="sampling-exercise-1" class="slide level2">
<h2>Sampling Exercise</h2>
<div class="cell">

<div class="cell-output cell-output-stdout">
<pre><code>[1] "Mean: " "4.95"  </code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Range: " "1"       "10"     </code></pre>
</div>
</div>
<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-72-1.png" style="width:50.0%" class="r-stretch"></section>
<section id="randomness-in-sampling" class="slide level2">
<h2>Randomness in Sampling</h2>
<ul>
<li>Because of the randomness of sampling, you may get close to the true estimates even with a small sample size - but your results will change each time you take a new sample of the same size</li>
<li>How do we get a feel for how accurate each sample size is? Or which sample size is recommended?</li>
</ul>
</section>
<section id="surveying-your-sampling" class="slide level2">
<h2>Surveying your Sampling</h2>

<img data-src="Week3a_files/figure-revealjs/unnamed-chunk-73-1.png" style="width:50.0%" class="r-stretch"></section>
<section id="surveying-your-sampling-1" class="slide level2">
<h2>Surveying your Sampling</h2>
<ul>
<li>We get close to the true mean (4.9) about 2/3rds of the time - is this good enough?</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
FALSE  TRUE 
   19    31 </code></pre>
</div>
</div>
</section>
<section id="surveying-your-sampling-2" class="slide level2">
<h2>Surveying your Sampling</h2>
<ul>
<li>This sampling variation is what we have to deal with, and account for, as empirical scientists.</li>
<li>If this had been a real-world scenario, we likely would be basing our estimate for the true mean on just a single sample mean.</li>
<li>Getting close to the idea of <strong>power</strong> - does our experimental design have the power to detect the parameters we are interested in?</li>
</ul>
</section></section>
<section>
<section id="sampling-distributions" class="title-slide slide level1 center">
<h1>Sampling Distributions</h1>
<ul>
<li>The previous exercise illustrates the concept of a sampling distribution.</li>
<li>We sampled over and over again (50 times) and calculated the mean for each sample to demonstrate the sampling distribution for the mean, our original parameter of interest.</li>
<li>One important point is that the sampling distribution for a given parameter is often very different from the variable’s distribution in the population. In many cases, the sampling distribution is normal or approximately so.</li>
</ul>
</section>
<section id="sampling-distributions---its-been-normal-this-whole-time" class="slide level2">
<h2>Sampling Distributions - It’s been Normal this whole time?!</h2>

<img data-src="images/week7_normalsampling.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section></section>
<section>
<section id="the-central-limit-theorem" class="title-slide slide level1 center">
<h1>The Central Limit Theorem</h1>

</section>
<section id="the-central-limit-theorem-1" class="slide level2">
<h2>The Central Limit Theorem</h2>
<ul>
<li>For most real world data sets we can’t empirically determine a sampling distribution by taking many actual samples, because we often have just the one sample.</li>
<li>Fortunately, we can rely on the <strong>Central Limit Theorem</strong> to make some assumptions about sampling distributions, particularly when estimating a mean from a single sample, or when estimating most any parameter using a “pseudo” or re-sampling process we refer to as “<strong>bootstrapping</strong>”</li>
</ul>
</section></section>
<section>
<section id="standard-error---the-range-of-the-sampling-distribution" class="title-slide slide level1 center">
<h1>Standard Error - the Range of the Sampling Distribution</h1>
<ul>
<li>The sampling distribution models all values we might have obtained from our sample and their probabilities of occurrence.</li>
<li>The standard error of an estimate can be conceptualized as the standard deviation of a sampling distribution.</li>
<li>So, whenever we obtain a parameter estimate, we need to include the standard error in some form or another, which is a measure of the precision of our estimate.</li>
</ul>
</section>
<section id="standard-error-1" class="slide level2">
<h2>Standard Error</h2>
<ul>
<li>The sample size and the spread of the distribution (range) - contribute to what is known as the standard error of a random variable.</li>
<li>The standard error for any given sample attribute (such as a sample mean), can be calculated either based on distributional assumptions, or by a process called “resampling.”</li>
</ul>
</section>
<section id="standard-error-2" class="slide level2">
<h2>Standard error</h2>

<img data-src="images/week_2.028.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="calculating-the-standard-error" class="slide level2">
<h2>Calculating the Standard Error</h2>
<ul>
<li>Standard Error of the Mean (SEM)</li>
<li>SEM = SD / sqrt(sample size)</li>
</ul>
</section>
<section id="calculating-the-standard-error-1" class="slide level2">
<h2>Calculating the Standard Error</h2>
<ul>
<li>Think conceptually - how will SEM change as sample size increases?</li>
</ul>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1299607</code></pre>
</div>
</div>
</section></section>
<section>
<section id="bootstrapping" class="title-slide slide level1 center">
<h1>Bootstrapping</h1>
<p>-Unfortunately, most other kinds of estimates (anything not the mean) do not have this amazing property, but we can rely on another approach to calculate the standard error. - This involves generating your own sampling distribution for the estimate using the “<strong>bootstrap</strong>,” a method invented by Efron (1979). - We call the bootstrap, and other methods that do not rely on distributional assumptions of the variable itself, “<strong>nonparametric</strong>” approaches.</p>
</section>
<section id="easy-steps-for-bootstrapping-in-r" class="slide level2">
<h2>Easy steps for bootstrapping in R</h2>
<ol type="1">
<li>Take a random sample (with replacement) from your sample data</li>
<li>Calculate the estimate using the measurements in the bootstrap sample (step 1). This is the first bootstrap replicate estimate</li>
<li>Repeat steps 1 and 2 a large number of times (1000 times is reasonable)</li>
<li>Calculate the sample standard deviation of all the bootstrap replicate estimates obtained in step 3 (SSD = sd(sample)/sqrt(sample size))</li>
</ol>
</section>
<section id="bootstrapping-1" class="slide level2">
<h2>Bootstrapping</h2>
<ul>
<li>The resulting quantity is called the “bootstrap standard error”</li>
<li>The bootstrap can be applied to almost any sample statistic, including means, proportions, correlations, and regression parameters.</li>
<li>It works when there is no ready formula for a standard error, for example when estimating the median, trimmed mean, correlation, eigenvalue, etc.</li>
<li>It is nonparametric, so doesn’t require normally-distributed data, as mentioned. - - It works well for parameter estimates that are based on complicated sampling procedures or calculations. For example, it is used to assess confidence in local relationships within phylogenetic trees.</li>
</ul>
</section>
<section id="confidence-intervals" class="slide level2">
<h2>Confidence Intervals</h2>
<ul>
<li>A confidence interval is a range of values about a parameter estimate, such that we are X% certain that the true population parameter value lies within that interval.</li>
<li>For now, know that for a normally distributed sample, a confidence interval about the population mean can be calculated using the t.test() function in base R.</li>
<li>The 95% confidence interval is commonly reported in statistical analysis results, by convention, but other values are occasionally reported as well.</li>
</ul>
</section></section>
<section>
<section id="relationship-between-mean-and-variance" class="title-slide slide level1 center">
<h1>Relationship between Mean and Variance</h1>
<ul>
<li>Population means and variances (and hence standard deviations) tend to have a strong, positive relationship.</li>
<li>So an otherwise similarly shaped distribution, but with a much larger mean, will by default have a much larger standard deviation</li>
</ul>
</section>
<section id="relationship-between-mean-and-variance-1" class="slide level2">
<h2>Relationship between mean and variance</h2>
<ul>
<li>This means it is inappropriate to compare variations of different populations with largely different means
<ul>
<li>For instance, comparing the standard deviation for a body measurement in a population of mice, with the same body measurement in a population of elephants is not meaningful.</li>
</ul></li>
</ul>
</section>
<section id="test-it-out-1" class="slide level2">
<h2>Test it out</h2>
<ul>
<li>Try changing the lambda of the original population, and see how the SD changes</li>
</ul>
</section>
<section id="coefficient-of-variation" class="slide level2">
<h2>Coefficient of Variation</h2>
<ul>
<li>To make standard deviations comparable across populations with very different means, we can instead compare a standardized metric called the “coefficient of variation” (CV), which is simply the sample standard deviation divided by the sample mean (and usually expressed as a % by multiplying by 100).</li>
</ul>
</section>
<section class="slide level2">


<div class="quarto-auto-generated-content">
<div class="footer footer-default">
<p>BioE_Stats_2025 - Knight Campus</p>
</div>
</div>
</section></section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'slow',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const isCodeAnnotation = (el) => {
        for (const clz of el.classList) {
          if (clz.startsWith('code-annotation-')) {                     
            return true;
          }
        }
        return false;
      }
      const onCopySuccess = function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      }
      const getTextToCopy = function(trigger) {
          const codeEl = trigger.previousElementSibling.cloneNode(true);
          for (const childEl of codeEl.children) {
            if (isCodeAnnotation(childEl)) {
              childEl.remove();
            }
          }
          return codeEl.innerText;
      }
      const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
        text: getTextToCopy
      });
      clipboard.on('success', onCopySuccess);
      if (window.document.getElementById('quarto-embedded-source-code-modal')) {
        // For code content inside modals, clipBoardJS needs to be initialized with a container option
        // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
        const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
          text: getTextToCopy,
          container: window.document.getElementById('quarto-embedded-source-code-modal')
        });
        clipboardModal.on('success', onCopySuccess);
      }
        var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
        var mailtoRegex = new RegExp(/^mailto:/);
          var filterRegex = new RegExp('/' + window.location.host + '/');
        var isInternal = (href) => {
            return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
        }
        // Inspect non-navigation links and adorn them if external
     	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
        for (var i=0; i<links.length; i++) {
          const link = links[i];
          if (!isInternal(link.href)) {
            // undo the damage that might have been done by quarto-nav.js in the case of
            // links that we want to consider external
            if (link.dataset.originalHref !== undefined) {
              link.href = link.dataset.originalHref;
            }
          }
        }
      function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
        const config = {
          allowHTML: true,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'light-border',
          placement: 'bottom-start',
        };
        if (contentFn) {
          config.content = contentFn;
        }
        if (onTriggerFn) {
          config.onTrigger = onTriggerFn;
        }
        if (onUntriggerFn) {
          config.onUntrigger = onUntriggerFn;
        }
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note) {
            return note.innerHTML;
          } else {
            return "";
          }
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>