<!DOCTYPE html>
<html lang="en"><head>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-html/tabby.min.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/light-border.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-de070a7b0ab54f8780927367ac907214.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.30">

  <meta name="author" content="Bill Cresko">
  <title>BioEngineering Statistics Spring 2025 – Week 8a - Statistics for Bioengineering</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="../site_libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for syntax highlighting */
    html { -webkit-text-size-adjust: 100%; }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; font-weight: bold; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
  </style>
  <link rel="stylesheet" href="../site_libs/revealjs/dist/theme/quarto-f563837468303362081e247dddd440d0.css">
  <link href="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="../site_libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Week 8a - Statistics for Bioengineering</h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Bill Cresko 
</div>
</div>
</div>

</section>
<section class="slide level2">

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.2     ✔ tibble    3.2.1
✔ lubridate 1.9.4     ✔ tidyr     1.3.1
✔ purrr     1.0.4     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors</code></pre>
</div>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a></a><span class="fu">library</span>(gt)</span>
<span id="cb3-2"><a></a><span class="fu">library</span>(readxl)</span>
<span id="cb3-3"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb3-4"><a></a><span class="fu">library</span>(tibble)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="this-week" class="slide level2 smaller">
<h2>This week</h2>
<ul>
<li><p>Key principles of experimental design</p></li>
<li><p>Linear Mixed Models (GLMMs)</p></li>
<li><p>Chi-square tests</p></li>
<li><p>Generalized Linear Models - Logistic and Poisson Regression</p></li>
</ul>
</section>
<section>
<section id="design-principles-for-planning-a-good-experiment" class="title-slide slide level1 center">
<h1>Design principles for planning a good experiment</h1>

</section>
<section id="what-is-an-experimental-study" class="slide level2 smaller">
<h2>What is an experimental study?</h2>
<ul>
<li>In an <strong>experimental study</strong> the researcher assigns treatments to units</li>
<li>In an <strong>observational study</strong> nature does the assigning of treatments to units</li>
<li>The crucial advantage of experiments derives from <strong>the random assignment of treatments to units</strong></li>
<li>Randomization <strong>minimizes the influence of confounding variables</strong></li>
<li>Allows us to infer <strong>cause and effect</strong> more easily (or at all)</li>
</ul>
</section>
<section id="mount-everest-example" class="slide level2 vcenter flexbox">
<h2>Mount Everest example</h2>
<p>Survival of climbers of Mount Everest is higher for individuals taking supplemental oxygen than those who don’t.</p>
<p><strong>Why?</strong></p>
</section>
<section id="mount-everest-example-1" class="slide level2 smaller">
<h2>Mount Everest example</h2>
<ul>
<li>One possibility is that supplemental oxygen (explanatory variable) really does cause higher survival (response variable).</li>
<li>The other is that the two variables are associated because other variables affect both supplemental oxygen and survival.</li>
<li>Use of supplemental oxygen might be a benign indicator of a greater overall preparedness of the climbers that use it.</li>
<li>Variables (like preparedness) that distort the causal relationship between the measured variables of interest (oxygen use and survival) are called <strong>confounding variables</strong></li>
</ul>
</section>
<section id="clinical-trials" class="slide level2 smaller">
<h2>Clinical Trials</h2>
<ul>
<li>The gold standard of experimental designs is the <strong>clinical trial</strong></li>
<li>Experimental design in all areas of biology have been informed by procedures used in clinical trials</li>
<li>A clinical trial is an experimental study in which two or more treatments are assigned to human subjects</li>
<li>The design of clinical trials has been refined because the cost of making a mistake with human subjects is so high</li>
<li>Experiments on nonhuman subjects are simply called “laboratory experiments”or “field experiments”</li>
</ul>
</section>
<section id="example-of-a-clinical-trial" class="slide level2 smaller">
<h2>Example of a clinical trial</h2>
<ul>
<li>Transmission of the HIV-1 virus via sex workers contributes to the rapid spread of AIDS in Africa</li>
<li>The spermicide nonoxynol-9 had shown <em>in vitro</em> activity against HIV-1, which motivated a clinical trial by van Damme et al.&nbsp;(2002).</li>
<li>They tested whether a vaginal gel containing the chemical would reduce the risk of acquiring the disease by female sex workers.</li>
<li>Data were gathered on a volunteer sample of 765 HIV-free sex-workers in six clinics in Asia and Africa.</li>
<li>Two gel treatments were assigned randomly to women at each clinic:
<ul>
<li>One gel contained nonoxynol-9</li>
<li>the other was a placebo.</li>
</ul></li>
<li>Neither the subjects nor the researchers making observations at the clinics knew who received the treatment and who got the placebo.</li>
</ul>
</section>
<section id="outcome-of-the-clinical-trial" class="slide level2 flexbox vcenter smaller">
<h2>Outcome of the clinical trial</h2>

<img data-src="images/images_6a.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"><ul>
<li>Note that these are count data but clinical trials could involve continuous data</li>
<li>We will use these data on Thursday for our Chi-square analyses</li>
<li>Also there is another variable - what is the nature of that variable?</li>
</ul>
</section>
<section id="design-components-of-a-clinical-trial" class="slide level2 smaller">
<h2>Design components of a clinical trial</h2>
<p>The goal of experimental design is to eliminate bias and to reduce sampling error when estimating and testing effects of one variable on another.</p>
<ul>
<li>To reduce bias, the experiment included:
<ul>
<li><strong>Simultaneous control group</strong>: study included both the treatment of interest and a control group (the women receiving the placebo).</li>
<li><strong>Randomization</strong>: treatments were randomly assigned to women at each clinic.</li>
<li><strong>Blinding</strong>: neither the subjects nor the clinicians knew which women were assigned which treatment.</li>
</ul></li>
<li>To reduce the effects of sampling error, the experiment included:
<ul>
<li><strong>Replication</strong>: study was carried out on multiple independent subjects.</li>
<li><strong>Balance</strong>: number of women was nearly equal in the two groups at every clinic.</li>
<li><strong>Blocking</strong>: subjects were grouped according to the clinic they attended, yielding multiple repetitions of the same experiment in different settings (“blocks”).</li>
</ul></li>
</ul>
</section>
<section id="simultaneous-control-group" class="slide level2 smaller">
<h2>1. Simultaneous control group</h2>
<ul>
<li>In clinical trials either a placebo or the currently accepted treatment should be provided.</li>
<li>In experiments requiring intrusive methods to administer treatment, such as
<ul>
<li>injections</li>
<li>surgery</li>
<li>restraint</li>
<li>confinement</li>
</ul></li>
<li>the control subjects should be perturbed in the same way as the other subjects, except for the treatment itself, as far as ethical considerations permit.</li>
<li>The “sham operation”, in which surgery is carried out without the experimental treatment itself, is an example.</li>
<li>In field experiments, applying a treatment of interest may physically disturb the plots receiving it and the surrounding areas, perhaps by trampling the ground by the researchers.</li>
<li>Ideally, the same disturbance should be applied to the control plots.</li>
</ul>
</section>
<section id="randomization" class="slide level2 smaller">
<h2>2. Randomization</h2>
<ul>
<li>The researcher should randomize assignment of treatments to units or subjects</li>
<li>Chance rather than conscious or unconscious decision determines which units end up receiving the treatment and which the control</li>
<li>Randomization breaks the association between possible confounding variables and the explanatory variable</li>
<li>Randomization doesn’t eliminate the variation contributed by confounding variables, only their correlation with treatment</li>
<li>Randomization ensures that variation from confounding variables is similar between the different treatment groups.</li>
</ul>
</section>
<section id="randomization-1" class="slide level2 smaller">
<h2>2. Randomization</h2>
<ul>
<li>Randomization should be carried out using a random process:
<ul>
<li>List all n subjects, one per row, in a computer spreadsheet.</li>
<li>Use the computer to give each individual a random number.</li>
<li>Assign treatment A to those subjects receiving the lowest numbers and treatment B to those with the highest numbers.</li>
</ul></li>
<li>Other ways of assigning treatments to subjects are almost always inferior because they do not eliminate the effects of confounding variables.</li>
<li>“Haphazard” assignment, in which the researcher chooses a treatment while trying to make it random, has repeatedly been shown to be non-random and prone to bias.</li>
</ul>
</section>
<section id="randomization-types" class="slide level2 smaller">
<h2>2. Randomization Types</h2>
<ul>
<li><p><strong>Completely randomized design</strong> - all subjects are placed to treatment or control with equal probability</p></li>
<li><p><strong>Randomized block</strong> - first broken into groups (e.g.&nbsp;age or gender) then assigned to treatment groups at random</p></li>
<li><p><strong>Matched pair design</strong> - subjects are paired by similarity before being randomly assigned to treatment groups</p></li>
</ul>
</section>
<section id="random-sampling-approaches" class="slide level2 smaller">
<h2>2. Random sampling approaches</h2>
<ul>
<li><strong>Simple random sample</strong> - every sample has equal probability of being chosen</li>
<li><strong>Stratified sample</strong> - subjects are randomly divided into groups then a simple random sample are taken from each</li>
<li><strong>Cluster sample</strong> - subjects are divided into similar groups, usually naturally occurring, and then a simple random samples of clusters is then taken and every member of the cluster is included in the sample</li>
<li><strong>Multistage sampling</strong> - First clusters are random sampled, and then random samples are take from each cluster.</li>
<li><strong>Systematic sample</strong> - members of a sample are chosen in a pre-determined ways. e.g Choose every 20th person coming into a store</li>
</ul>
</section>
<section id="blinding" class="slide level2 smaller">
<h2>3. Blinding</h2>
<ul>
<li><strong>Blinding</strong> is the process of concealing information from participants (sometimes including researchers) about which subjects receive which treatment.</li>
<li>Blinding prevents <strong>subjects</strong> and <strong>researchers</strong> from changing their behavior, consciously or unconsciously, as a result of knowing which treatment they were receiving or administering.</li>
<li>For example, studies showing that acupuncture has a significant effect on back pain are limited to those without blinding.</li>
<li>Reviews of medical studies have revealed that studies carried out without double- blinding exaggerated treatment effects by 16% on average compared with studies carried out with double-blinding (Jüni et al.&nbsp;2001).</li>
<li>Bebarta et al.(2003) reviewed 290 two-treatment experiments carried out on animals or on cell lines. The odds of detecting a positive effect of treatment were more than threefold higher in studies without blinding than in studies with blinding.</li>
<li>Blinding can be incorporated into experiments on nonhuman subjects using coded tags that identify the subject to a “blind” observer without revealing the treatment (and who measures units from different treatments in random order).</li>
</ul>
</section>
<section id="blinding-1" class="slide level2 smaller">
<h2>3. Blinding</h2>
<ul>
<li>In a <strong>single-blind experiment</strong>, the subjects are unaware of the treatment that they have been assigned.
<ul>
<li>Treatments must be indistinguishable to subjects, which prevents them from responding differently according to knowledge of treatment.</li>
<li>Blinding can also be a concern in non-human studies where animals respond to stimuli</li>
</ul></li>
<li>In a <strong>double-blind experiment</strong> the researchers administering the treatments and measuring the response are also unaware of which subjects are receiving which treatments
<ul>
<li>Researchers sometimes have pet hypotheses, and they might treat experimental subjects in different ways depending on their hopes for the outcome</li>
<li>Many response variables are difficult to measure and require some subjective interpretation, which makes the results prone to a bias</li>
<li>Researchers are naturally more interested in the treated subjects than the control subjects, and this increased attention can itself result in improved response</li>
</ul></li>
</ul>
</section>
<section id="replication" class="slide level2 smaller">
<h2>4. Replication</h2>
<ul>
<li><p>Replication is the assignment of each treatment to multiple, independent experimental units.</p></li>
<li><p>The goal of experiments is to estimate and test treatment effects against the background of variation between individuals (“noise”) caused by other variables</p></li>
<li><p>Without replication, we would not know whether response differences were due to the treatments or just chance differences between the treatments caused by other factors.</p></li>
<li><p>One way to reduce noise is to make the experimental conditions constant</p></li>
<li><p>However, highly constant experimental conditions might not be feasible nor desirable</p></li>
<li><p>By limiting the conditions of an experiment, we also limit the generality of the results</p></li>
<li><p>Another way to make treatment effects stand out is to include extreme treatments and to replicate the data.</p></li>
</ul>
</section>
<section id="replication-1" class="slide level2 smaller">
<h2>4. Replication</h2>
<ul>
<li>Studies that use more units (i.e.&nbsp;that have larger sample sizes) will have smaller standard errors and a higher probability of getting the correct answer from a hypothesis test.</li>
<li>Larger samples mean more information, and more information means better estimates and more powerful tests.</li>
<li>Replication is not about the number of plants or animals used, but the number of independent units in the experiment. An “experimental unit” is the independent unit to which treatments are assigned.</li>
<li>The figure shows three experimental designs used to compare plant growth under two temperature treatments (indicated by the shading of the pots). The first two designs are un-replicated.</li>
</ul>
</section>
<section id="pseudoreplication" class="slide level2 smaller">
<h2>4. (Pseudo)replication</h2>

<img data-src="images/images_6a.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="balance" class="slide level2 smaller">
<h2>5. Balance</h2>
<ul>
<li>A study design is balanced if all treatments have the same sample size.</li>
<li>Conversely, a design is unbalanced if there are unequal sample sizes between treatments.</li>
<li>Balance is a second way to reduce the influence of sampling error on estimation and hypothesis testing.</li>
<li>To appreciate this, look again at the equation for the standard error of the difference between two treatment means.</li>
<li>For a fixed total number of experimental units, n1 + n2, the standard error is smallest when n1 and n2 are equal.</li>
<li>Balance has other benefits. For example, ANOVA is more robust to departures from the assumption of equal variances when designs are balanced or nearly so.</li>
</ul>

<img data-src="images/images_6a.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="blocking" class="slide level2 smaller">
<h2>6. Blocking</h2>
<ul>
<li>Blocking is the grouping of experimental units that have similar properties. Within each block, treatments are randomly assigned to experimental units.</li>
<li>Blocking essentially repeats the same, completely randomized experiment multiple times, once for each block.</li>
<li>Differences between treatments are only evaluated within blocks, and in this way the component of variation arising from differences between blocks are controlled for.</li>
</ul>

<img data-src="images/images_6a.008.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:80.0%"></section>
<section id="what-if-you-cant-do-experiments" class="slide level2 smaller">
<h2>What if you can’t do experiments?</h2>
<ul>
<li>Experimental studies are not always feasible, in which case we must fall back upon observational studies.</li>
<li>The best observational studies incorporate as many of the features of good experimental design as possible to minimize bias (e.g., blinding) and the impact of sampling error (e.g., replication, balance, blocking, and even extreme treatments) except for one: randomization.</li>
<li>Randomization is out of the question, because in an observational study the researcher does not assign treatments to subjects.</li>
<li>Two strategies are used to limit the effects of confounding variables on a difference between treatments in a controlled observational study: matching; and adjusting for known confounding variables (covariates).</li>
<li>But beware - there could always be a co-variate that you didn’t include!</li>
</ul>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="general-linear-mixed-models" class="title-slide slide level1 center">
<h1>General Linear Mixed Models</h1>

</section>
<section id="what-is-a-glmm" class="slide level2">
<h2>What is a GLMM?</h2>
<ul>
<li>General Linear Mixed Model (<strong>GLMM</strong>)</li>
<li>A continuous response variable with a mix of continuous and categorical predictor variables</li>
<li>GLMM’s can be used to model random effects</li>
</ul>
</section>
<section id="attributes-of-mixed-effects-models" class="slide level2 flexbox vcenter smaller">
<h2>Attributes of mixed effects models</h2>
<ul>
<li>Linear models that include both fixed and random effects.</li>
<li>The model is split into fixed and random parts:
<ul>
<li>Fixed effects influence mean of the response variable Y.</li>
<li>Random effects influence the variance of Y.</li>
</ul></li>
<li>There is a different error variance for each level of grouping.</li>
<li>Estimation and testing is based on restricted maximum likelihood, which can handle unequal sample size.</li>
<li>P-values for fixed effects are conservative when design unbalanced.</li>
<li>We can use <code>lm</code> for simple models (which we’ll do today)</li>
<li>More complicated models are better performed using the <code>nlme</code> &amp; <code>lme4</code> packages in <code>R</code>.</li>
</ul>
</section>
<section id="assumptions-of-mixed-effects-models" class="slide level2 flexbox vcenter smaller">
<h2>Assumptions of mixed-effects models</h2>
<ul>
<li>Variation within groups follows a normal distribution with equal variance among groups.</li>
<li>Groups are randomly sampled from “population” of groups.</li>
<li>Group means follow a normal distribution.</li>
<li>Measurements within groups are independent.</li>
</ul>
</section>
<section id="hypotheses-for-model-3-anova-factorial-design-with-mixed-effects" class="slide level2 flexbox vcenter">
<h2>Hypotheses for Model 3 ANOVA Factorial Design With Mixed Effects</h2>

<img data-src="images/images_7b.003.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section></section>
<section>
<section id="parallel-slopes-mixed-model-aka-analysis-of-covariance-ancova" class="title-slide slide level1 center">
<h1>Parallel Slopes Mixed Model (aka Analysis of Covariance (ANCOVA))</h1>

</section>
<section id="brain-body-size-neaderthals-as-compared-to-humans" class="slide level2 flexbox vcenter">
<h2>Brain &amp; body size | neaderthals as compared to humans</h2>

<img data-src="images/images_7b.005.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="brain-body-size-neaderthals-as-compared-to-humans-1" class="slide level2 flexbox vcenter">
<h2>Brain &amp; body size | neaderthals as compared to humans</h2>

<img data-src="images/images_7b.006.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="brain-body-size-neaderthals-as-compared-to-humans-2" class="slide level2 flexbox vcenter">
<h2>Brain &amp; body size | neaderthals as compared to humans</h2>

<img data-src="images/images_7b.007.jpeg" class="quarto-figure quarto-figure-center r-stretch" style="width:90.0%"></section>
<section id="parallel-slopes-mixed-model" class="slide level2 flexbox vcenter">
<h2>Parallel Slopes Mixed Model</h2>
<ul>
<li>Mixture of regression and ANOVA</li>
<li>Response is still a normally distributed continuous variable</li>
<li>One or more continuous predictor variables (covariates)</li>
<li>Sometimes the covariates are of biological interest</li>
<li>Most often we want to remove unexplained variance</li>
<li>In this way they are similar to a blocking variable in ANOVA</li>
<li>Operationally, similar to regular ANOVA in which the group and overall means are replaced by group and overall relationships</li>
</ul>
</section>
<section id="parallel-slopes-model---r-interlude" class="slide level2">
<h2>Parallel Slopes Model - R Interlude</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb4-2"><a></a><span class="fu">library</span>(tibble)</span>
<span id="cb4-3"><a></a>heart_rate_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"mouse_rat_heart.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb4-4"><a></a></span>
<span id="cb4-5"><a></a><span class="fu">ggplot</span>(heart_rate_df, <span class="fu">aes</span>(<span class="at">x =</span> temp, <span class="at">y =</span> heart_rate, <span class="at">col =</span> species)) <span class="sc">+</span> </span>
<span id="cb4-6"><a></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb4-7"><a></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>

</div>
<img data-src="Week8a_files/figure-html/unnamed-chunk-11-1.png" width="672" class="r-stretch"></section>
<section id="parallel-slopes-model---r-interlude-1" class="slide level2 smaller">
<h2>Parallel Slopes Model - R Interlude</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb6-2"><a></a><span class="fu">library</span>(tibble)</span>
<span id="cb6-3"><a></a>heart_rate_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"mouse_rat_heart.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb6-4"><a></a></span>
<span id="cb6-5"><a></a>model_additive <span class="ot">&lt;-</span> <span class="fu">lm</span>(heart_rate <span class="sc">~</span> temp <span class="sc">+</span> species, <span class="at">data =</span> heart_rate_df)</span>
<span id="cb6-6"><a></a><span class="fu">summary</span>(model_additive)</span>
<span id="cb6-7"><a></a></span>
<span id="cb6-8"><a></a>new_heart_rate_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb6-9"><a></a>  <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"mouse"</span>, <span class="st">"mouse"</span>, <span class="st">"mouse"</span>, <span class="st">"rat"</span>, <span class="st">"rat"</span>, <span class="st">"rat"</span>),</span>
<span id="cb6-10"><a></a>  <span class="at">temp =</span> <span class="fu">c</span>(<span class="dv">24</span>, <span class="dv">26</span>, <span class="dv">28</span>, <span class="dv">24</span>, <span class="dv">26</span>, <span class="dv">28</span>)</span>
<span id="cb6-11"><a></a>)</span>
<span id="cb6-12"><a></a></span>
<span id="cb6-13"><a></a><span class="fu">predict</span>(model_additive, new_heart_rate_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = heart_rate ~ temp + species, data = heart_rate_df)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.0128 -1.1296 -0.3912  0.9650  3.7800 

Coefficients:
             Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)  -7.21091    2.55094  -2.827  0.00858 ** 
temp          3.60275    0.09729  37.032  &lt; 2e-16 ***
speciesrat  -10.06529    0.73526 -13.689 6.27e-14 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.786 on 28 degrees of freedom
Multiple R-squared:  0.9896,    Adjusted R-squared:  0.9888 
F-statistic:  1331 on 2 and 28 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>       1        2        3        4        5        6 
79.25516 86.46067 93.66617 69.18987 76.39538 83.60088 </code></pre>
</div>
</div>
</section>
<section id="parallel-slopes-model---r-interlude-2" class="slide level2 smaller">
<h2>Parallel Slopes Model - R Interlude</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb9-2"><a></a><span class="fu">library</span>(tibble)</span>
<span id="cb9-3"><a></a>heart_rate_df <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"mouse_rat_heart.csv"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb9-4"><a></a></span>
<span id="cb9-5"><a></a>model_interactive <span class="ot">&lt;-</span> <span class="fu">lm</span>(heart_rate <span class="sc">~</span> temp <span class="sc">*</span> species, <span class="at">data =</span> heart_rate_df)</span>
<span id="cb9-6"><a></a><span class="fu">summary</span>(model_interactive)</span>
<span id="cb9-7"><a></a></span>
<span id="cb9-8"><a></a>new_heart_rate_df <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb9-9"><a></a>  <span class="at">species =</span> <span class="fu">c</span>(<span class="st">"mouse"</span>, <span class="st">"mouse"</span>, <span class="st">"mouse"</span>, <span class="st">"rat"</span>, <span class="st">"rat"</span>, <span class="st">"rat"</span>),</span>
<span id="cb9-10"><a></a>  <span class="at">temp =</span> <span class="fu">c</span>(<span class="dv">24</span>, <span class="dv">26</span>, <span class="dv">28</span>, <span class="dv">24</span>, <span class="dv">26</span>, <span class="dv">28</span>)</span>
<span id="cb9-11"><a></a>)</span>
<span id="cb9-12"><a></a></span>
<span id="cb9-13"><a></a><span class="fu">predict</span>(model_interactive, new_heart_rate_df)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="parallel-slopes-model---r-interlude-3" class="slide level2 smaller">
<h2>Parallel Slopes Model - R Interlude</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = heart_rate ~ temp * species, data = heart_rate_df)

Residuals:
    Min      1Q  Median      3Q     Max 
-3.7031 -1.3417 -0.1235  0.8100  3.6330 

Coefficients:
                Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)     -11.0408     4.1515  -2.659    0.013 *  
temp              3.7514     0.1601  23.429   &lt;2e-16 ***
speciesrat       -4.3484     4.9617  -0.876    0.389    
temp:speciesrat  -0.2340     0.2009  -1.165    0.254    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 1.775 on 27 degrees of freedom
Multiple R-squared:  0.9901,    Adjusted R-squared:  0.989 
F-statistic: 898.9 on 3 and 27 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>       1        2        3        4        5        6 
78.99389 86.49678 93.99967 69.02983 76.06475 83.09967 </code></pre>
</div>
</div>
</section>
<section id="parallel-slopes-mixed-model-1" class="slide level2 smaller">
<h2>Parallel Slopes Mixed Model</h2>
<p><span class="math display">\[ y = \beta_0 + \beta_1x_1 + \beta_2x_2 + \epsilon \]</span> where</p>
<p><span class="math display">\[
y = \left\{
\begin{aligned}
    &amp; \beta_0 + \beta_1x_1 + \epsilon \\
    &amp;(\beta_0 + \beta_2) +\beta_1x_1 + \epsilon
\end{aligned}
\right.
\]</span></p>
<ul>
<li>Where the intercepts are different from one another but the slopes are not - R gives those estimates in that exact order</li>
</ul>
<p><span class="math display">\[
y = \left\{
\begin{aligned}
    &amp; -7.21091 + 3.60275x_1  \\
    &amp;(-7.21091 -10.06529) +3.60275x_1
\end{aligned}
\right.
\]</span></p>
</section></section>
<section>
<section id="interaction-mixed-model" class="title-slide slide level1 center">
<h1>Interaction Mixed Model</h1>

</section>
<section id="interaction-mixed-model-1" class="slide level2">
<h2>Interaction Mixed Model</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb12-2"><a></a><span class="fu">library</span>(modeldata)</span>
<span id="cb12-3"><a></a><span class="fu">theme_set</span>(<span class="fu">theme_minimal</span>())</span>
<span id="cb12-4"><a></a><span class="fu">data</span>(penguins)</span>
<span id="cb12-5"><a></a>penguins_sm <span class="ot">&lt;-</span> <span class="fu">filter</span>(penguins, species <span class="sc">!=</span> <span class="st">"Adelie"</span>)</span>
<span id="cb12-6"><a></a></span>
<span id="cb12-7"><a></a><span class="fu">ggplot</span>(penguins_sm, <span class="fu">aes</span>(<span class="at">x =</span> flipper_length_mm,</span>
<span id="cb12-8"><a></a>                        <span class="at">y =</span> body_mass_g,</span>
<span id="cb12-9"><a></a>                        <span class="at">col =</span> species)) <span class="sc">+</span> </span>
<span id="cb12-10"><a></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb12-11"><a></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="interaction-mixed-model-2" class="slide level2">
<h2>Interaction Mixed Model</h2>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>`geom_smooth()` using formula = 'y ~ x'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing non-finite outside the scale range
(`stat_smooth()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>

</div>
<img data-src="Week8a_files/figure-html/unnamed-chunk-17-1.png" width="672" class="r-stretch"></section>
<section id="interaction-mixed-model-3" class="slide level2">
<h2>Interaction Mixed Model</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb16-2"><a></a><span class="fu">library</span>(modeldata)</span>
<span id="cb16-3"><a></a><span class="fu">theme_set</span>(<span class="fu">theme_minimal</span>())</span>
<span id="cb16-4"><a></a><span class="fu">data</span>(penguins)</span>
<span id="cb16-5"><a></a>penguins_sm <span class="ot">&lt;-</span> <span class="fu">filter</span>(penguins, species <span class="sc">!=</span> <span class="st">"Adelie"</span>)</span>
<span id="cb16-6"><a></a></span>
<span id="cb16-7"><a></a>model <span class="ot">&lt;-</span> <span class="fu">lm</span>(body_mass_g <span class="sc">~</span> flipper_length_mm <span class="sc">*</span> species,</span>
<span id="cb16-8"><a></a>            <span class="at">data =</span> penguins_sm)</span>
<span id="cb16-9"><a></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="interaction-mixed-model-4" class="slide level2">
<h2>Interaction Mixed Model</h2>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
lm(formula = body_mass_g ~ flipper_length_mm * species, data = penguins_sm)

Residuals:
    Min      1Q  Median      3Q     Max 
-911.18 -215.38  -42.69  162.67 1015.71 

Coefficients:
                                 Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                     -3037.196   1138.679  -2.667  0.00832 ** 
flipper_length_mm                  34.573      5.811   5.950  1.3e-08 ***
speciesGentoo                   -3750.085   1534.769  -2.443  0.01548 *  
flipper_length_mm:speciesGentoo    20.049      7.496   2.674  0.00815 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 339.2 on 187 degrees of freedom
  (1 observation deleted due to missingness)
Multiple R-squared:  0.8205,    Adjusted R-squared:  0.8176 
F-statistic: 284.9 on 3 and 187 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
</div>
<ul>
<li><p>Make a dummy data set of values for Gentoo and Chinstrap Penguins</p></li>
<li><p>Use the interaction model that you fit to make new predictions for those values</p></li>
</ul>
</section>
<section id="interaction-mixed-model-5" class="slide level2 smaller">
<h2>Interaction Mixed Model</h2>
<p><span class="math display">\[
x_2 = \left\{
\begin{aligned}
    &amp; 0 \qquad \text{for Chinstrap} \\
    &amp; 1 \qquad \text{for Gentoo}
\end{aligned}
\right.
\]</span> <span class="math display">\[ y = \beta_0 + \beta_1x_1 + \beta_2x_2 + \beta_3x_1x_2 + \epsilon \]</span></p>
<p><span class="math display">\[
y = \left\{
\begin{aligned}
    &amp; \beta_0 + \beta_1x_1 + \epsilon \\
    &amp;(\beta_0 + \beta_2) +(\beta_1 +\beta_3)x_1 + \epsilon
\end{aligned}
\right.
\]</span> - Where the intercepts are different from one another but the slopes are not - Again R gives those estimates in that exact order</p>
<p><span class="math display">\[
y = \left\{
\begin{aligned}
    &amp; -3037.196 + 34.573x_1  \\
    &amp;(-3037.196 - 3750.085) + (34.573 + 20.049)x_1
\end{aligned}
\right.
\]</span></p>
</section>
<section class="slide level2">

</section></section>
<section>
<section id="more-complicated-models" class="title-slide slide level1 center">
<h1>More Complicated Models</h1>

</section>
<section id="how-do-we-factor-in-categorical-variables" class="slide level2">
<h2>How do we factor in categorical variables?</h2>
<ul>
<li>For a single categorical predictor, we can include effects of each factor level:</li>
</ul>
<p><span class="math display">\[ y = B_0 + B_1(x~level1) + B_2(x~level2) +~... ~+ e \]</span></p>
<ul>
<li>Each factor level (ex: for Wolbachia: Yes or No) becomes it’s own “effector” that will change the value of y</li>
<li>Instead of x being a continuous numerical value, for categorical data it will be either a 0 or 1</li>
</ul>
</section>
<section id="a-generalized-linear-model" class="slide level2">
<h2>A Generalized Linear Model</h2>
<ul>
<li>Check out the help page in RStudio for <code>glm()</code></li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a></a><span class="fu">glm</span>(<span class="at">formula =</span> y <span class="sc">~</span> x1 <span class="sc">+</span> x2, <span class="at">family =</span> <span class="fu">gaussian</span>(<span class="at">link =</span> <span class="st">"identity"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="first-steps-use-a-glm-to-test-wolbachia-infection" class="slide level2">
<h2>First steps: use a glm to test Wolbachia infection</h2>
<ul>
<li>Set up a <code>glm()</code> to test for the effect of Wolbachia infection on Recombinant Fraction</li>
</ul>
</section>
<section id="first-steps-use-a-glm-to-test-wolbachia-infection-1" class="slide level2">
<h2>First steps: use a glm to test Wolbachia infection</h2>
<ul>
<li>What do these components mean?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a></a>fly <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"Mostoufi2022_Recombination_Edit.csv"</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">','</span>)</span>
<span id="cb19-2"><a></a>model <span class="ot">&lt;-</span> <span class="fu">glm</span>(fly<span class="sc">$</span>RecombinantFraction <span class="sc">~</span> fly<span class="sc">$</span>Wolbachia, <span class="at">family =</span> <span class="fu">gaussian</span>(<span class="at">link=</span><span class="st">"identity"</span>))</span>
<span id="cb19-3"><a></a><span class="fu">summary</span>(model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = fly$RecombinantFraction ~ fly$Wolbachia, family = gaussian(link = "identity"))

Coefficients:
                 Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)      0.348054   0.005845  59.545   &lt;2e-16 ***
fly$WolbachiaYes 0.021664   0.008474   2.557   0.0112 *  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for gaussian family taken to be 0.00406577)

    Null deviance: 0.94137  on 226  degrees of freedom
Residual deviance: 0.91480  on 225  degrees of freedom
AIC: -601.48

Number of Fisher Scoring iterations: 2</code></pre>
</div>
</div>
</section>
<section id="interpreting-the-glm" class="slide level2">
<h2>Interpreting the GLM</h2>
<ul>
<li>Why is our intercept significant?
<ul>
<li>Remember that the null hypothesis is a model with an intercept and slope of 0</li>
</ul></li>
<li>How would you interpret the “WolbachiaYes” coefficient, especially the estimate and the p value?</li>
</ul>
</section>
<section id="second-steps---adding-more-variables" class="slide level2">
<h2>Second steps - adding more variables</h2>
<ul>
<li>Now let’s add Food to our model - how does this change our model results?</li>
</ul>
</section>
<section id="second-steps---adding-more-variables-1" class="slide level2">
<h2>Second steps - adding more variables</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a></a>model2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(fly<span class="sc">$</span>RecombinantFraction <span class="sc">~</span> fly<span class="sc">$</span>Wolbachia <span class="sc">+</span> fly<span class="sc">$</span>Food, <span class="at">family =</span> <span class="fu">gaussian</span>(<span class="at">link =</span> <span class="st">"identity"</span>))</span>
<span id="cb21-2"><a></a><span class="fu">summary</span>(model2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = fly$RecombinantFraction ~ fly$Wolbachia + fly$Food, 
    family = gaussian(link = "identity"))

Coefficients:
                 Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)      0.344691   0.008121  42.445   &lt;2e-16 ***
fly$WolbachiaYes 0.021731   0.008504   2.555   0.0113 *  
fly$FoodSucrose  0.002933   0.010481   0.280   0.7799    
fly$FoodYeast    0.007366   0.010128   0.727   0.4678    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for gaussian family taken to be 0.00409246)

    Null deviance: 0.94137  on 226  degrees of freedom
Residual deviance: 0.91262  on 223  degrees of freedom
AIC: -598.02

Number of Fisher Scoring iterations: 2</code></pre>
</div>
</div>
</section>
<section id="interpreting-the-glm-1" class="slide level2">
<h2>Interpreting the GLM</h2>
<ul>
<li>How would you interpret the Intercept and Food coefficients, especially the estimate and the p value?</li>
</ul>
</section>
<section id="interaction-effects-in-glms" class="slide level2">
<h2>Interaction effects in GLMs</h2>
<ul>
<li>You can also model interactions between two categorical variables in glms
<ul>
<li>What if variable x and variable z interact in non-additive ways?</li>
</ul></li>
<li>Use the notation x*z in the formula</li>
</ul>
</section>
<section id="third-steps---adding-interaction-effects" class="slide level2">
<h2>Third steps - adding interaction effects</h2>
<ul>
<li>Let’s finally add an interaction between Wolbachia infection and Food to represent Wolbachia titer in our model</li>
<li>Run the model - what are the results?</li>
</ul>
</section>
<section id="third-steps---adding-interaction-effects-1" class="slide level2">
<h2>Third steps - adding interaction effects</h2>
<ul>
<li>How do we interpret these findings?</li>
</ul>
<div class="cell">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a></a>model3 <span class="ot">&lt;-</span> <span class="fu">glm</span>(fly<span class="sc">$</span>RecombinantFraction <span class="sc">~</span> fly<span class="sc">$</span>Wolbachia <span class="sc">+</span> fly<span class="sc">$</span>Food <span class="sc">+</span> fly<span class="sc">$</span>Wolbachia<span class="sc">*</span>fly<span class="sc">$</span>Food, <span class="at">family =</span> <span class="fu">gaussian</span>(<span class="at">link =</span> <span class="st">"identity"</span>))</span>
<span id="cb23-2"><a></a><span class="fu">summary</span>(model3)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
Call:
glm(formula = fly$RecombinantFraction ~ fly$Wolbachia + fly$Food + 
    fly$Wolbachia * fly$Food, family = gaussian(link = "identity"))

Coefficients:
                                  Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)                       0.344520   0.009760  35.298   &lt;2e-16 ***
fly$WolbachiaYes                  0.022080   0.013970   1.581    0.115    
fly$FoodSucrose                   0.010286   0.014458   0.711    0.478    
fly$FoodYeast                     0.001256   0.014059   0.089    0.929    
fly$WolbachiaYes:fly$FoodSucrose -0.015873   0.021002  -0.756    0.451    
fly$WolbachiaYes:fly$FoodYeast    0.012910   0.020282   0.637    0.525    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

(Dispersion parameter for gaussian family taken to be 0.004096248)

    Null deviance: 0.94137  on 226  degrees of freedom
Residual deviance: 0.90527  on 221  degrees of freedom
AIC: -595.86

Number of Fisher Scoring iterations: 2</code></pre>
</div>
</div>
</section>
<section id="how-do-we-know-which-model-to-use" class="slide level2">
<h2>How do we know which model to use?</h2>
<ul>
<li>Want to be careful not to <em>overfit</em> your linear model (give too many variables)</li>
<li>The Akaike information criterion (AIC) is a metric for comparing different models to find the best fit
<ul>
<li>Notice the AIC score in your output?</li>
<li>For more on this, take Advanced Bio Stats! Or do some reading on the internet.</li>
</ul></li>
</ul>
</section>
<section id="r-interlude-variance-components-with-2-random-factors-using-lme4" class="slide level2 smaller">
<h2>R INTERLUDE | Variance components with 2 random factors using LME4</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a></a>rnadata <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">'RNAseq.tsv'</span>, <span class="at">header=</span>T, <span class="at">sep=</span><span class="st">''</span>)</span>
<span id="cb25-2"><a></a><span class="fu">head</span>(rnadata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>variables excluding first 5 and last 5 observations</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a></a>gene <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Gene80[<span class="dv">6</span><span class="sc">:</span><span class="dv">75</span>] </span>
<span id="cb26-2"><a></a>microbiota <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Microbiota[<span class="dv">6</span><span class="sc">:</span><span class="dv">75</span>]</span>
<span id="cb26-3"><a></a>genotype <span class="ot">&lt;-</span> rnadata<span class="sc">$</span>Genotype[<span class="dv">6</span><span class="sc">:</span><span class="dv">75</span>]</span>
<span id="cb26-4"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota)</span>
<span id="cb26-5"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> genotype)</span>
<span id="cb26-6"><a></a><span class="fu">boxplot</span>(gene <span class="sc">~</span> microbiota<span class="sc">*</span>genotype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Estimate the variance components using Restricted Maximum Likelihood (REML)</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a></a><span class="fu">library</span>(lme4)</span>
<span id="cb27-2"><a></a><span class="fu">lmer</span>(gene <span class="sc">~</span> <span class="dv">1</span> <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> microbiota) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> genotype) <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">|</span> microbiota<span class="sc">:</span>genotype))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Based on the REML sd estimates, what are the relative contributions of the factors to total variance in gene expression?</p>


</section></section>
    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="../site_libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="../site_libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="../site_libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="../site_libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="../site_libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="../site_libs/revealjs/plugin/notes/notes.js"></script>
  <script src="../site_libs/revealjs/plugin/search/search.js"></script>
  <script src="../site_libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="../site_libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'fade',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'slow',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>