---
title: "EDA, t-test and Power Calculations"
format: html
editor: visual
---

## Problem Set 1 - Directions

**Assignment:** Your task is to use R and Quarto Markdown to write a short report including the following tasks. You will use both the **microbiota/RNAseq data** set that you have been using in class, as well as **the dataset that you've identified for your term long project**. Try to accomplish a version of the tasks I lay out below on your own dataset, but you're undoubtedly going to need to make some modifications and adjustments. In addition, you will be performing one set of power analyses using simulation on a hypothetical problem

**Due:** Submit your work via Canvas by the end of the day (midnight) on **Tuesday, April 29th.** Please submit **both the Quarto_md file *and* the resulting html or pdf file**. You can work with other members of class, but I expect each of you to construct and run all of the scripts yourself.

## Problem 1 - Exploratory Data Analysis

### Read in the data

-   Read in the files
-   Use `str`, `head` and `glimpse` to get an idea of the data structure
-   Create a `tibble` data object using this dataset (the function is `as_tibble`)

### Perform some wrangling

-   `Select` all of the categorical variables and only 4 of the gene count variables and put them into a new data object.
-   `Mutate` each of the 4 gene expression values by performing a square root transformation making a new variable for each of the original (keep all eight variables in the dataset).
-   `Summarise` the mean and standard deviation for each of the gene count variables grouped by the ‘Microbiota’ and ‘Genotype’ categorical variables (hint, using `group_by` may make this easier; also remember your `na.rm=TRUE` to remove the `na` values)

### Graphical Communication with `ggplot2`

Insert a code chunk and complete the following tasks using your gene expression tibble:

-   Create a barplot of microbiota and genotype- what does this tell you about experimental design?

-   Create a histogram and frequency polygram of the expression of one particular gene.

-   Create a boxplot of the expression of one gene by microbiota/genotype combination. Reorder it by median expression and flip the coordinates.

-   Explore the coexpression patterns among your genes by plotting a couple pairwise combinations using `geom_point`. Plot your most interesting coexpression by treatment using `facet_wrap`.

## Problem 2 - Parametric t-test in R

-   Using R to make a dummy data set that contains one continuous and one categorical value with two levels. To do so draw individuals of the two different factor levels from normal distributions (check the 'norm' family of functions) with slightly different means but equal standard deviations. Take 100 observations per factor level.

-   Now, perform a t-test to see if the two populations are statistically different from one another (see below)

```{r, eval=FALSE, echo=TRUE}
boxplot(continuous_variable~cat_variable, dataset name) 
t.test(continuous_variable~cat_variable, dataset name) 
```

-   Repeat steps 1 and 2 above but use different sample sizes (e.g. 10 , then 100, then 1000, then 10,000 obsv. per factor level). How does the sample size affect the statistics?

-   What if you run the test again and make the means of the categorical groups more dissimilar?

-   What happens if you change the standard deviation of your normal distributions to be very small or very large?

-   Now perform the t-test using the dataset provided in class and your own data for two variables.

## Problem 3 - *a priori* power analyses via simulation

You have been asked by your colleagues to aid in the design of an experiment to determine how many mice might be needed to test the efficacy of a specific medical device on bone growth. The *in vivo* trials are expensive, and so the researchers would like your help to figure out **how many mice are needed** to achieve a reasonable **level of precision**. In particular, they are interested in both the number of microns that bones grow over a 3 month period, and whether mice that have the implant have a greater rate of growth than those that do not.

To help them, you should do a *power analysis*, and write up the results in a short, readable section. Your colleagues will be interested in the cost-precision tradeoff, so you should include graphical displays that convey both (a) how accurately the mean number microns of growth can be estimated, and (b) how likely it is that a real difference in mean between treated and control of a given, reasonable size will be detected (have a $p$-value less than 0.05), both as a function of sample size. I'm leaving it up to you precisely how to convey these things.

I'd like you to do this power analysis using *simulation*: by using the computer to do the experiment *in silico* in the case where you know the truth, and seeing how close your estimates are to the truth. To do this, you'll have to *make some stuff up*: for instance, plausible values for what the mean growth rate *is*, and what sort of mean difference between the treated and control that your colleagues might plausibly care about. You don't have to do independent research to find good values, but you should discuss the choices you made. You will also have to decide what a good range of sample sizes is to display your results across. I'll save you the burden of making *one* thing up: you can assume that the distribution of distances traveled is **Poisson distributed**, so you can simulate your fake surveys using the function `rpois( )`. Your final report for this section should be readable and *not* include visible code in the compiled version. However, you should explain clearly in words what you did to arrive at your answers.

The purposes of this section of the homework are to practice writing reports in Quarto markdown; doing simulation-based power analyses, and communicating results in plain language.
