---
title: "Linear Regression and ANOVA"
format: html
editor: visual
---

## Problem Set 2 - Directions

**Assignment:** Your task is to use R and Quarto Markdown to write a short report including the following tasks. You will use a new data set of **effects of IGF-1 on growth rate in mouse (`igf.tsv`)**, and a simplified **stickleback microbiota/RNAseq data (`RNAseq_microbiome.csv`)** set that you have been using in class. In addition, perform either a simple linear regression (2.1) or an ANOVA (2.2) on **the dataset that you've identified for your term long project** depending on the nature of your data.

**Due:** Submit your work via Canvas by the end of the day (midnight) on **Wednesday, May 14th.** Please submit **both the Quarto_md file *and* the resulting html or pdf file**. You can work with other members of class, but I expect each of you to construct and run all of the scripts yourself.

## Problem 2.1 : Multiple linear regression to examine the effect of IGF-1 on the early growth of mice

**The Study**. Researchers are studying how levels of the hormone IGF-1 (insulin-like growth factor 1) affect early growth, and have measured IGF-1 levels along with length, weight, and a few maturity-related traits in 710 young mice. Your goal is to investigate how IGF-1 is related to differences in length after accounting for other factors.

The dataset (`igf.tsv`) can be found in the data folder on Canvas. This dataset has the following variables

-   `age` (in weeks),
-   `sex`
-   `igf` (IGF-1 levels),
-   `maturity` (1-5, larger values are more sexually mature),
-   `weight` (in g), and
-   `length` (body and tail, in cm).

Perform the following steps:

1.  Describe the data using your skills in exploratory data analysis. Which variables are categorical, which are continuous? Are they coded correctly in r, or do you need to coerce them to a different data type?

2.  Fit two linear models, one with `weight` as the response and the other with `length` as a response, and both with `igf` as the predictor variable.

3.  Summarize the fitted model in terms of key statistics such as $R^2$, F-value, degrees of freedom and p-values

4.  Perform a residual analysis to see if you have satisfied the assumptions of parametric linear regression, and identify outliers that might have a significant effect on the model

5.  Provide visualizations and summaries of what you find.

------------------------------------------------------------------------

## Exercise 2.2: Factorial ANOVA examining the effects of microbiome treatment on RNA expression levels in the guts of threespine stickleback

Load the `RNAseq_microbiome.csv` dataset which represents RNAseq data collected for three different genes (FGF3, Sox10 and SLC5) from the guts of threespine stickleback fish. In this experiment two different genotypes of stickleback (ocean and freshwater) were tested by subjecting each to two different microbiome treatments (conventional or mono-associated). Stickleback can be made germ free and then either exposed to the conventional microbiome or just one microbe. After exposure the guts were extracted, RNA isolated, and RNAseq data collected.

The dataset (`igf.tsv`) can be found in the data folder on Canvas. This dataset has the following variables:

-   `Genotype` - whether the fish has an oceanic or freshwater genotype
-   `Microbiota` - whether the fish was germ free or conventionalized
-   `FGF3` - the level of gene expression of the signalling molecule Fibroblast Growth Factor 3
-   `Sox10` - the level of gene expression of the transcription factor gene Sox10
-   `SLC5` - the level of gene expression of the small molecule transporter SLC5

Is `R` correctly interpreting the class of your variables? It is always good to check this before doing anything.

1.  Make boxplots to graphically illustrate differences in gene expression distributions among genotype and microbiota treatment for each gene.

-   Use the `mutate` function in `dpylr` to combine the two categorical variables into a new variable
-   Any obvious patterns?
-   Would you guess there is a difference among means?
-   Does it look like there may be an interaction?

2.  Fit the factorial linear model

-   Use the summary function to look at your results.
-   Is there a statistically significant model?
-   Is there a significant interaction between genotype and treatment?

3.  Perform a Tukey's post-hoc means test across treatment groups

-   Perform the test separately for each of the two levels for each categorical variable
-   Now perform it on your new variable with four levels and rerun the Tukey's test
-   Do all groups differ from one another?
